# Claudia Samples

## Experiment

## Metadata

```{r}
designdf <- read.table("data/sample_mapping.tsv", header = T, sep = "\t")
DT::datatable(designdf)
```

## OligoMM

|  ID | phylum | species | 
|------|-------| -------|
| YL44 | _Verrucomicrobia_ | _A. muciniphila_ |
| I48 |  _Bacteroidetes_ | _B. caecimuris_ |
| YL27 | _Bacteroidetes_ | _M. intestinale_ |
| YL45 | _Proteobacteria_ | _T. muris_ |
| YL2 | _Actinobacteria_ | _B. longum_ |
| KB1 | _Firmicutes_ | _E. faecalis_  |
| KB18 | _Firmicutes_ | _A. muris_ |
| YL32 | _Firmicutes_ | _C. clostridioforme_ |
| YL31 | _Firmicutes_ | _F. plautii_ |
| YL58 | _Firmicutes_ | _B. coccoides_ |
| I49 | _Firmicutes_ | _L. reuteri_ |
| I46 | _Firmicutes_ | _C. innocuum_ |

## Load in variants

```{r}
require(data.table)
source("utils.R")
vcftodataframe <- function(vcf_files, contig_mapping = contig_mapping, 
  gff_df = gff_df) {
	require(vcfR)
	res <- list()
	for (file in vcf_files) {
   # message(file)
		vcf_content <- vcfR::read.vcfR(file, verbose = FALSE)
		vcf_fix <- as.data.frame(vcf_content@fix) 
		vcf_info <- vcfR::INFO2df(vcf_content) # contains DP and AF info
		if (nrow(vcf_fix) > 0) { # there are variants
		    dat <-  as.data.frame(cbind(vcf_fix[,c(1, 2, 4, 5, 6)],
          vcf_info[,c(1, 2)]))
			dat$majorAF <- sapply(dat$AF, minorAfToMajorAf)
      dat$dp <- as.numeric(as.matrix(vcf_info$DP))
			dat$genome <- contig_mapping[match(dat$CHROM, 
        contig_mapping$contig),]$genome
			dat$genome_hr <- translateGenomeIdToFullName(tolower(dat$genome))
			dat$mouse.id <-  substr(tools::file_path_sans_ext(basename(file)), 1, 4)
      # add studz type specific annotations
      dat$mouse.group <- designdf[match(dat$mouse.id, designdf$mouse.id),]$desc
      dat$day <- designdf[match(dat$mouse.id, designdf$mouse.id),]$day
      dat$generation <- designdf[match(dat$mouse.id, designdf$mouse.id),]$generation
      dat$ecoli <- designdf[match(dat$mouse.id, designdf$mouse.id),]$ecoli
      dat$sample <- tools::file_path_sans_ext(basename(file))
			# annotate overlay of gene
			dt_gff <- data.table(start = gff_df$start, end = gff_df$end,
				chr = as.character(as.matrix(gff_df$chr)), feature = gff_df$product)
			colnames(dat)[1:2] <- c("chr", "start")
			dat$start <- as.integer(as.matrix(dat$start))
			dat$chr <- as.character(as.matrix(dat$chr))
			dat$end <- dat$start
			dat2 <- as.data.table(dat)
			setkey(dt_gff, chr, start, end)
			annotated <- foverlaps(dat2, dt_gff, type="within", mult="first")
			res[[tools::file_path_sans_ext(basename(file))]] <- annotated # add vcf df to list
		} else{
			message("Skipping")
		}
	}
	df <- as.data.frame(do.call(rbind, res)) # merge list to df
	return(df)
}
```

Merge vcf and annotate with metadata

```{r}
# load in reference information
gff_files <- Sys.glob("data/references/joined_reference_curated_ecoli/*.gff")
gff_df <- NULL
for (gff_file in gff_files){
	message(gff_file)
	gff <- rtracklayer::readGFF(gff_file)
	# subset since different columns are present on gff files
	relevant <- data.frame(start = gff$start, end = gff$end,
    type = as.character(as.matrix(gff$type)), 
    gene = as.character(as.matrix(gff$gene)),
    product = as.character(as.matrix(gff$product)),
    chr = as.character(as.matrix(gff$seqid)))
	relevant$genome <-  substr(basename(gff_file),
    1, nchar(basename(gff_file)) - 4)
	gff_df <- rbind(gff_df, relevant)
}

# load in contig information
contig_mapping <- read.csv2("data/contig_mapping_new_ref.csv", 
  sep =";", header=T, stringsAsFactors = F)

# load in vcf files
vcf_files <- Sys.glob("out_claudia/all_vcf/*.vcf")
vcf_samples <- suppressWarnings(vcftodataframe(vcf_files,
  contig_mapping, gff_df = gff_df))
vcf_samples$feature <- as.character(as.matrix(vcf_samples$feature)) 

vcf_samples[which(is.na(vcf_samples$feature)),]$feature <- "outside ORFs"
vcf_samples $start <- NULL
vcf_samples $end <- NULL
vcf_samples $i.end <- NULL
colnames(vcf_samples)[3] <- "POS"

vcf_samples$ref_size <- nchar(as.character(as.matrix(vcf_samples$REF)))
vcf_samples$alt_size <- nchar(as.character(as.matrix(vcf_samples$ALT)))
vcf_samples$alteration <- paste(as.character(vcf_samples$REF), "->",as.character(vcf_samples$ALT))
vcf_samples$alteration_type <- "SNP"
vcf_samples[which(vcf_samples$ref_size < vcf_samples$alt_size), ]$alteration_type <- "insertion"
vcf_samples[which(vcf_samples$ref_size > vcf_samples$alt_size), ]$alteration_type <- "deletion"
saveRDS(vcf_samples, file = "data/rds/omm_claudia_new.rds")
```

## Filter out of abnormal high mutation

We filter out samples that have a mutation rate of the global mean. 

```{r, fig.cap="Mutation profile before removal. Vertical line is the global mean number of mutation rates", fig.width=9, fig.height=15}
dat <- readRDS("data/rds/omm_claudia_new.rds")
dat <- dat[which(dat$alteration_type == "SNP"),]
dat$dummy <- 1
# summarize by alteration type
dat.agg <- aggregate(dummy ~ alteration + sample, dat, sum)
dat.agg <- dat.agg[order(-dat.agg$dummy),]

median_threshold <- mean(dat.agg$dummy)

p <- ggplot(dat.agg, aes(x = reorder(sample, dummy), y = dummy, group = alteration)) + ylab("number of SNPs (log10)") + xlab("samples")
p <- p + geom_bar(stat = "identity") + coord_flip() + facet_wrap(~ alteration) + scale_y_log10()
p <- p + theme_bw() + geom_hline(yintercept=median_threshold, colour="grey50")
print(p)

# remove all C>A and G>T on selected samples
affected_samples <- dat.agg[which(dat.agg$dummy > median_threshold),]$sample 
dat <- readRDS("data/rds/omm_claudia_new.rds")
dat_outlier <- which(dat$alteration_type == "SNP" &
	dat$sample %in% affected_samples &
	(dat$alteration == "C -> A" | dat$alteration == "G -> T"))
dat_corrected <- dat[-dat_outlier,]
saveRDS(dat_corrected, file = "data/rds/omm_claudia_new.rds")
nrow(dat_corrected) # number of variants in total
```

```{r, fig.cap="Mutation profile after removal. Vertical line is the global mean number of mutation rates befor filtering", fig.align="center", fig.width=9, fig.height=15}
# plot again
dat_corrected2 <- dat_corrected[which(dat_corrected$alteration_type == "SNP"),]
dat_corrected2$dummy <- 1
# summarize by alteration type
dat_corrected2.agg <- aggregate(dummy ~ alteration + sample, dat_corrected2, sum)

p <- ggplot(dat_corrected2.agg, aes(x = reorder(sample, dummy), y = dummy, group = alteration)) + ylab("number of SNPs (log10)") + xlab("samples")
p <- p + geom_bar(stat = "identity") + coord_flip() + facet_wrap(~ alteration) + scale_y_log10()
p <- p + theme_bw() + geom_hline(yintercept=median_threshold, colour="grey50")
print(p)
```

## AF frequency

```{r, fig.cap="AF of resequenced strains", fig.align="center", fig.width=9, fig.height=15}
dat <- readRDS("data/rds/omm_claudia_new.rds")
library(scales)

p <- ggplot(dat, aes(AF, fill = mouse.group)) + geom_histogram()
p <- p + facet_wrap(~ genome + genome_hr, scales= "free", ncol = 3)
p <- p + xlab("AF") + ylab("occurence") + theme_minimal() 
plotly::ggplotly(p)
```

```{r, fig.cap="major AF of resequenced strains", fig.align="center", fig.width=9, fig.height=15}
dat <- readRDS("data/rds/omm_claudia_new.rds")
p <- ggplot(dat, aes(majorAF, fill = mouse.group)) + geom_histogram()
p <- p + theme_minimal() 
p <- p + facet_wrap(~ genome+ genome_hr, scales= "free", ncol = 3)
p <- p + xlab("AF") + ylab("occurence")
plotly::ggplotly(p)
```

## number of variants per group

```{r}
dat <- readRDS("data/rds/omm_claudia_new.rds")
dat$variants <- 1
dat.agg <- aggregate(variants ~ mouse.id + alteration_type + genome_hr, dat, sum)
DT::datatable(dat.agg)
```

### number of variants per treatment group 

### deletion
```{r, fig.align="center", fig.width=14, fig.height=14, fig.margin=TRUE, fig.cap="total number of variants of all 12 OMM genomes by mouse and grouyp stratified by variant type. Seems there is still a outlier (20 cecal content caecimuris, where some samples have more than 1000 varaints)"}
dat <- readRDS("data/rds/omm_claudia_new.rds")
dat$variants <- 1
dat.agg <- aggregate(variants ~ mouse.id + mouse.group + alteration_type+ genome_hr, dat, sum)
p <- ggplot(dat.agg, aes(x = reorder(mouse.id, -variants) , y = variants, color = alteration_type, group = alteration_type, shape = factor(alteration_type)))
p <- p + geom_jitter(size = .4) + facet_grid(genome_hr ~ mouse.group,space = "free", scales = "free_x")
p <- p + geom_line() + scale_y_log10()
p <- p  + theme_minimal() + ylab("number variants")
p <- p + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
p
```

## Heatmap


```{r heatmap-claudia-1, fig.align="center", fig.width=9, fig.height=10, fig.margin=TRUE}
library(circlize)
library(ComplexHeatmap)

dat <- readRDS("data/rds/omm_claudia_new.rds")
dat$variant.id <- paste0(dat$POS, "-", dat$REF, "-", dat$ALT)
data.wide <- dcast(dat, variant.id ~ sample, value.var = "AF")
data.wide[is.na(data.wide)] <- 0
rownames(data.wide) <-  data.wide$variant.id
data.wide$variant.id <- NULL

heat <- data.matrix(data.wide)
# limit to variants that are present in at least 10% of samples
heat_num <- rowSums(heat != 0)
heat2 <- heat[which(heat_num > ncol(heat)/10),]

dat$dummy <- 1
annot.data <- aggregate(dummy ~ sample + mouse.group + day, dat, sum)

heat3.day <- annot.data[match(colnames(heat2), annot.data$sample),]$day
heat3.mouse.group <- annot.data[match(colnames(heat2), annot.data$sample),]$mouse.group


#data.wide.sub <- dat[match(colnames(heat3), dat$sample.id),]

col_fun = colorRamp2(c(0, 0.5, 1), c("white", "yellow", "red"))

Heatmap(heat2
, name = "AF",  border = TRUE,  col = col_fun,
right_annotation = rowAnnotation(prev = anno_lines(rowSums(heat2))),
cluster_columns = T,
 row_gap = unit(0, "mm"), column_gap = unit(0, "mm"),
  column_names_gp = gpar(fontsize =5),
 row_names_gp = gpar(fontsize = 3),
 show_row_dend = F,
  show_row_names = F,
 show_column_dend = T,

top_annotation = HeatmapAnnotation(day=anno_simple(heat3.day),
mouse_group = anno_simple(heat3.mouse.group))
)
```
