# Variant statistics {#Variant_statistics}

## Process of _raw_ data

Since the raw output of the **Snakemake** pipeline produces one `vcf` file per sample (including all contigs of the OligoMM reference genomes) we now produce a single `rds` file that contains the merged variant information of the whole study.

We define a function that iterates over a set of `vcf` files (`vcf.files`)

```{r process-vcf1}
#' Processes a list of vcf files
#' 
#' @param vcf.files a list of file path of files in vcf format
#' @param contig_mapping mapping file of vcf CHR to OMM reference genomes
#' @param gff.dff merged gff data from reference genomes
#' @return a dataframe holding the merged annotated variant information
vcfToDataframe <- function(vcf.files, contig_mapping = read.csv2("data/contig_genome_mapping.csv"), gff.df = gff.df) {
	require(vcfR)
	res <- list()
	for (file in vcf.files) {
		vcf.content <- vcfR::read.vcfR(file, verbose = FALSE )
		vcf.fix <- as.data.frame(vcf.content@fix) # contains chr, position and substitution informations
		vcf.info <- vcfR::INFO2df(vcf.content) # get INFO field, contains DP, AF informations
	#  vcf.ann <- data.frame(do.call('rbind', strsplit(as.character(vcf.info$ANN),'|',fixed = TRUE))) # split ANN field, contains information if files are processed by snpEff
		if(nrow(vcf.fix) > 0) {
			dat <-  as.data.frame(cbind(vcf.fix[,c(1, 2, 4, 5, 6)], vcf.info[,c(1, 2)]))
			dat$majorAF <- sapply(dat$AF, minorAfToMajorAf) # transforms e.g. AF of 0.1 to 0.9, 0.9 stays 0.9 and 0.5 stays 0.5
			dat$genome <- contig_mapping[match(dat$CHROM, contig_mapping$contig),]$genome # map chr information to genome name e.g. NHMU01000001.1 -> i48
			dat$genome_hr <- translateGenomeIdToFullName(dat$genome)
			dat$mouse.id <- substr(basename(file), 1, 4) # get mouse ID from file name, e.g. "1681"
			dat$dp <- as.numeric(as.matrix(vcf.info$DP))

			# annotate overlay of gene
			dat$genome_desc <- apply(dat, 1, getGeneByPosition, gff = gff, pos.column = 2, chr.column = 1)
			res[[basename(file)]] <- dat # add vcf df to list
		}
	}
	df <- as.data.frame(do.call(rbind, res)) # merge list to df
	return(df)
}
```

We apply this function on all  `r length(Sys.glob("data/raw/lofreq/*.vcf"))` files that live in `data/raw/lofreq/*.vcf`

```{r, results="hide", message=FALSE}
# merge gff annotaions
gff.files <- Sys.glob("data/annotation/gff/*.gff")
gff.df <- NULL
for (gff.file in gff.files){
	message(gff.file)
	gff <- rtracklayer::readGFF(gff.file)
	# subset since different columns are present on gff files
	relevant <- data.frame(start = gff$start, end = gff$end, type = as.character(as.matrix(gff$type)), gene = as.character(as.matrix(gff$gene)), product = as.character(as.matrix(gff$product)), chr = as.character(as.matrix(gff$seqid)))
	relevant$genome <-  substr(basename(gff.file), 1, nchar(basename(gff.file))-4)
	gff.df <- rbind(gff.df, relevant)
}

contig_mapping <- read.csv2("data/contig_genome_mapping.csv", sep =";", header=T, stringsAsFactors = F) # this file contains contig names of the 12 OligoMM genomes
vcf.files <- Sys.glob("data/raw/lofreq/*.vcf")
dat <- suppressWarnings(vcfToDataframe(vcf.files, contig_mapping, gff.df = gff.df))
# annotate study metadata
dat <- merge(dat, design.df, by = "mouse.id")
dat$mouse.id <- as.character(as.matrix(dat$mouse.id))
saveRDS(dat, file = "data/rds/variants.rds")
```

The dataset is written to `data/rds/variants.rds`).

## Number of variants 

### by sample

only non-ecoli samples

```{r load-vcf}
dat <- readRDS("data/rds/variants.rds")
dat <- dat[which(dat$ecoli== FALSE),]
```

This set contains `r nrow(dat)` variants found in `r length(unique(dat$genome))` genomes. The detailed breakdow of variants per genome is as follows.

```{r}
dat$dummy <- 1
dat.by.sample.genome <- aggregate(dummy ~ mouse.id + genome, dat, sum)
colnames(dat.by.sample.genome) <- c("sample", "genome", "number_variants")
DT::datatable(dat.by.sample.genome)
```

```{r, fig.cap="Number of variants by genome (color) found in samples"}
p <- ggplot(dat.by.sample.genome, aes(x = reorder(sample, number_variants), y = number_variants, fill = genome))
p <- p + geom_bar(stat="identity") + coord_flip() + theme_classic()
p <- p + ylab("number of variants") + xlab("sample") + scale_fill_manual(values = omm_colors)
plotly::ggplotly(p)
```

### by mouse and genome

only non-ecoli samples


```{r}
dat$dummy <- 1
dat.by.sample <- aggregate(dummy ~ mouse.id, dat, sum)
colnames(dat.by.sample) <- c("sample", "number_variants")
DT::datatable(dat.by.sample)
```

```{r, fig.align="center", fig.cap="number of variants of all 12 OMM genomes by mouse"}
p <- ggplot(dat.by.sample, aes(x = reorder(sample, number_variants), y = number_variants))
p <- p + geom_bar(stat="identity") + coord_flip() + theme_classic()
p <- p + xlab("number of variants") + ylab("sample")
plotly::ggplotly(p)
```

### by extendet metadata

only non-ecoli samples

```{r}
dat$dummy <- 1
dat.by.sample.genome2 <- aggregate(dummy ~ mouse.id + genome + generation + ecoli + desc + day, dat, sum)
colnames(dat.by.sample.genome2) <- c("sample", "genome", "generation", "ecoli", "desc","day","number_variants")
summary(dat$DP)
```
DP is the filtered depth, at the sample level. This gives you the number of filtered reads that support each of the reported alleles. You can check the variant caller’s documentation to see which filters are applied by default. Only reads that passed the variant caller’s filters are included in this number. However, unlike the AD calculation, uninformative reads are included in DP.


```{r}
DT::datatable(dat.by.sample.genome2)
dat$DP <- NULL
```

```{r, fig.cap="Number of variants by genome (color) found in samples"}
#dat.by.sample.genome2 <- dat.by.sample.genome2[which(dat.by.sample.genome2$sample != "I_cc" & dat.by.sample.genome2$sample != "I_mi"),]
p <- ggplot(dat.by.sample.genome2, aes(x = sample, y = number_variants, fill = genome))
p <- p + geom_bar(stat="identity") + theme_classic()
p <- p + facet_grid(.~ generation + day, scales = "free_x")
p <- p + ylab("number of variants") + xlab("sample") + scale_fill_manual(values = omm_colors)
print(p)
#plotly::ggplotly(p)
```


