<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Section 4 Introduction | 01-load-data</title>
  <meta name="description" content="Section 4 Introduction | 01-load-data" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="Section 4 Introduction | 01-load-data" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Section 4 Introduction | 01-load-data" />
  
  
  

<meta name="author" content="Philipp C. MÃ¼nch" />


<meta name="date" content="2020-03-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="number-of-variants.html"/>
<link rel="next" href="literature.html"/>
<script src="libs/header-attrs-2.1.1/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.12/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<script src="libs/plotly-binding-4.9.2/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-1.52.2/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-1.52.2/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Analysis Report</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#Data_structures"><i class="fa fa-check"></i><b>1.1</b> Data structure</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="loading-raw-variant-data.html"><a href="loading-raw-variant-data.html"><i class="fa fa-check"></i><b>2</b> Loading raw variant data</a></li>
<li class="chapter" data-level="3" data-path="number-of-variants.html"><a href="number-of-variants.html"><i class="fa fa-check"></i><b>3</b> Number of variants</a>
<ul>
<li class="chapter" data-level="3.1" data-path="number-of-variants.html"><a href="number-of-variants.html#number-of-variants-1"><i class="fa fa-check"></i><b>3.1</b> number of variants</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="number-of-variants.html"><a href="number-of-variants.html#by-mouse-and-genome"><i class="fa fa-check"></i><b>3.1.1</b> by mouse and genome</a></li>
<li class="chapter" data-level="3.1.2" data-path="number-of-variants.html"><a href="number-of-variants.html#by-mouse"><i class="fa fa-check"></i><b>3.1.2</b> by mouse</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="number-of-variants.html"><a href="number-of-variants.html#major-allele-frequency"><i class="fa fa-check"></i><b>3.2</b> major allele frequency</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>4</b> Introduction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="intro.html"><a href="intro.html#functions"><i class="fa fa-check"></i><b>4.1</b> functions</a></li>
<li class="chapter" data-level="4.2" data-path="intro.html"><a href="intro.html#iterate-over-files"><i class="fa fa-check"></i><b>4.2</b> iterate over files</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="intro.html"><a href="intro.html#annotate-variants-based-on-gff-file-information-of-reference-genomes"><i class="fa fa-check"></i><b>4.2.1</b> annotate variants based on <em>gff</em> file information of reference genomes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="literature.html"><a href="literature.html"><i class="fa fa-check"></i><b>5</b> Literature</a></li>
<li class="chapter" data-level="6" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>6</b> Methods</a></li>
<li class="chapter" data-level="7" data-path="applications.html"><a href="applications.html"><i class="fa fa-check"></i><b>7</b> Applications</a>
<ul>
<li class="chapter" data-level="7.1" data-path="applications.html"><a href="applications.html#example-one"><i class="fa fa-check"></i><b>7.1</b> Example one</a></li>
<li class="chapter" data-level="7.2" data-path="applications.html"><a href="applications.html#example-two"><i class="fa fa-check"></i><b>7.2</b> Example two</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>8</b> Final Words</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://pmuench.com" target="blank">Philipp C. Muench</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">01-load-data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="intro" class="section level1" number="4">
<h1><span class="header-section-number">Section 4</span> Introduction</h1>
<p>You can label chapter and section titles using <code>{#label}</code> after them, e.g., we can reference Chapter <a href="intro.html#intro">4</a>. If you do not manually label them, there will be automatic labels anyway, e.g., Chapter <a href="methods.html#methods">6</a>.</p>
<p>Figures and tables with captions will be placed in <code>figure</code> and <code>table</code> environments, respectively.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="intro.html#cb11-1"></a><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">4</span>, <span class="fl">.1</span>, <span class="fl">.1</span>))</span>
<span id="cb11-2"><a href="intro.html#cb11-2"></a><span class="kw">plot</span>(pressure, <span class="dt">type =</span> <span class="st">&#39;b&#39;</span>, <span class="dt">pch =</span> <span class="dv">19</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:nice-fig"></span>
<img src="oligomm-claudia_files/figure-html/nice-fig-1.png" alt="Here is a nice figure!" width="80%" />
<p class="caption">
Figure 4.1: Here is a nice figure!
</p>
</div>
<p>Reference a figure by its code chunk label with the <code>fig:</code> prefix, e.g., see Figure <a href="intro.html#fig:nice-fig">4.1</a>. Similarly, you can reference tables generated from <code>knitr::kable()</code>, e.g., see Table <a href="intro.html#tab:nice-tab">4.1</a>.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="intro.html#cb12-1"></a>knitr<span class="op">::</span><span class="kw">kable</span>(</span>
<span id="cb12-2"><a href="intro.html#cb12-2"></a>  <span class="kw">head</span>(iris, <span class="dv">20</span>), <span class="dt">caption =</span> <span class="st">&#39;Here is a nice table!&#39;</span>,</span>
<span id="cb12-3"><a href="intro.html#cb12-3"></a>  <span class="dt">booktabs =</span> <span class="ot">TRUE</span></span>
<span id="cb12-4"><a href="intro.html#cb12-4"></a>)</span></code></pre></div>
<table>
<caption>
<span id="tab:nice-tab">Table 4.1: </span>Here is a nice table!
</caption>
<thead>
<tr>
<th style="text-align:right;">
Sepal.Length
</th>
<th style="text-align:right;">
Sepal.Width
</th>
<th style="text-align:right;">
Petal.Length
</th>
<th style="text-align:right;">
Petal.Width
</th>
<th style="text-align:left;">
Species
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
5.1
</td>
<td style="text-align:right;">
3.5
</td>
<td style="text-align:right;">
1.4
</td>
<td style="text-align:right;">
0.2
</td>
<td style="text-align:left;">
setosa
</td>
</tr>
<tr>
<td style="text-align:right;">
4.9
</td>
<td style="text-align:right;">
3.0
</td>
<td style="text-align:right;">
1.4
</td>
<td style="text-align:right;">
0.2
</td>
<td style="text-align:left;">
setosa
</td>
</tr>
<tr>
<td style="text-align:right;">
4.7
</td>
<td style="text-align:right;">
3.2
</td>
<td style="text-align:right;">
1.3
</td>
<td style="text-align:right;">
0.2
</td>
<td style="text-align:left;">
setosa
</td>
</tr>
<tr>
<td style="text-align:right;">
4.6
</td>
<td style="text-align:right;">
3.1
</td>
<td style="text-align:right;">
1.5
</td>
<td style="text-align:right;">
0.2
</td>
<td style="text-align:left;">
setosa
</td>
</tr>
<tr>
<td style="text-align:right;">
5.0
</td>
<td style="text-align:right;">
3.6
</td>
<td style="text-align:right;">
1.4
</td>
<td style="text-align:right;">
0.2
</td>
<td style="text-align:left;">
setosa
</td>
</tr>
<tr>
<td style="text-align:right;">
5.4
</td>
<td style="text-align:right;">
3.9
</td>
<td style="text-align:right;">
1.7
</td>
<td style="text-align:right;">
0.4
</td>
<td style="text-align:left;">
setosa
</td>
</tr>
<tr>
<td style="text-align:right;">
4.6
</td>
<td style="text-align:right;">
3.4
</td>
<td style="text-align:right;">
1.4
</td>
<td style="text-align:right;">
0.3
</td>
<td style="text-align:left;">
setosa
</td>
</tr>
<tr>
<td style="text-align:right;">
5.0
</td>
<td style="text-align:right;">
3.4
</td>
<td style="text-align:right;">
1.5
</td>
<td style="text-align:right;">
0.2
</td>
<td style="text-align:left;">
setosa
</td>
</tr>
<tr>
<td style="text-align:right;">
4.4
</td>
<td style="text-align:right;">
2.9
</td>
<td style="text-align:right;">
1.4
</td>
<td style="text-align:right;">
0.2
</td>
<td style="text-align:left;">
setosa
</td>
</tr>
<tr>
<td style="text-align:right;">
4.9
</td>
<td style="text-align:right;">
3.1
</td>
<td style="text-align:right;">
1.5
</td>
<td style="text-align:right;">
0.1
</td>
<td style="text-align:left;">
setosa
</td>
</tr>
<tr>
<td style="text-align:right;">
5.4
</td>
<td style="text-align:right;">
3.7
</td>
<td style="text-align:right;">
1.5
</td>
<td style="text-align:right;">
0.2
</td>
<td style="text-align:left;">
setosa
</td>
</tr>
<tr>
<td style="text-align:right;">
4.8
</td>
<td style="text-align:right;">
3.4
</td>
<td style="text-align:right;">
1.6
</td>
<td style="text-align:right;">
0.2
</td>
<td style="text-align:left;">
setosa
</td>
</tr>
<tr>
<td style="text-align:right;">
4.8
</td>
<td style="text-align:right;">
3.0
</td>
<td style="text-align:right;">
1.4
</td>
<td style="text-align:right;">
0.1
</td>
<td style="text-align:left;">
setosa
</td>
</tr>
<tr>
<td style="text-align:right;">
4.3
</td>
<td style="text-align:right;">
3.0
</td>
<td style="text-align:right;">
1.1
</td>
<td style="text-align:right;">
0.1
</td>
<td style="text-align:left;">
setosa
</td>
</tr>
<tr>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
4.0
</td>
<td style="text-align:right;">
1.2
</td>
<td style="text-align:right;">
0.2
</td>
<td style="text-align:left;">
setosa
</td>
</tr>
<tr>
<td style="text-align:right;">
5.7
</td>
<td style="text-align:right;">
4.4
</td>
<td style="text-align:right;">
1.5
</td>
<td style="text-align:right;">
0.4
</td>
<td style="text-align:left;">
setosa
</td>
</tr>
<tr>
<td style="text-align:right;">
5.4
</td>
<td style="text-align:right;">
3.9
</td>
<td style="text-align:right;">
1.3
</td>
<td style="text-align:right;">
0.4
</td>
<td style="text-align:left;">
setosa
</td>
</tr>
<tr>
<td style="text-align:right;">
5.1
</td>
<td style="text-align:right;">
3.5
</td>
<td style="text-align:right;">
1.4
</td>
<td style="text-align:right;">
0.3
</td>
<td style="text-align:left;">
setosa
</td>
</tr>
<tr>
<td style="text-align:right;">
5.7
</td>
<td style="text-align:right;">
3.8
</td>
<td style="text-align:right;">
1.7
</td>
<td style="text-align:right;">
0.3
</td>
<td style="text-align:left;">
setosa
</td>
</tr>
<tr>
<td style="text-align:right;">
5.1
</td>
<td style="text-align:right;">
3.8
</td>
<td style="text-align:right;">
1.5
</td>
<td style="text-align:right;">
0.3
</td>
<td style="text-align:left;">
setosa
</td>
</tr>
</tbody>
</table>
<p>You can write citations, too. For example, we are using the <strong>bookdown</strong> package <span class="citation">[@R-bookdown]</span> in this sample book, which was built on top of R Markdown and <strong>knitr</strong> <span class="citation">[@xie2015]</span>.</p>

<p>This file produces the <code>rds</code> file that contains the variant information processed with Lofreq to <code>data/rds/variants_unannotated.rds</code></p>
<div id="functions" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> functions</h2>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="intro.html#cb13-1"></a><span class="co"># processes all vcf files located in @file.paths</span></span>
<span id="cb13-2"><a href="intro.html#cb13-2"></a><span class="co"># and returns a data frame with combined variant</span></span>
<span id="cb13-3"><a href="intro.html#cb13-3"></a><span class="co"># informations with annotated metadata</span></span>
<span id="cb13-4"><a href="intro.html#cb13-4"></a>vcfToDataframe &lt;-<span class="st"> </span><span class="cf">function</span>(vcf.files, contig_mapping) {</span>
<span id="cb13-5"><a href="intro.html#cb13-5"></a>    res &lt;-<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb13-6"><a href="intro.html#cb13-6"></a>    <span class="cf">for</span> (file <span class="cf">in</span> vcf.files) {</span>
<span id="cb13-7"><a href="intro.html#cb13-7"></a>        <span class="kw">message</span>(file)</span>
<span id="cb13-8"><a href="intro.html#cb13-8"></a>        vcf.content &lt;-<span class="st"> </span>vcfR<span class="op">::</span><span class="kw">read.vcfR</span>(file, <span class="dt">verbose =</span> <span class="ot">FALSE</span> )</span>
<span id="cb13-9"><a href="intro.html#cb13-9"></a>        vcf.fix &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(vcf.content<span class="op">@</span>fix) <span class="co"># contains chr, position and substitution informations</span></span>
<span id="cb13-10"><a href="intro.html#cb13-10"></a>        vcf.info &lt;-<span class="st"> </span>vcfR<span class="op">::</span><span class="kw">INFO2df</span>(vcf.content) <span class="co"># get INFO field, contains DP, AF informations</span></span>
<span id="cb13-11"><a href="intro.html#cb13-11"></a>    <span class="co">#  vcf.ann &lt;- data.frame(do.call(&#39;rbind&#39;, strsplit(as.character(vcf.info$ANN),&#39;|&#39;,fixed = TRUE))) # split ANN field, contains information if files are processed by snpEff</span></span>
<span id="cb13-12"><a href="intro.html#cb13-12"></a>        <span class="cf">if</span>(<span class="kw">nrow</span>(vcf.fix) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>) {</span>
<span id="cb13-13"><a href="intro.html#cb13-13"></a>            dat &lt;-<span class="st">  </span><span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(vcf.fix[,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">5</span>, <span class="dv">6</span>)], vcf.info[,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>)]))</span>
<span id="cb13-14"><a href="intro.html#cb13-14"></a>            dat<span class="op">$</span>majorAF &lt;-<span class="st"> </span><span class="kw">sapply</span>(dat<span class="op">$</span>AF, minorAfToMajorAf) <span class="co"># transforms e.g. AF of 0.1 to 0.9, 0.9 stays 0.9 and 0.5 stays 0.5</span></span>
<span id="cb13-15"><a href="intro.html#cb13-15"></a>            dat<span class="op">$</span>genome &lt;-<span class="st"> </span>contig_mapping[<span class="kw">match</span>(dat<span class="op">$</span>CHROM, contig_mapping<span class="op">$</span>contig),]<span class="op">$</span>genome <span class="co"># map chr information to genome name e.g. NHMU01000001.1 -&gt; i48</span></span>
<span id="cb13-16"><a href="intro.html#cb13-16"></a>            dat<span class="op">$</span>genome_hr &lt;-<span class="st"> </span><span class="kw">translateGenomeIdToFullName</span>(dat<span class="op">$</span>genome)</span>
<span id="cb13-17"><a href="intro.html#cb13-17"></a>            dat<span class="op">$</span>day &lt;-<span class="st"> </span><span class="kw">as.integer</span>(<span class="kw">substr</span>(<span class="kw">basename</span>(file), <span class="dv">6</span>, <span class="dv">7</span>)) <span class="co"># get day information from file name</span></span>
<span id="cb13-18"><a href="intro.html#cb13-18"></a>            dat<span class="op">$</span>mouse.id &lt;-<span class="st"> </span><span class="kw">substr</span>(<span class="kw">basename</span>(file), <span class="dv">1</span>, <span class="dv">4</span>) <span class="co"># get mouse ID from file name, e.g. &quot;1681&quot;</span></span>
<span id="cb13-19"><a href="intro.html#cb13-19"></a>            dat<span class="op">$</span>phase &lt;-<span class="st"> </span><span class="kw">translateMouseIdToClaudiaGroup</span>(<span class="kw">as.character</span>(<span class="kw">as.matrix</span>(dat<span class="op">$</span>mouse.id)))</span>
<span id="cb13-20"><a href="intro.html#cb13-20"></a>            dat<span class="op">$</span>phase_num &lt;-<span class="st"> </span><span class="kw">binDaysByPhaseGroup</span>(dat<span class="op">$</span>day)</span>
<span id="cb13-21"><a href="intro.html#cb13-21"></a>            dat<span class="op">$</span>dp &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">as.matrix</span>(vcf.info<span class="op">$</span>DP))</span>
<span id="cb13-22"><a href="intro.html#cb13-22"></a>            res[[<span class="kw">basename</span>(file)]] &lt;-<span class="st"> </span>dat <span class="co"># add vcf df to list</span></span>
<span id="cb13-23"><a href="intro.html#cb13-23"></a>        }</span>
<span id="cb13-24"><a href="intro.html#cb13-24"></a>    }</span>
<span id="cb13-25"><a href="intro.html#cb13-25"></a>    df &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">do.call</span>(rbind, res)) <span class="co"># merge list to df</span></span>
<span id="cb13-26"><a href="intro.html#cb13-26"></a>    <span class="kw">return</span>(df)</span>
<span id="cb13-27"><a href="intro.html#cb13-27"></a>}</span></code></pre></div>
</div>
<div id="iterate-over-files" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> iterate over files</h2>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="intro.html#cb14-1"></a>contig_mapping &lt;-<span class="st"> </span><span class="kw">read.csv2</span>(<span class="st">&quot;data/contig_genome_mapping.csv&quot;</span>, <span class="dt">sep =</span><span class="st">&quot;;&quot;</span>, <span class="dt">header=</span>T, <span class="dt">stringsAsFactors =</span> F) <span class="co"># this file contains contig names of the 12 OligoMM genomes</span></span>
<span id="cb14-2"><a href="intro.html#cb14-2"></a>vcf.files &lt;-<span class="st"> </span><span class="kw">Sys.glob</span>(<span class="st">&quot;data/raw/lofreq/*.vcf&quot;</span>)</span>
<span id="cb14-3"><a href="intro.html#cb14-3"></a>dat &lt;-<span class="st"> </span><span class="kw">vcfToDataframe</span>(vcf.files, contig_mapping)</span></code></pre></div>
<pre><code>## data/raw/lofreq/1423_S34.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1425_S35.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1607_S23.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1612_S24.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1660_S26.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1664_S28.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1750_S27.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1753_S29.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1779_S30.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1783_S31.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1789_S32.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1801_S33.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1807_S42.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1814_S43.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1815_S44.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1877_S25.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1880_S37.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1881_S38.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1882_S39.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1883_S40.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1884_S41.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/1885_S36.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/I_cc_S46.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<pre><code>## data/raw/lofreq/I_mix_S45.sorted.vcf</code></pre>
<pre><code>## Warning in vcfToDataframe(vcf.files, contig_mapping): NAs introduced by coercion</code></pre>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="intro.html#cb63-1"></a><span class="kw">saveRDS</span>(dat, <span class="dt">file =</span> <span class="st">&quot;data/rds/variants_unannotated.rds&quot;</span>)</span></code></pre></div>
<div id="annotate-variants-based-on-gff-file-information-of-reference-genomes" class="section level3" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> annotate variants based on <em>gff</em> file information of reference genomes</h3>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="intro.html#cb64-1"></a><span class="kw">head</span>(dat)</span></code></pre></div>
<pre><code>##                            CHROM     POS REF ALT QUAL DP       AF  majorAF
## 1423_S34.sorted.vcf.1 CP021420.1 2358462   A   G  202 84 0.773810 0.773810
## 1423_S34.sorted.vcf.2 CP021421.1   47859   A   G   93 74 0.824324 0.824324
## 1423_S34.sorted.vcf.3 CP021421.1  155759   G   T  326 73 0.972603 0.972603
## 1423_S34.sorted.vcf.4 CP021421.1  788493   G   A  161 82 0.695122 0.695122
## 1423_S34.sorted.vcf.5 CP021421.1 1198557   T   A  232 72 0.833333 0.833333
##                       genome      genome_hr day mouse.id phase phase_num dp
## 1423_S34.sorted.vcf.1   yl44 A. muciniphila  NA     1423    14        NA 84
## 1423_S34.sorted.vcf.2   yl27 M. intestinale  NA     1423    14        NA 74
## 1423_S34.sorted.vcf.3   yl27 M. intestinale  NA     1423    14        NA 73
## 1423_S34.sorted.vcf.4   yl27 M. intestinale  NA     1423    14        NA 82
## 1423_S34.sorted.vcf.5   yl27 M. intestinale  NA     1423    14        NA 72
##  [ reached &#39;max&#39; / getOption(&quot;max.print&quot;) -- omitted 1 rows ]</code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="number-of-variants.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="literature.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["oligomm-claudia.pdf", "oligomm-claudia.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
