<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Section 4 AB Samples | OMM report</title>
  <meta name="description" content="Section 4 AB Samples | OMM report" />
  <meta name="generator" content="bookdown 0.19 and GitBook 2.6.7" />

  <meta property="og:title" content="Section 4 AB Samples | OMM report" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Section 4 AB Samples | OMM report" />
  
  
  

<meta name="author" content="Philipp C. Muench" />


<meta name="date" content="2020-06-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="resequencing-samples.html"/>

<script src="libs/header-attrs-2.1/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.10/datatables.js"></script>
<link href="libs/dt-core-1.10.19/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.19/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.19/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<script src="libs/plotly-binding-4.9.1/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-1.49.4/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-1.49.4/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Setup</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.1</b> Installation</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#generate-report"><i class="fa fa-check"></i><b>1.2</b> Generate report</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#data-structure"><i class="fa fa-check"></i><b>1.3</b> Data structure</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="claudia-samples.html"><a href="claudia-samples.html"><i class="fa fa-check"></i><b>2</b> Claudia Samples</a>
<ul>
<li class="chapter" data-level="2.1" data-path="claudia-samples.html"><a href="claudia-samples.html#experiment"><i class="fa fa-check"></i><b>2.1</b> Experiment</a></li>
<li class="chapter" data-level="2.2" data-path="claudia-samples.html"><a href="claudia-samples.html#metadata"><i class="fa fa-check"></i><b>2.2</b> Metadata</a></li>
<li class="chapter" data-level="2.3" data-path="claudia-samples.html"><a href="claudia-samples.html#oligomm"><i class="fa fa-check"></i><b>2.3</b> OligoMM</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="claudia-samples.html"><a href="claudia-samples.html#crispr"><i class="fa fa-check"></i><b>2.3.1</b> CRISPR</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="resequencing-samples.html"><a href="resequencing-samples.html"><i class="fa fa-check"></i><b>3</b> Resequencing Samples</a>
<ul>
<li class="chapter" data-level="3.1" data-path="resequencing-samples.html"><a href="resequencing-samples.html#metadata-1"><i class="fa fa-check"></i><b>3.1</b> Metadata</a></li>
<li class="chapter" data-level="3.2" data-path="resequencing-samples.html"><a href="resequencing-samples.html#load-in-variants"><i class="fa fa-check"></i><b>3.2</b> Load in variants</a></li>
<li class="chapter" data-level="3.3" data-path="resequencing-samples.html"><a href="resequencing-samples.html#kb1"><i class="fa fa-check"></i><b>3.3</b> KB1</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="resequencing-samples.html"><a href="resequencing-samples.html#snp"><i class="fa fa-check"></i><b>3.3.1</b> SNP</a></li>
<li class="chapter" data-level="3.3.2" data-path="resequencing-samples.html"><a href="resequencing-samples.html#insertion"><i class="fa fa-check"></i><b>3.3.2</b> insertion</a></li>
<li class="chapter" data-level="3.3.3" data-path="resequencing-samples.html"><a href="resequencing-samples.html#deletion"><i class="fa fa-check"></i><b>3.3.3</b> deletion</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="resequencing-samples.html"><a href="resequencing-samples.html#i46"><i class="fa fa-check"></i><b>3.4</b> I46</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="resequencing-samples.html"><a href="resequencing-samples.html#snp-1"><i class="fa fa-check"></i><b>3.4.1</b> SNP</a></li>
<li class="chapter" data-level="3.4.2" data-path="resequencing-samples.html"><a href="resequencing-samples.html#insertion-1"><i class="fa fa-check"></i><b>3.4.2</b> insertion</a></li>
<li class="chapter" data-level="3.4.3" data-path="resequencing-samples.html"><a href="resequencing-samples.html#deletion-1"><i class="fa fa-check"></i><b>3.4.3</b> deletion</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="resequencing-samples.html"><a href="resequencing-samples.html#yl58"><i class="fa fa-check"></i><b>3.5</b> YL58</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="resequencing-samples.html"><a href="resequencing-samples.html#snp-2"><i class="fa fa-check"></i><b>3.5.1</b> SNP</a></li>
<li class="chapter" data-level="3.5.2" data-path="resequencing-samples.html"><a href="resequencing-samples.html#insertion-2"><i class="fa fa-check"></i><b>3.5.2</b> insertion</a></li>
<li class="chapter" data-level="3.5.3" data-path="resequencing-samples.html"><a href="resequencing-samples.html#deletion-2"><i class="fa fa-check"></i><b>3.5.3</b> deletion</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="resequencing-samples.html"><a href="resequencing-samples.html#mutation-bias"><i class="fa fa-check"></i><b>3.6</b> mutation bias?</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="resequencing-samples.html"><a href="resequencing-samples.html#snp-3"><i class="fa fa-check"></i><b>3.6.1</b> SNP</a></li>
<li class="chapter" data-level="3.6.2" data-path="resequencing-samples.html"><a href="resequencing-samples.html#insertions"><i class="fa fa-check"></i><b>3.6.2</b> insertions</a></li>
<li class="chapter" data-level="3.6.3" data-path="resequencing-samples.html"><a href="resequencing-samples.html#deletions"><i class="fa fa-check"></i><b>3.6.3</b> deletions</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="resequencing-samples.html"><a href="resequencing-samples.html#heatmap"><i class="fa fa-check"></i><b>3.7</b> heatmap</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="resequencing-samples.html"><a href="resequencing-samples.html#kb1-1"><i class="fa fa-check"></i><b>3.7.1</b> KB1</a></li>
<li class="chapter" data-level="3.7.2" data-path="resequencing-samples.html"><a href="resequencing-samples.html#i46-1"><i class="fa fa-check"></i><b>3.7.2</b> I46</a></li>
<li class="chapter" data-level="3.7.3" data-path="resequencing-samples.html"><a href="resequencing-samples.html#yl58-1"><i class="fa fa-check"></i><b>3.7.3</b> YL58</a></li>
<li class="chapter" data-level="3.7.4" data-path="resequencing-samples.html"><a href="resequencing-samples.html#filter-present-in-all"><i class="fa fa-check"></i><b>3.7.4</b> Filter: present in all</a></li>
<li class="chapter" data-level="3.7.5" data-path="resequencing-samples.html"><a href="resequencing-samples.html#kb1-2"><i class="fa fa-check"></i><b>3.7.5</b> KB1</a></li>
<li class="chapter" data-level="3.7.6" data-path="resequencing-samples.html"><a href="resequencing-samples.html#yl58-2"><i class="fa fa-check"></i><b>3.7.6</b> YL58</a></li>
<li class="chapter" data-level="3.7.7" data-path="resequencing-samples.html"><a href="resequencing-samples.html#i46-2"><i class="fa fa-check"></i><b>3.7.7</b> I46</a></li>
<li class="chapter" data-level="3.7.8" data-path="resequencing-samples.html"><a href="resequencing-samples.html#filter-not-in-water-but-in-ab"><i class="fa fa-check"></i><b>3.7.8</b> Filter: not in water but in AB</a></li>
<li class="chapter" data-level="3.7.9" data-path="resequencing-samples.html"><a href="resequencing-samples.html#kb1-3"><i class="fa fa-check"></i><b>3.7.9</b> KB1</a></li>
<li class="chapter" data-level="3.7.10" data-path="resequencing-samples.html"><a href="resequencing-samples.html#yl58-3"><i class="fa fa-check"></i><b>3.7.10</b> YL58</a></li>
<li class="chapter" data-level="3.7.11" data-path="resequencing-samples.html"><a href="resequencing-samples.html#i46-3"><i class="fa fa-check"></i><b>3.7.11</b> I46</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="resequencing-samples.html"><a href="resequencing-samples.html#visualize-coverage"><i class="fa fa-check"></i><b>3.8</b> visualize coverage</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="ab-samples.html"><a href="ab-samples.html"><i class="fa fa-check"></i><b>4</b> AB Samples</a>
<ul>
<li class="chapter" data-level="4.1" data-path="ab-samples.html"><a href="ab-samples.html#load-in-variants-1"><i class="fa fa-check"></i><b>4.1</b> Load in variants</a></li>
<li class="chapter" data-level="4.2" data-path="ab-samples.html"><a href="ab-samples.html#af-frequency"><i class="fa fa-check"></i><b>4.2</b> AF frequency</a></li>
<li class="chapter" data-level="4.3" data-path="ab-samples.html"><a href="ab-samples.html#number-of-variants-per-samples"><i class="fa fa-check"></i><b>4.3</b> number of variants per samples</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="ab-samples.html"><a href="ab-samples.html#number-of-variants-per-treatment-group"><i class="fa fa-check"></i><b>4.3.1</b> number of variants per treatment group</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="ab-samples.html"><a href="ab-samples.html#heatmap-1"><i class="fa fa-check"></i><b>4.4</b> Heatmap</a></li>
<li class="chapter" data-level="4.5" data-path="ab-samples.html"><a href="ab-samples.html#focus-on-mouse-where-we-have-many-time-points"><i class="fa fa-check"></i><b>4.5</b> Focus on mouse where we have many time points</a></li>
<li class="chapter" data-level="4.6" data-path="ab-samples.html"><a href="ab-samples.html#akkermansia-muciniphila"><i class="fa fa-check"></i><b>4.6</b> Akkermansia Muciniphila</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="ab-samples.html"><a href="ab-samples.html#area-plot-1"><i class="fa fa-check"></i><b>4.6.1</b> area plot 1</a></li>
<li class="chapter" data-level="4.6.2" data-path="ab-samples.html"><a href="ab-samples.html#line-plot"><i class="fa fa-check"></i><b>4.6.2</b> line plot</a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">OMM report</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="ab-samples" class="section level1" number="4">
<h1><span class="header-section-number">Section 4</span> AB Samples</h1>
<div id="load-in-variants-1" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Load in variants</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="ab-samples.html#cb1-1"></a>vcfToDataframe &lt;-<span class="st"> </span><span class="cf">function</span>(vcf.files, <span class="dt">contig_mapping =</span> contig_mapping, <span class="dt">gff.df =</span> gff.df) {</span>
<span id="cb1-2"><a href="ab-samples.html#cb1-2"></a>    <span class="kw">require</span>(vcfR)</span>
<span id="cb1-3"><a href="ab-samples.html#cb1-3"></a>    res &lt;-<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb1-4"><a href="ab-samples.html#cb1-4"></a>    <span class="cf">for</span> (file <span class="cf">in</span> vcf.files) {</span>
<span id="cb1-5"><a href="ab-samples.html#cb1-5"></a>        <span class="kw">library</span>(data.table)</span>
<span id="cb1-6"><a href="ab-samples.html#cb1-6"></a>        vcf.content &lt;-<span class="st"> </span>vcfR<span class="op">::</span><span class="kw">read.vcfR</span>(file, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</span>
<span id="cb1-7"><a href="ab-samples.html#cb1-7"></a>        vcf.fix &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(vcf.content<span class="op">@</span>fix) <span class="co"># contains chr, position and substitution informations</span></span>
<span id="cb1-8"><a href="ab-samples.html#cb1-8"></a>        vcf.info &lt;-<span class="st"> </span>vcfR<span class="op">::</span><span class="kw">INFO2df</span>(vcf.content) <span class="co"># get INFO field, contains DP, AF informations</span></span>
<span id="cb1-9"><a href="ab-samples.html#cb1-9"></a>        <span class="cf">if</span>(<span class="kw">nrow</span>(vcf.fix) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>) { <span class="co"># there are variants</span></span>
<span id="cb1-10"><a href="ab-samples.html#cb1-10"></a>            dat &lt;-<span class="st">  </span><span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(vcf.fix[,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">5</span>, <span class="dv">6</span>)], vcf.info[,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>)]))</span>
<span id="cb1-11"><a href="ab-samples.html#cb1-11"></a>            dat<span class="op">$</span>majorAF &lt;-<span class="st"> </span><span class="kw">sapply</span>(dat<span class="op">$</span>AF, minorAfToMajorAf) <span class="co"># transforms e.g. AF of 0.1 to 0.9, 0.9 stays 0.9 and 0.5 stays 0.5</span></span>
<span id="cb1-12"><a href="ab-samples.html#cb1-12"></a>            dat<span class="op">$</span>genome &lt;-<span class="st"> </span>contig_mapping[<span class="kw">match</span>(dat<span class="op">$</span>CHROM, contig_mapping<span class="op">$</span>contig),]<span class="op">$</span>genome <span class="co"># map chr information to genome name e.g. NHMU01000001.1 -&gt; i48</span></span>
<span id="cb1-13"><a href="ab-samples.html#cb1-13"></a>            dat<span class="op">$</span>genome_hr &lt;-<span class="st"> </span><span class="kw">translateGenomeIdToFullName</span>(<span class="kw">tolower</span>(dat<span class="op">$</span>genome))</span>
<span id="cb1-14"><a href="ab-samples.html#cb1-14"></a>            dat<span class="op">$</span>mouse.id &lt;-<span class="st">  </span><span class="kw">substr</span>(tools<span class="op">::</span><span class="kw">file_path_sans_ext</span>(<span class="kw">basename</span>(file)), <span class="dv">1</span>, <span class="dv">4</span>)</span>
<span id="cb1-15"><a href="ab-samples.html#cb1-15"></a>            dat<span class="op">$</span>mouse.group &lt;-<span class="st"> </span><span class="kw">translateMouseIdToTreatmentGroup</span>(dat<span class="op">$</span>mouse.id)</span>
<span id="cb1-16"><a href="ab-samples.html#cb1-16"></a>            dat<span class="op">$</span>day &lt;-<span class="st"> </span><span class="kw">as.integer</span>(<span class="kw">substr</span>(<span class="kw">basename</span>(file), <span class="dv">6</span>, <span class="dv">7</span>))</span>
<span id="cb1-17"><a href="ab-samples.html#cb1-17"></a>            dat<span class="op">$</span>phase &lt;-<span class="st"> </span><span class="kw">binDaysByPhase</span>(<span class="kw">as.numeric</span>(<span class="kw">as.matrix</span>(dat<span class="op">$</span>day)))</span>
<span id="cb1-18"><a href="ab-samples.html#cb1-18"></a>            dat<span class="op">$</span>phase_num &lt;-<span class="st"> </span><span class="kw">binDaysByPhaseGroup</span>(dat<span class="op">$</span>day)</span>
<span id="cb1-19"><a href="ab-samples.html#cb1-19"></a>            dat<span class="op">$</span>dp &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">as.matrix</span>(vcf.info<span class="op">$</span>DP))</span>
<span id="cb1-20"><a href="ab-samples.html#cb1-20"></a>            <span class="co"># annotate overlay of gene</span></span>
<span id="cb1-21"><a href="ab-samples.html#cb1-21"></a>            dt.gff &lt;-<span class="st"> </span><span class="kw">data.table</span>(<span class="dt">start =</span> gff.df<span class="op">$</span>start, <span class="dt">end =</span> gff.df<span class="op">$</span>end,</span>
<span id="cb1-22"><a href="ab-samples.html#cb1-22"></a>                <span class="dt">chr =</span> <span class="kw">as.character</span>(<span class="kw">as.matrix</span>(gff.df<span class="op">$</span>chr)), <span class="dt">feature =</span> gff.df<span class="op">$</span>product)</span>
<span id="cb1-23"><a href="ab-samples.html#cb1-23"></a>            <span class="kw">colnames</span>(dat)[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;chr&quot;</span>, <span class="st">&quot;start&quot;</span>)</span>
<span id="cb1-24"><a href="ab-samples.html#cb1-24"></a>            dat<span class="op">$</span>start &lt;-<span class="st"> </span><span class="kw">as.integer</span>(<span class="kw">as.matrix</span>(dat<span class="op">$</span>start))</span>
<span id="cb1-25"><a href="ab-samples.html#cb1-25"></a>            dat<span class="op">$</span>chr &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="kw">as.matrix</span>(dat<span class="op">$</span>chr))</span>
<span id="cb1-26"><a href="ab-samples.html#cb1-26"></a>            dat<span class="op">$</span>end &lt;-<span class="st"> </span>dat<span class="op">$</span>start</span>
<span id="cb1-27"><a href="ab-samples.html#cb1-27"></a>            dat2 &lt;-<span class="st"> </span><span class="kw">as.data.table</span>(dat)</span>
<span id="cb1-28"><a href="ab-samples.html#cb1-28"></a>            <span class="kw">setkey</span>(dt.gff, chr, start, end)</span>
<span id="cb1-29"><a href="ab-samples.html#cb1-29"></a>            annotated &lt;-<span class="st"> </span><span class="kw">foverlaps</span>(dat2, dt.gff, <span class="dt">type=</span><span class="st">&quot;within&quot;</span>, <span class="dt">mult=</span><span class="st">&quot;first&quot;</span>)</span>
<span id="cb1-30"><a href="ab-samples.html#cb1-30"></a>            res[[tools<span class="op">::</span><span class="kw">file_path_sans_ext</span>(<span class="kw">basename</span>(file))]] &lt;-<span class="st"> </span>annotated <span class="co"># add vcf df to list</span></span>
<span id="cb1-31"><a href="ab-samples.html#cb1-31"></a>        } <span class="cf">else</span>{</span>
<span id="cb1-32"><a href="ab-samples.html#cb1-32"></a>            <span class="kw">message</span>(<span class="st">&quot;Skipping&quot;</span>)</span>
<span id="cb1-33"><a href="ab-samples.html#cb1-33"></a>        }</span>
<span id="cb1-34"><a href="ab-samples.html#cb1-34"></a>    }</span>
<span id="cb1-35"><a href="ab-samples.html#cb1-35"></a>    df &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">do.call</span>(rbind, res)) <span class="co"># merge list to df</span></span>
<span id="cb1-36"><a href="ab-samples.html#cb1-36"></a>    <span class="kw">return</span>(df)</span>
<span id="cb1-37"><a href="ab-samples.html#cb1-37"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="ab-samples.html#cb2-1"></a><span class="co"># load in reference information</span></span>
<span id="cb2-2"><a href="ab-samples.html#cb2-2"></a>gff.files &lt;-<span class="st"> </span><span class="kw">Sys.glob</span>(<span class="st">&quot;data/references/joined_reference_curated_ecoli/*.gff&quot;</span>)</span>
<span id="cb2-3"><a href="ab-samples.html#cb2-3"></a>gff.df &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb2-4"><a href="ab-samples.html#cb2-4"></a><span class="cf">for</span> (gff.file <span class="cf">in</span> gff.files){</span>
<span id="cb2-5"><a href="ab-samples.html#cb2-5"></a>    <span class="kw">message</span>(gff.file)</span>
<span id="cb2-6"><a href="ab-samples.html#cb2-6"></a>    gff &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">readGFF</span>(gff.file)</span>
<span id="cb2-7"><a href="ab-samples.html#cb2-7"></a>    <span class="co"># subset since different columns are present on gff files</span></span>
<span id="cb2-8"><a href="ab-samples.html#cb2-8"></a>    relevant &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">start =</span> gff<span class="op">$</span>start, <span class="dt">end =</span> gff<span class="op">$</span>end, <span class="dt">type =</span> <span class="kw">as.character</span>(<span class="kw">as.matrix</span>(gff<span class="op">$</span>type)), <span class="dt">gene =</span> <span class="kw">as.character</span>(<span class="kw">as.matrix</span>(gff<span class="op">$</span>gene)), <span class="dt">product =</span> <span class="kw">as.character</span>(<span class="kw">as.matrix</span>(gff<span class="op">$</span>product)), <span class="dt">chr =</span> <span class="kw">as.character</span>(<span class="kw">as.matrix</span>(gff<span class="op">$</span>seqid)))</span>
<span id="cb2-9"><a href="ab-samples.html#cb2-9"></a>    relevant<span class="op">$</span>genome &lt;-<span class="st">  </span><span class="kw">substr</span>(<span class="kw">basename</span>(gff.file), <span class="dv">1</span>, <span class="kw">nchar</span>(<span class="kw">basename</span>(gff.file))<span class="op">-</span><span class="dv">4</span>)</span>
<span id="cb2-10"><a href="ab-samples.html#cb2-10"></a>    gff.df &lt;-<span class="st"> </span><span class="kw">rbind</span>(gff.df, relevant)</span>
<span id="cb2-11"><a href="ab-samples.html#cb2-11"></a>}</span></code></pre></div>
<pre><code>## data/references/joined_reference_curated_ecoli/joined_reference_curated_ecoli.gff</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="ab-samples.html#cb4-1"></a><span class="co"># load in contig information</span></span>
<span id="cb4-2"><a href="ab-samples.html#cb4-2"></a>contig_mapping &lt;-<span class="st"> </span><span class="kw">read.csv2</span>(<span class="st">&quot;data/contig_mapping_new_ref.csv&quot;</span>, <span class="dt">sep =</span><span class="st">&quot;;&quot;</span>, <span class="dt">header=</span>T, <span class="dt">stringsAsFactors =</span> F) <span class="co"># this file contains contig names of the 12 OligoMM genomes</span></span>
<span id="cb4-3"><a href="ab-samples.html#cb4-3"></a></span>
<span id="cb4-4"><a href="ab-samples.html#cb4-4"></a><span class="co"># load in vcf files</span></span>
<span id="cb4-5"><a href="ab-samples.html#cb4-5"></a>vcf.files &lt;-<span class="st"> </span><span class="kw">Sys.glob</span>(<span class="st">&quot;out_philipp/all_vcf/*.vcf&quot;</span>)</span>
<span id="cb4-6"><a href="ab-samples.html#cb4-6"></a>vcf.samples &lt;-<span class="st"> </span><span class="kw">suppressWarnings</span>(<span class="kw">vcfToDataframe</span>(vcf.files, contig_mapping, <span class="dt">gff.df =</span> gff.df))</span></code></pre></div>
<pre><code>## Skipping</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="ab-samples.html#cb6-1"></a>vcf.samples<span class="op">$</span>feature &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="kw">as.matrix</span>(vcf.samples<span class="op">$</span>feature)) </span>
<span id="cb6-2"><a href="ab-samples.html#cb6-2"></a></span>
<span id="cb6-3"><a href="ab-samples.html#cb6-3"></a>vcf.samples[<span class="kw">which</span>(<span class="kw">is.na</span>(vcf.samples<span class="op">$</span>feature)),]<span class="op">$</span>feature &lt;-<span class="st"> &quot;outside ORFs&quot;</span></span>
<span id="cb6-4"><a href="ab-samples.html#cb6-4"></a></span>
<span id="cb6-5"><a href="ab-samples.html#cb6-5"></a>vcf.samples <span class="op">$</span>start &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb6-6"><a href="ab-samples.html#cb6-6"></a>vcf.samples <span class="op">$</span>end &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb6-7"><a href="ab-samples.html#cb6-7"></a>vcf.samples <span class="op">$</span>i.end &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb6-8"><a href="ab-samples.html#cb6-8"></a><span class="kw">colnames</span>(vcf.samples)[<span class="dv">3</span>] &lt;-<span class="st"> &quot;POS&quot;</span></span>
<span id="cb6-9"><a href="ab-samples.html#cb6-9"></a></span>
<span id="cb6-10"><a href="ab-samples.html#cb6-10"></a><span class="kw">saveRDS</span>(vcf.samples, <span class="dt">file =</span> <span class="st">&quot;data/rds/omm_ab.rds&quot;</span>)</span></code></pre></div>
</div>
<div id="af-frequency" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> AF frequency</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="ab-samples.html#cb7-1"></a>p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(vcf.samples, <span class="kw">aes</span>(AF, <span class="dt">fill =</span> genome)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_histogram</span>()</span>
<span id="cb7-2"><a href="ab-samples.html#cb7-2"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">facet_grid</span>(mouse.id <span class="op">+</span><span class="st"> </span>mouse.group <span class="op">~</span><span class="st"> </span>genome <span class="op">+</span><span class="st"> </span>genome_hr)</span>
<span id="cb7-3"><a href="ab-samples.html#cb7-3"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;AF&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;occurence&quot;</span>)</span>
<span id="cb7-4"><a href="ab-samples.html#cb7-4"></a><span class="kw">print</span>(p)</span></code></pre></div>
<pre><code>## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-3"></span>
<img src="oligomm-claudia_files/figure-html/unnamed-chunk-3-1.png" alt="AF of resequenced strains" width="1440" />
<p class="caption">
Figure 4.1: AF of resequenced strains
</p>
</div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="ab-samples.html#cb9-1"></a>p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(vcf.samples, <span class="kw">aes</span>(majorAF, <span class="dt">fill =</span> genome)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_histogram</span>()</span>
<span id="cb9-2"><a href="ab-samples.html#cb9-2"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">facet_grid</span>(mouse.id <span class="op">+</span><span class="st"> </span>mouse.group <span class="op">~</span><span class="st"> </span>genome <span class="op">+</span><span class="st"> </span>genome_hr)</span>
<span id="cb9-3"><a href="ab-samples.html#cb9-3"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;AF&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;occurence&quot;</span>)</span>
<span id="cb9-4"><a href="ab-samples.html#cb9-4"></a><span class="kw">print</span>(p)</span></code></pre></div>
<pre><code>## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-4"></span>
<img src="oligomm-claudia_files/figure-html/unnamed-chunk-4-1.png" alt="major AF of resequenced strains" width="1440" />
<p class="caption">
Figure 4.2: major AF of resequenced strains
</p>
</div>
</div>
<div id="number-of-variants-per-samples" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> number of variants per samples</h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="ab-samples.html#cb11-1"></a>dat &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;data/rds/omm_ab.rds&quot;</span>)</span>
<span id="cb11-2"><a href="ab-samples.html#cb11-2"></a>dat<span class="op">$</span>dummy &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb11-3"><a href="ab-samples.html#cb11-3"></a>dat.agg &lt;-<span class="st"> </span><span class="kw">aggregate</span>(dummy <span class="op">~</span><span class="st"> </span>mouse.id <span class="op">+</span><span class="st"> </span>mouse.group <span class="op">+</span><span class="st"> </span>day <span class="op">+</span><span class="st"> </span>phase, dat, sum)</span>
<span id="cb11-4"><a href="ab-samples.html#cb11-4"></a>DT<span class="op">::</span><span class="kw">datatable</span>(dat.agg)</span></code></pre></div>
<div id="htmlwidget-b38e203ab589678e8b44" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-b38e203ab589678e8b44">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110"],["1688","1692","1699","1688","1692","1699","1683","1686","1688","1690","1692","1693","1694","1697","1698","1699","1681","1683","1684","1688","1692","1699","1683","1688","1692","1699","1683","1688","1692","1699","1683","1688","1692","1699","1683","1686","1688","1690","1692","1693","1694","1697","1698","1699","1681","1683","1684","1688","1692","1699","1686","1688","1690","1692","1693","1694","1697","1698","1699","1681","1683","1684","1688","1692","1699","1683","1686","1688","1690","1692","1693","1694","1697","1698","1699","1681","1683","1684","1688","1692","1699","1683","1686","1688","1690","1692","1693","1694","1697","1698","1699","1681","1683","1684","1688","1692","1699","1683","1686","1688","1690","1692","1693","1694","1697","1698","1699","1681","1683","1684"],["Ciprofloxacin","Tetracyclin","Vancomycin","Ciprofloxacin","Tetracyclin","Vancomycin","Water","Ciprofloxacin","Ciprofloxacin","Ciprofloxacin","Tetracyclin","Tetracyclin","Tetracyclin","Vancomycin","Vancomycin","Vancomycin","Water","Water","Water","Ciprofloxacin","Tetracyclin","Vancomycin","Water","Ciprofloxacin","Tetracyclin","Vancomycin","Water","Ciprofloxacin","Tetracyclin","Vancomycin","Water","Ciprofloxacin","Tetracyclin","Vancomycin","Water","Ciprofloxacin","Ciprofloxacin","Ciprofloxacin","Tetracyclin","Tetracyclin","Tetracyclin","Vancomycin","Vancomycin","Vancomycin","Water","Water","Water","Ciprofloxacin","Tetracyclin","Vancomycin","Ciprofloxacin","Ciprofloxacin","Ciprofloxacin","Tetracyclin","Tetracyclin","Tetracyclin","Vancomycin","Vancomycin","Vancomycin","Water","Water","Water","Ciprofloxacin","Tetracyclin","Vancomycin","Water","Ciprofloxacin","Ciprofloxacin","Ciprofloxacin","Tetracyclin","Tetracyclin","Tetracyclin","Vancomycin","Vancomycin","Vancomycin","Water","Water","Water","Ciprofloxacin","Tetracyclin","Vancomycin","Water","Ciprofloxacin","Ciprofloxacin","Ciprofloxacin","Tetracyclin","Tetracyclin","Tetracyclin","Vancomycin","Vancomycin","Vancomycin","Water","Water","Water","Ciprofloxacin","Tetracyclin","Vancomycin","Water","Ciprofloxacin","Ciprofloxacin","Ciprofloxacin","Tetracyclin","Tetracyclin","Tetracyclin","Vancomycin","Vancomycin","Vancomycin","Water","Water","Water"],[37,37,37,44,44,44,44,49,49,49,49,49,49,49,49,49,49,49,49,4,4,4,4,18,18,18,18,53,53,53,53,67,67,67,67,0,0,0,0,0,0,0,0,0,0,0,0,9,9,9,14,14,14,14,14,14,14,14,14,14,14,14,23,23,23,23,30,30,30,30,30,30,30,30,30,30,30,30,58,58,58,58,63,63,63,63,63,63,63,63,63,63,63,63,72,72,72,72,79,79,79,79,79,79,79,79,79,79,79,79],["long-recovery","long-recovery","long-recovery","long-recovery","long-recovery","long-recovery","long-recovery","long-recovery","long-recovery","long-recovery","long-recovery","long-recovery","long-recovery","long-recovery","long-recovery","long-recovery","long-recovery","long-recovery","long-recovery","post-treatment","post-treatment","post-treatment","post-treatment","post-treatment","post-treatment","post-treatment","post-treatment","post-treatment","post-treatment","post-treatment","post-treatment","post-treatment","post-treatment","post-treatment","post-treatment","pre-treatment","pre-treatment","pre-treatment","pre-treatment","pre-treatment","pre-treatment","pre-treatment","pre-treatment","pre-treatment","pre-treatment","pre-treatment","pre-treatment","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery","short-recovery"],[1331,1448,1314,1683,1405,1154,1612,1740,1480,1547,1438,1648,1667,1313,1051,1104,1412,1584,1345,1499,661,999,1448,1313,1562,851,1471,1525,1288,880,1664,1637,1021,808,1555,1576,1292,1512,1548,1561,1236,1245,1560,1690,1179,1669,1061,1449,1339,1345,1747,1323,1851,1698,1767,1502,1248,1016,1239,1245,1423,1081,1497,1652,1212,1424,1775,1369,1664,1457,1656,1808,1191,1168,1410,1396,1549,1255,1674,1542,1182,1500,1544,1579,1648,1556,1659,1598,1157,1302,1342,1204,1624,1286,1475,2410,1349,1577,1425,1382,1557,1584,1840,1300,1026,1384,1292,1113,1504,1051]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>mouse.id<\/th>\n      <th>mouse.group<\/th>\n      <th>day<\/th>\n      <th>phase<\/th>\n      <th>dummy<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="number-of-variants-per-treatment-group" class="section level3" number="4.3.1">
<h3><span class="header-section-number">4.3.1</span> number of variants per treatment group</h3>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="ab-samples.html#cb12-1"></a>p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(dat.agg, <span class="kw">aes</span>(<span class="dt">x =</span> mouse.id, <span class="dt">y =</span> dummy, <span class="dt">color =</span> day))</span>
<span id="cb12-2"><a href="ab-samples.html#cb12-2"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_jitter</span>(<span class="dt">shape =</span> <span class="dv">4</span>) <span class="op">+</span><span class="st"> </span><span class="kw">facet_grid</span>(. <span class="op">~</span><span class="st"> </span>mouse.group, <span class="dt">scales =</span> <span class="st">&quot;free_x&quot;</span>)</span>
<span id="cb12-3"><a href="ab-samples.html#cb12-3"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_boxplot</span>()  <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Mouse ID&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;number of variants&quot;</span>)</span>
<span id="cb12-4"><a href="ab-samples.html#cb12-4"></a>plotly<span class="op">::</span><span class="kw">ggplotly</span>(p)</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-6"></span>
<div id="htmlwidget-d91bfcaa49714bd3e6da" style="width:960px;height:480px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-d91bfcaa49714bd3e6da">{"x":{"data":[{"x":[2.14420470464975,2.28560303077102,1.35455107465386,2.14761564582586,2.71315700225532,2.029408878088,2.32909690663219,1.94093199633062,1.75939907804132,1.39370612930506,1.68315814789385,3.34120140131563,2.38682488203049,0.966100102104247,1.92218360342085,3.39204970505089,2.05385542307049,1.2524000801146,1.84189064521343,3.13324021808803,1.89325969293714,1.21106515675783,2.12961795646697,3.36428221669048,1.70368656888604,0.986045814864337,1.71312683597207,3.29021167140454],"y":[1330.62401380427,1683.22509880234,1739.87065263093,1480.38888953328,1546.90211129785,1498.92210396919,1312.89909046516,1525.19230176844,1636.72689191327,1576.11303645205,1291.86844814345,1511.90241512693,1449.2548344953,1746.92712405343,1323.32811147105,1851.27901006937,1496.88367914613,1775.13810267746,1369.13472398147,1663.83505787216,1673.71496141721,1544.00832501017,1579.08352425341,1648.06618524827,1475.26832887884,1424.79208079726,1382.10364304353,1556.65161862224],"text":["mouse.id: 1688<br />dummy: 1331<br />day: 37","mouse.id: 1688<br />dummy: 1683<br />day: 44","mouse.id: 1686<br />dummy: 1740<br />day: 49","mouse.id: 1688<br />dummy: 1480<br />day: 49","mouse.id: 1690<br />dummy: 1547<br />day: 49","mouse.id: 1688<br />dummy: 1499<br />day:  4","mouse.id: 1688<br />dummy: 1313<br />day: 18","mouse.id: 1688<br />dummy: 1525<br />day: 53","mouse.id: 1688<br />dummy: 1637<br />day: 67","mouse.id: 1686<br />dummy: 1576<br />day:  0","mouse.id: 1688<br />dummy: 1292<br />day:  0","mouse.id: 1690<br />dummy: 1512<br />day:  0","mouse.id: 1688<br />dummy: 1449<br />day:  9","mouse.id: 1686<br />dummy: 1747<br />day: 14","mouse.id: 1688<br />dummy: 1323<br />day: 14","mouse.id: 1690<br />dummy: 1851<br />day: 14","mouse.id: 1688<br />dummy: 1497<br />day: 23","mouse.id: 1686<br />dummy: 1775<br />day: 30","mouse.id: 1688<br />dummy: 1369<br />day: 30","mouse.id: 1690<br />dummy: 1664<br />day: 30","mouse.id: 1688<br />dummy: 1674<br />day: 58","mouse.id: 1686<br />dummy: 1544<br />day: 63","mouse.id: 1688<br />dummy: 1579<br />day: 63","mouse.id: 1690<br />dummy: 1648<br />day: 63","mouse.id: 1688<br />dummy: 1475<br />day: 72","mouse.id: 1686<br />dummy: 1425<br />day: 79","mouse.id: 1688<br />dummy: 1382<br />day: 79","mouse.id: 1690<br />dummy: 1557<br />day: 79"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":["rgba(49,102,146,1)","rgba(55,114,162,1)","rgba(59,123,174,1)","rgba(59,123,174,1)","rgba(59,123,174,1)","rgba(22,49,75,1)","rgba(33,71,104,1)","rgba(62,130,183,1)","rgba(75,155,217,1)","rgba(19,43,67,1)","rgba(19,43,67,1)","rgba(19,43,67,1)","rgba(26,57,85,1)","rgba(30,64,96,1)","rgba(30,64,96,1)","rgba(30,64,96,1)","rgba(37,79,115,1)","rgba(43,90,130,1)","rgba(43,90,130,1)","rgba(43,90,130,1)","rgba(67,139,195,1)","rgba(71,148,207,1)","rgba(71,148,207,1)","rgba(71,148,207,1)","rgba(80,164,230,1)","rgba(86,177,247,1)","rgba(86,177,247,1)","rgba(86,177,247,1)"],"opacity":1,"size":5.66929133858268,"symbol":"x-thin-open","line":{"width":1.88976377952756,"color":["rgba(49,102,146,1)","rgba(55,114,162,1)","rgba(59,123,174,1)","rgba(59,123,174,1)","rgba(59,123,174,1)","rgba(22,49,75,1)","rgba(33,71,104,1)","rgba(62,130,183,1)","rgba(75,155,217,1)","rgba(19,43,67,1)","rgba(19,43,67,1)","rgba(19,43,67,1)","rgba(26,57,85,1)","rgba(30,64,96,1)","rgba(30,64,96,1)","rgba(30,64,96,1)","rgba(37,79,115,1)","rgba(43,90,130,1)","rgba(43,90,130,1)","rgba(43,90,130,1)","rgba(67,139,195,1)","rgba(71,148,207,1)","rgba(71,148,207,1)","rgba(71,148,207,1)","rgba(80,164,230,1)","rgba(86,177,247,1)","rgba(86,177,247,1)","rgba(86,177,247,1)"]}},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[1.19828758202493,1.36409250628203,1.03369914423674,1.70402902346104,3.0054625146091,0.668998756445944,0.671074180491269,0.993541984632611,0.695912179164588,1.07570874206722,1.71672355663031,3.02342988867313,1.09290930256248,0.737083206139505,1.93324844297022,2.62847222238779,1.38129326831549,0.726938569173217,1.76577528510243,3.2067618643865,1.17409958951175,0.95397399906069,1.76431052256376,3.24790931157768,0.780959309637547,1.13553717061877,1.7682259997353,2.64081861991435],"y":[1447.93757177368,1404.69354360215,1438.3857851563,1647.67087182011,1666.85040749274,661.343668434583,1561.67055865303,1287.85450244993,1020.88981749807,1547.92873281892,1561.11217683498,1235.67836128064,1339.18551096339,1697.97472938206,1767.18764609359,1501.97801972721,1652.21375785153,1456.70529490598,1655.60031796526,1808.14183153827,1542.25049069617,1555.62262726165,1658.80324183423,1598.06701196972,2410.30038407575,1584.23580133319,1839.95722494721,1299.73993641715],"text":["mouse.id: 1692<br />dummy: 1448<br />day: 37","mouse.id: 1692<br />dummy: 1405<br />day: 44","mouse.id: 1692<br />dummy: 1438<br />day: 49","mouse.id: 1693<br />dummy: 1648<br />day: 49","mouse.id: 1694<br />dummy: 1667<br />day: 49","mouse.id: 1692<br />dummy:  661<br />day:  4","mouse.id: 1692<br />dummy: 1562<br />day: 18","mouse.id: 1692<br />dummy: 1288<br />day: 53","mouse.id: 1692<br />dummy: 1021<br />day: 67","mouse.id: 1692<br />dummy: 1548<br />day:  0","mouse.id: 1693<br />dummy: 1561<br />day:  0","mouse.id: 1694<br />dummy: 1236<br />day:  0","mouse.id: 1692<br />dummy: 1339<br />day:  9","mouse.id: 1692<br />dummy: 1698<br />day: 14","mouse.id: 1693<br />dummy: 1767<br />day: 14","mouse.id: 1694<br />dummy: 1502<br />day: 14","mouse.id: 1692<br />dummy: 1652<br />day: 23","mouse.id: 1692<br />dummy: 1457<br />day: 30","mouse.id: 1693<br />dummy: 1656<br />day: 30","mouse.id: 1694<br />dummy: 1808<br />day: 30","mouse.id: 1692<br />dummy: 1542<br />day: 58","mouse.id: 1692<br />dummy: 1556<br />day: 63","mouse.id: 1693<br />dummy: 1659<br />day: 63","mouse.id: 1694<br />dummy: 1598<br />day: 63","mouse.id: 1692<br />dummy: 2410<br />day: 72","mouse.id: 1692<br />dummy: 1584<br />day: 79","mouse.id: 1693<br />dummy: 1840<br />day: 79","mouse.id: 1694<br />dummy: 1300<br />day: 79"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":["rgba(49,102,146,1)","rgba(55,114,162,1)","rgba(59,123,174,1)","rgba(59,123,174,1)","rgba(59,123,174,1)","rgba(22,49,75,1)","rgba(33,71,104,1)","rgba(62,130,183,1)","rgba(75,155,217,1)","rgba(19,43,67,1)","rgba(19,43,67,1)","rgba(19,43,67,1)","rgba(26,57,85,1)","rgba(30,64,96,1)","rgba(30,64,96,1)","rgba(30,64,96,1)","rgba(37,79,115,1)","rgba(43,90,130,1)","rgba(43,90,130,1)","rgba(43,90,130,1)","rgba(67,139,195,1)","rgba(71,148,207,1)","rgba(71,148,207,1)","rgba(71,148,207,1)","rgba(80,164,230,1)","rgba(86,177,247,1)","rgba(86,177,247,1)","rgba(86,177,247,1)"],"opacity":1,"size":5.66929133858268,"symbol":"x-thin-open","line":{"width":1.88976377952756,"color":["rgba(49,102,146,1)","rgba(55,114,162,1)","rgba(59,123,174,1)","rgba(59,123,174,1)","rgba(59,123,174,1)","rgba(22,49,75,1)","rgba(33,71,104,1)","rgba(62,130,183,1)","rgba(75,155,217,1)","rgba(19,43,67,1)","rgba(19,43,67,1)","rgba(19,43,67,1)","rgba(26,57,85,1)","rgba(30,64,96,1)","rgba(30,64,96,1)","rgba(30,64,96,1)","rgba(37,79,115,1)","rgba(43,90,130,1)","rgba(43,90,130,1)","rgba(43,90,130,1)","rgba(67,139,195,1)","rgba(71,148,207,1)","rgba(71,148,207,1)","rgba(71,148,207,1)","rgba(80,164,230,1)","rgba(86,177,247,1)","rgba(86,177,247,1)","rgba(86,177,247,1)"]}},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null},{"x":[3.32593559231609,3.30316386818886,0.751257736794651,2.33520994391292,2.95041866283864,3.20660052411258,2.82043528016657,2.97027577273548,2.93388118837029,0.980354832485318,2.37811781670898,2.76660826746374,3.34234547335654,1.03325146064162,2.02236271649599,2.61143268179148,3.34827230386436,0.970928431674838,2.14466817025095,3.11103716939688,3.05536238402128,0.608497302979231,1.91972585171461,2.87313463240862,2.70433387812227,1.32929426804185,2.38500399421901,2.74663059022278],"y":[1313.7199231958,1153.69116432425,1312.66500555445,1050.99844625536,1104.3367509855,999.031112464331,851.308397310972,880.300851889141,808.00631196741,1245.07202777471,1559.75180669837,1689.68827329725,1345.35406356622,1247.93300660867,1015.75449003652,1239.0436520787,1212.24248128552,1191.17893366758,1167.88855751846,1410.33628203999,1181.95142330155,1156.91736596841,1301.89601559956,1342.36420281343,1349.09338674471,1026.03889674395,1383.93801244125,1291.99954334088],"text":["mouse.id: 1699<br />dummy: 1314<br />day: 37","mouse.id: 1699<br />dummy: 1154<br />day: 44","mouse.id: 1697<br />dummy: 1313<br />day: 49","mouse.id: 1698<br />dummy: 1051<br />day: 49","mouse.id: 1699<br />dummy: 1104<br />day: 49","mouse.id: 1699<br />dummy:  999<br />day:  4","mouse.id: 1699<br />dummy:  851<br />day: 18","mouse.id: 1699<br />dummy:  880<br />day: 53","mouse.id: 1699<br />dummy:  808<br />day: 67","mouse.id: 1697<br />dummy: 1245<br />day:  0","mouse.id: 1698<br />dummy: 1560<br />day:  0","mouse.id: 1699<br />dummy: 1690<br />day:  0","mouse.id: 1699<br />dummy: 1345<br />day:  9","mouse.id: 1697<br />dummy: 1248<br />day: 14","mouse.id: 1698<br />dummy: 1016<br />day: 14","mouse.id: 1699<br />dummy: 1239<br />day: 14","mouse.id: 1699<br />dummy: 1212<br />day: 23","mouse.id: 1697<br />dummy: 1191<br />day: 30","mouse.id: 1698<br />dummy: 1168<br />day: 30","mouse.id: 1699<br />dummy: 1410<br />day: 30","mouse.id: 1699<br />dummy: 1182<br />day: 58","mouse.id: 1697<br />dummy: 1157<br />day: 63","mouse.id: 1698<br />dummy: 1302<br />day: 63","mouse.id: 1699<br />dummy: 1342<br />day: 63","mouse.id: 1699<br />dummy: 1349<br />day: 72","mouse.id: 1697<br />dummy: 1026<br />day: 79","mouse.id: 1698<br />dummy: 1384<br />day: 79","mouse.id: 1699<br />dummy: 1292<br />day: 79"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":["rgba(49,102,146,1)","rgba(55,114,162,1)","rgba(59,123,174,1)","rgba(59,123,174,1)","rgba(59,123,174,1)","rgba(22,49,75,1)","rgba(33,71,104,1)","rgba(62,130,183,1)","rgba(75,155,217,1)","rgba(19,43,67,1)","rgba(19,43,67,1)","rgba(19,43,67,1)","rgba(26,57,85,1)","rgba(30,64,96,1)","rgba(30,64,96,1)","rgba(30,64,96,1)","rgba(37,79,115,1)","rgba(43,90,130,1)","rgba(43,90,130,1)","rgba(43,90,130,1)","rgba(67,139,195,1)","rgba(71,148,207,1)","rgba(71,148,207,1)","rgba(71,148,207,1)","rgba(80,164,230,1)","rgba(86,177,247,1)","rgba(86,177,247,1)","rgba(86,177,247,1)"],"opacity":1,"size":5.66929133858268,"symbol":"x-thin-open","line":{"width":1.88976377952756,"color":["rgba(49,102,146,1)","rgba(55,114,162,1)","rgba(59,123,174,1)","rgba(59,123,174,1)","rgba(59,123,174,1)","rgba(22,49,75,1)","rgba(33,71,104,1)","rgba(62,130,183,1)","rgba(75,155,217,1)","rgba(19,43,67,1)","rgba(19,43,67,1)","rgba(19,43,67,1)","rgba(26,57,85,1)","rgba(30,64,96,1)","rgba(30,64,96,1)","rgba(30,64,96,1)","rgba(37,79,115,1)","rgba(43,90,130,1)","rgba(43,90,130,1)","rgba(43,90,130,1)","rgba(67,139,195,1)","rgba(71,148,207,1)","rgba(71,148,207,1)","rgba(71,148,207,1)","rgba(80,164,230,1)","rgba(86,177,247,1)","rgba(86,177,247,1)","rgba(86,177,247,1)"]}},"hoveron":"points","showlegend":false,"xaxis":"x3","yaxis":"y","hoverinfo":"text","frame":null},{"x":[1.62740011252463,0.981026475317776,2.28829694315791,3.14414447322488,1.87942777406424,1.90169148519635,2.31713452059776,1.71219500936568,0.886759524419904,1.8677228519693,3.04309800919145,1.33784955628216,2.34578842986375,3.23977579288185,2.30831927452236,0.762213982082903,2.12923604361713,2.63341678902507,2.3901833800599,1.33634471017867,1.74168569911271,2.64633751865476,1.68183555137366,0.677601812407374,2.03708385918289,2.93263548817486],"y":[1611.70055123735,1411.60621163547,1583.97446338106,1344.64392035883,1448.12975777481,1470.90562838912,1663.61526757479,1554.78365479074,1178.9214022845,1668.70865974166,1061.01247749813,1245.04520283211,1422.84137163926,1081.36721903272,1423.60433050971,1396.27639552858,1549.23597699311,1254.73672283199,1500.06788878962,1203.67594593093,1624.09809336495,1286.35412197541,1576.63733424451,1112.78494223244,1503.82755382601,1050.85737924427],"text":["mouse.id: 1683<br />dummy: 1612<br />day: 44","mouse.id: 1681<br />dummy: 1412<br />day: 49","mouse.id: 1683<br />dummy: 1584<br />day: 49","mouse.id: 1684<br />dummy: 1345<br />day: 49","mouse.id: 1683<br />dummy: 1448<br />day:  4","mouse.id: 1683<br />dummy: 1471<br />day: 18","mouse.id: 1683<br />dummy: 1664<br />day: 53","mouse.id: 1683<br />dummy: 1555<br />day: 67","mouse.id: 1681<br />dummy: 1179<br />day:  0","mouse.id: 1683<br />dummy: 1669<br />day:  0","mouse.id: 1684<br />dummy: 1061<br />day:  0","mouse.id: 1681<br />dummy: 1245<br />day: 14","mouse.id: 1683<br />dummy: 1423<br />day: 14","mouse.id: 1684<br />dummy: 1081<br />day: 14","mouse.id: 1683<br />dummy: 1424<br />day: 23","mouse.id: 1681<br />dummy: 1396<br />day: 30","mouse.id: 1683<br />dummy: 1549<br />day: 30","mouse.id: 1684<br />dummy: 1255<br />day: 30","mouse.id: 1683<br />dummy: 1500<br />day: 58","mouse.id: 1681<br />dummy: 1204<br />day: 63","mouse.id: 1683<br />dummy: 1624<br />day: 63","mouse.id: 1684<br />dummy: 1286<br />day: 63","mouse.id: 1683<br />dummy: 1577<br />day: 72","mouse.id: 1681<br />dummy: 1113<br />day: 79","mouse.id: 1683<br />dummy: 1504<br />day: 79","mouse.id: 1684<br />dummy: 1051<br />day: 79"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":["rgba(55,114,162,1)","rgba(59,123,174,1)","rgba(59,123,174,1)","rgba(59,123,174,1)","rgba(22,49,75,1)","rgba(33,71,104,1)","rgba(62,130,183,1)","rgba(75,155,217,1)","rgba(19,43,67,1)","rgba(19,43,67,1)","rgba(19,43,67,1)","rgba(30,64,96,1)","rgba(30,64,96,1)","rgba(30,64,96,1)","rgba(37,79,115,1)","rgba(43,90,130,1)","rgba(43,90,130,1)","rgba(43,90,130,1)","rgba(67,139,195,1)","rgba(71,148,207,1)","rgba(71,148,207,1)","rgba(71,148,207,1)","rgba(80,164,230,1)","rgba(86,177,247,1)","rgba(86,177,247,1)","rgba(86,177,247,1)"],"opacity":1,"size":5.66929133858268,"symbol":"x-thin-open","line":{"width":1.88976377952756,"color":["rgba(55,114,162,1)","rgba(59,123,174,1)","rgba(59,123,174,1)","rgba(59,123,174,1)","rgba(22,49,75,1)","rgba(33,71,104,1)","rgba(62,130,183,1)","rgba(75,155,217,1)","rgba(19,43,67,1)","rgba(19,43,67,1)","rgba(19,43,67,1)","rgba(30,64,96,1)","rgba(30,64,96,1)","rgba(30,64,96,1)","rgba(37,79,115,1)","rgba(43,90,130,1)","rgba(43,90,130,1)","rgba(43,90,130,1)","rgba(67,139,195,1)","rgba(71,148,207,1)","rgba(71,148,207,1)","rgba(71,148,207,1)","rgba(80,164,230,1)","rgba(86,177,247,1)","rgba(86,177,247,1)","rgba(86,177,247,1)"]}},"hoveron":"points","showlegend":false,"xaxis":"x4","yaxis":"y","hoverinfo":"text","frame":null},{"x":[2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,1,1,1,1,1,1,3,3,3,3,3,3],"y":[1331,1525,1369,1683,1480,1637,1499,1292,1323,1674,1313,1475,1449,1579,1497,1382,1576,1740,1747,1425,1775,1544,1851,1547,1648,1557,1664,1512],"hoverinfo":"y","type":"box","fillcolor":"rgba(255,255,255,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"showlegend":false,"xaxis":"x","yaxis":"y","frame":null},{"x":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,2,2,2,2,2,3,3,3,3,3,3],"y":[661,1562,1339,1288,1405,1457,1448,1021,1556,1548,2410,1698,1652,1438,1584,1542,1656,1648,1561,1767,1840,1659,1502,1667,1300,1236,1808,1598],"hoverinfo":"y","type":"box","fillcolor":"rgba(255,255,255,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"showlegend":false,"xaxis":"x2","yaxis":"y","frame":null},{"x":[3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,1,1,1,1,1,1,2,2,2,2,2,2],"y":[1154,880,1410,1690,1314,808,851,999,1239,1104,1182,1212,1349,1292,1342,1345,1191,1313,1157,1245,1026,1248,1016,1560,1302,1168,1051,1384],"hoverinfo":"y","type":"box","fillcolor":"rgba(255,255,255,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"showlegend":false,"xaxis":"x3","yaxis":"y","frame":null},{"x":[2,2,2,2,2,2,2,2,2,2,2,2,2,2,1,1,1,1,1,1,3,3,3,3,3,3],"y":[1549,1471,1612,1500,1584,1624,1669,1448,1555,1664,1504,1424,1423,1577,1412,1179,1204,1245,1113,1396,1061,1345,1286,1255,1081,1051],"hoverinfo":"y","type":"box","fillcolor":"rgba(255,255,255,1)","marker":{"opacity":null,"outliercolor":"rgba(0,0,0,1)","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)"},"size":5.66929133858268},"line":{"color":"rgba(51,51,51,1)","width":1.88976377952756},"showlegend":false,"xaxis":"x4","yaxis":"y","frame":null},{"x":[1],"y":[1000],"name":"99_7f1c47b14b9868e6aa379be97eb94aeb","type":"scatter","mode":"markers","opacity":0,"hoverinfo":"skip","showlegend":false,"marker":{"color":[0,1],"colorscale":[[0,"#132B43"],[0.0526315789473684,"#16314B"],[0.105263157894737,"#193754"],[0.157894736842105,"#1D3E5C"],[0.210526315789474,"#204465"],[0.263157894736842,"#234B6E"],[0.315789473684211,"#275277"],[0.368421052631579,"#2A5980"],[0.421052631578947,"#2E608A"],[0.473684210526316,"#316793"],[0.526315789473684,"#356E9D"],[0.578947368421053,"#3875A6"],[0.631578947368421,"#3C7CB0"],[0.684210526315789,"#3F83BA"],[0.736842105263158,"#438BC4"],[0.789473684210526,"#4792CE"],[0.842105263157895,"#4B9AD8"],[0.894736842105263,"#4EA2E2"],[0.947368421052632,"#52A9ED"],[1,"#56B1F7"]],"colorbar":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"thickness":23.04,"title":"day","titlefont":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"tickmode":"array","ticktext":["0","20","40","60"],"tickvals":[0,0.253164556962025,0.506329113924051,0.759493670886076],"tickfont":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"ticklen":2,"len":0.5}},"xaxis":"x","yaxis":"y","frame":null}],"layout":{"margin":{"t":42.3013698630137,"r":18.9954337899543,"b":44.5662100456621,"l":48.9497716894977},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,0.242389649923896],"automargin":true,"type":"linear","autorange":false,"range":[0.4,3.6],"tickmode":"array","ticktext":["1686","1688","1690"],"tickvals":[1,2,3],"categoryorder":"array","categoryarray":["1686","1688","1690"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"annotations":[{"text":"Mouse ID","x":0.5,"y":-0.0471841704718417,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"top","annotationType":"axis"},{"text":"number of variants","x":-0.0357686453576865,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis"},{"text":"Ciprofloxacin","x":0.121194824961948,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":11.689497716895},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"Tetracyclin","x":0.375,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":11.689497716895},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"Vancomycin","x":0.625,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":11.689497716895},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"Water","x":0.878805175038052,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":11.689497716895},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[573.534980796212,2497.76540327954],"tickmode":"array","ticktext":["1000","1500","2000"],"tickvals":[1000,1500,2000],"categoryorder":"array","categoryarray":["1000","1500","2000"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":0.242389649923896,"y0":0,"y1":1},{"type":"rect","fillcolor":"rgba(255,255,255,1)","line":{"color":"rgba(0,0,0,1)","width":1.32835201328352,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0,"x1":0.242389649923896,"y0":0,"y1":23.37899543379,"yanchor":1,"ysizemode":"pixel"},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.257610350076104,"x1":0.492389649923896,"y0":0,"y1":1},{"type":"rect","fillcolor":"rgba(255,255,255,1)","line":{"color":"rgba(0,0,0,1)","width":1.32835201328352,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0.257610350076104,"x1":0.492389649923896,"y0":0,"y1":23.37899543379,"yanchor":1,"ysizemode":"pixel"},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.507610350076104,"x1":0.742389649923896,"y0":0,"y1":1},{"type":"rect","fillcolor":"rgba(255,255,255,1)","line":{"color":"rgba(0,0,0,1)","width":1.32835201328352,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0.507610350076104,"x1":0.742389649923896,"y0":0,"y1":23.37899543379,"yanchor":1,"ysizemode":"pixel"},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.757610350076104,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":"rgba(255,255,255,1)","line":{"color":"rgba(0,0,0,1)","width":1.32835201328352,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0.757610350076104,"x1":1,"y0":0,"y1":23.37899543379,"yanchor":1,"ysizemode":"pixel"}],"xaxis2":{"type":"linear","autorange":false,"range":[0.4,3.6],"tickmode":"array","ticktext":["1692","1693","1694"],"tickvals":[1,2,3],"categoryorder":"array","categoryarray":["1692","1693","1694"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":false,"domain":[0.257610350076104,0.492389649923896],"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"xaxis3":{"type":"linear","autorange":false,"range":[0.4,3.6],"tickmode":"array","ticktext":["1697","1698","1699"],"tickvals":[1,2,3],"categoryorder":"array","categoryarray":["1697","1698","1699"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":false,"domain":[0.507610350076104,0.742389649923896],"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"xaxis4":{"type":"linear","autorange":false,"range":[0.4,3.6],"tickmode":"array","ticktext":["1681","1683","1684"],"tickvals":[1,2,3],"categoryorder":"array","categoryarray":["1681","1683","1684"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":false,"domain":[0.757610350076104,1],"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"5df1c1722c9":{"x":{},"y":{},"colour":{},"type":"scatter"},"5df32ecc887":{"x":{},"y":{},"colour":{}}},"cur_data":"5df1c1722c9","visdat":{"5df1c1722c9":["function (y) ","x"],"5df32ecc887":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 4.3: number of variants of all 12 OMM genomes by mouse
</p>
</div>
</div>
</div>
<div id="heatmap-1" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Heatmap</h2>
<p>All mice</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="ab-samples.html#cb13-1"></a>dat &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;data/rds/omm_ab.rds&quot;</span>)</span>
<span id="cb13-2"><a href="ab-samples.html#cb13-2"></a>dat<span class="op">$</span>sample.id &lt;-<span class="st"> </span><span class="kw">paste0</span>(dat<span class="op">$</span>mouse.id, <span class="st">&quot;-&quot;</span>,dat<span class="op">$</span>day)</span>
<span id="cb13-3"><a href="ab-samples.html#cb13-3"></a>dat<span class="op">$</span>variant.id &lt;-<span class="st"> </span><span class="kw">paste0</span>(dat<span class="op">$</span>POS, <span class="st">&quot;-&quot;</span>, dat<span class="op">$</span>REF, <span class="st">&quot;-&quot;</span>, dat<span class="op">$</span>ALT)</span>
<span id="cb13-4"><a href="ab-samples.html#cb13-4"></a>data.wide &lt;-<span class="st"> </span><span class="kw">dcast</span>(dat, variant.id <span class="op">~</span><span class="st"> </span>sample.id, <span class="dt">value.var =</span> <span class="st">&quot;AF&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in dcast(dat, variant.id ~ sample.id, value.var = &quot;AF&quot;): The dcast generic in data.table has been passed a data.frame and will attempt to redirect to the reshape2::dcast; please note that reshape2 is
## deprecated, and this redirection is now deprecated as well. Please do this redirection yourself like reshape2::dcast(dat). In the next version, this warning will become an error.</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="ab-samples.html#cb15-1"></a>data.wide[<span class="kw">is.na</span>(data.wide)] &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb15-2"><a href="ab-samples.html#cb15-2"></a><span class="kw">rownames</span>(data.wide) &lt;-<span class="st">  </span>data.wide<span class="op">$</span>variant.id</span>
<span id="cb15-3"><a href="ab-samples.html#cb15-3"></a>data.wide<span class="op">$</span>variant.id &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb15-4"><a href="ab-samples.html#cb15-4"></a></span>
<span id="cb15-5"><a href="ab-samples.html#cb15-5"></a><span class="kw">library</span>(circlize)</span>
<span id="cb15-6"><a href="ab-samples.html#cb15-6"></a><span class="kw">library</span>(ComplexHeatmap)</span>
<span id="cb15-7"><a href="ab-samples.html#cb15-7"></a></span>
<span id="cb15-8"><a href="ab-samples.html#cb15-8"></a>heat &lt;-<span class="st"> </span><span class="kw">data.matrix</span>(data.wide)</span>
<span id="cb15-9"><a href="ab-samples.html#cb15-9"></a><span class="co"># limit to variants that are present in at least 10% of samples</span></span>
<span id="cb15-10"><a href="ab-samples.html#cb15-10"></a>heat_num &lt;-<span class="st"> </span><span class="kw">rowSums</span>(heat <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>)</span>
<span id="cb15-11"><a href="ab-samples.html#cb15-11"></a>heat2 &lt;-<span class="st"> </span>heat[<span class="kw">which</span>(heat_num <span class="op">&gt;</span><span class="st"> </span><span class="kw">ncol</span>(heat)<span class="op">/</span><span class="dv">10</span>),]</span>
<span id="cb15-12"><a href="ab-samples.html#cb15-12"></a></span>
<span id="cb15-13"><a href="ab-samples.html#cb15-13"></a><span class="co"># limit to variants that have a high variance</span></span>
<span id="cb15-14"><a href="ab-samples.html#cb15-14"></a>heat_var_num &lt;-<span class="st"> </span>matrixStats<span class="op">::</span><span class="kw">rowVars</span>(heat2)</span>
<span id="cb15-15"><a href="ab-samples.html#cb15-15"></a>heat3 &lt;-<span class="st"> </span>heat2[<span class="kw">which</span>(heat_var_num <span class="op">&gt;</span><span class="st">  </span><span class="kw">quantile</span>(heat_var_num,  <span class="fl">0.5</span>)) ,]</span>
<span id="cb15-16"><a href="ab-samples.html#cb15-16"></a></span>
<span id="cb15-17"><a href="ab-samples.html#cb15-17"></a>dat<span class="op">$</span>dummy &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb15-18"><a href="ab-samples.html#cb15-18"></a>annot.data &lt;-<span class="st"> </span><span class="kw">aggregate</span>(dummy <span class="op">~</span><span class="st"> </span>mouse.id <span class="op">+</span><span class="st"> </span>mouse.group <span class="op">+</span><span class="st"> </span>day <span class="op">+</span><span class="st"> </span>phase, dat, sum)</span>
<span id="cb15-19"><a href="ab-samples.html#cb15-19"></a>annot.data<span class="op">$</span>sample.id &lt;-<span class="st"> </span><span class="kw">paste0</span>(annot.data<span class="op">$</span>mouse.id, <span class="st">&quot;-&quot;</span>,annot.data<span class="op">$</span>day)</span>
<span id="cb15-20"><a href="ab-samples.html#cb15-20"></a></span>
<span id="cb15-21"><a href="ab-samples.html#cb15-21"></a>heat3.day &lt;-<span class="st"> </span>annot.data[<span class="kw">match</span>(<span class="kw">colnames</span>(heat3), annot.data<span class="op">$</span>sample.id),]<span class="op">$</span>day</span>
<span id="cb15-22"><a href="ab-samples.html#cb15-22"></a>heat3.mouse.group &lt;-<span class="st"> </span>annot.data[<span class="kw">match</span>(<span class="kw">colnames</span>(heat3), annot.data<span class="op">$</span>sample.id),]<span class="op">$</span>mouse.group</span>
<span id="cb15-23"><a href="ab-samples.html#cb15-23"></a>heat3.phase &lt;-<span class="st"> </span>annot.data[<span class="kw">match</span>(<span class="kw">colnames</span>(heat3), annot.data<span class="op">$</span>sample.id),]<span class="op">$</span>phase</span>
<span id="cb15-24"><a href="ab-samples.html#cb15-24"></a></span>
<span id="cb15-25"><a href="ab-samples.html#cb15-25"></a>data.wide.sub &lt;-<span class="st"> </span>dat[<span class="kw">match</span>(<span class="kw">colnames</span>(heat3), dat<span class="op">$</span>sample.id),]</span>
<span id="cb15-26"><a href="ab-samples.html#cb15-26"></a></span>
<span id="cb15-27"><a href="ab-samples.html#cb15-27"></a>col_fun =<span class="st"> </span><span class="kw">colorRamp2</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>, <span class="dv">1</span>), <span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;yellow&quot;</span>, <span class="st">&quot;red&quot;</span>))</span>
<span id="cb15-28"><a href="ab-samples.html#cb15-28"></a></span>
<span id="cb15-29"><a href="ab-samples.html#cb15-29"></a><span class="co"># order the heatmap by treatment group</span></span>
<span id="cb15-30"><a href="ab-samples.html#cb15-30"></a></span>
<span id="cb15-31"><a href="ab-samples.html#cb15-31"></a><span class="co">#pdf(&quot;heat.pdf&quot;, width= 10, height = 10)</span></span>
<span id="cb15-32"><a href="ab-samples.html#cb15-32"></a><span class="kw">Heatmap</span>(heat3, <span class="dt">name =</span> <span class="st">&quot;AF&quot;</span>, <span class="dt">col =</span> col_fun, <span class="dt">border =</span> <span class="ot">TRUE</span>,</span>
<span id="cb15-33"><a href="ab-samples.html#cb15-33"></a><span class="dt">top_annotation =</span> <span class="kw">HeatmapAnnotation</span>(<span class="dt">num =</span> <span class="kw">anno_lines</span>(<span class="kw">colSums</span>(heat3),</span>
<span id="cb15-34"><a href="ab-samples.html#cb15-34"></a><span class="dt">smooth =</span> <span class="ot">TRUE</span>,<span class="dt">border =</span> <span class="ot">TRUE</span>),</span>
<span id="cb15-35"><a href="ab-samples.html#cb15-35"></a><span class="dt">day=</span>heat3.day),</span>
<span id="cb15-36"><a href="ab-samples.html#cb15-36"></a><span class="dt">cluster_columns =</span>F,</span>
<span id="cb15-37"><a href="ab-samples.html#cb15-37"></a><span class="dt">right_annotation =</span> <span class="kw">rowAnnotation</span>(<span class="dt">prev =</span> <span class="kw">anno_barplot</span>(<span class="kw">rowSums</span>(heat3))),</span>
<span id="cb15-38"><a href="ab-samples.html#cb15-38"></a> <span class="dt">row_gap =</span> <span class="kw">unit</span>(<span class="dv">0</span>, <span class="st">&quot;mm&quot;</span>), <span class="dt">column_gap =</span> <span class="kw">unit</span>(<span class="dv">0</span>, <span class="st">&quot;mm&quot;</span>),</span>
<span id="cb15-39"><a href="ab-samples.html#cb15-39"></a><span class="dt">column_split =</span> heat3.mouse.group,</span>
<span id="cb15-40"><a href="ab-samples.html#cb15-40"></a> <span class="dt">column_names_gp =</span> <span class="kw">gpar</span>(<span class="dt">fontsize =</span><span class="dv">5</span>),</span>
<span id="cb15-41"><a href="ab-samples.html#cb15-41"></a> <span class="dt">row_names_gp =</span> <span class="kw">gpar</span>(<span class="dt">fontsize =</span> <span class="dv">3</span>),</span>
<span id="cb15-42"><a href="ab-samples.html#cb15-42"></a> <span class="dt">show_row_dend =</span> F,</span>
<span id="cb15-43"><a href="ab-samples.html#cb15-43"></a>  <span class="dt">show_row_names =</span> F,</span>
<span id="cb15-44"><a href="ab-samples.html#cb15-44"></a> <span class="dt">show_column_dend =</span> F</span>
<span id="cb15-45"><a href="ab-samples.html#cb15-45"></a>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-7"></span>
<img src="oligomm-claudia_files/figure-html/unnamed-chunk-7-1.png" alt="number of variants of all 12 OMM genomes by mouse" width="1440" />
<p class="caption">
Figure 4.4: number of variants of all 12 OMM genomes by mouse
</p>
</div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="ab-samples.html#cb16-1"></a><span class="co">#dev.off()</span></span></code></pre></div>
</div>
<div id="focus-on-mouse-where-we-have-many-time-points" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> Focus on mouse where we have many time points</h2>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="ab-samples.html#cb17-1"></a>dat &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;data/rds/omm_ab.rds&quot;</span>)</span>
<span id="cb17-2"><a href="ab-samples.html#cb17-2"></a>dat<span class="op">$</span>rep.group &lt;-<span class="st"> </span><span class="kw">translateMouseIdToReplicateGroup</span>(dat<span class="op">$</span>mouse.id)</span>
<span id="cb17-3"><a href="ab-samples.html#cb17-3"></a>dat &lt;-<span class="st"> </span>dat[<span class="kw">which</span>(dat<span class="op">$</span>rep.group <span class="op">==</span><span class="st"> &quot;Full&quot;</span>),]</span>
<span id="cb17-4"><a href="ab-samples.html#cb17-4"></a></span>
<span id="cb17-5"><a href="ab-samples.html#cb17-5"></a>dat<span class="op">$</span>sample.id &lt;-<span class="st"> </span><span class="kw">paste0</span>(dat<span class="op">$</span>mouse.id, <span class="st">&quot;-&quot;</span>,dat<span class="op">$</span>day)</span>
<span id="cb17-6"><a href="ab-samples.html#cb17-6"></a>dat<span class="op">$</span>variant.id &lt;-<span class="st"> </span><span class="kw">paste0</span>(dat<span class="op">$</span>POS, <span class="st">&quot;-&quot;</span>, dat<span class="op">$</span>REF, <span class="st">&quot;-&quot;</span>, dat<span class="op">$</span>ALT)</span>
<span id="cb17-7"><a href="ab-samples.html#cb17-7"></a>data.wide &lt;-<span class="st"> </span><span class="kw">dcast</span>(dat, variant.id <span class="op">~</span><span class="st"> </span>sample.id, <span class="dt">value.var =</span> <span class="st">&quot;AF&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in dcast(dat, variant.id ~ sample.id, value.var = &quot;AF&quot;): The dcast generic in data.table has been passed a data.frame and will attempt to redirect to the reshape2::dcast; please note that reshape2 is
## deprecated, and this redirection is now deprecated as well. Please do this redirection yourself like reshape2::dcast(dat). In the next version, this warning will become an error.</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="ab-samples.html#cb19-1"></a>data.wide[<span class="kw">is.na</span>(data.wide)] &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb19-2"><a href="ab-samples.html#cb19-2"></a><span class="kw">rownames</span>(data.wide) &lt;-<span class="st">  </span>data.wide<span class="op">$</span>variant.id</span>
<span id="cb19-3"><a href="ab-samples.html#cb19-3"></a>data.wide<span class="op">$</span>variant.id &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb19-4"><a href="ab-samples.html#cb19-4"></a></span>
<span id="cb19-5"><a href="ab-samples.html#cb19-5"></a>heat &lt;-<span class="st"> </span><span class="kw">data.matrix</span>(data.wide)</span>
<span id="cb19-6"><a href="ab-samples.html#cb19-6"></a><span class="co"># limit to variants that are present in at least 10% of samples</span></span>
<span id="cb19-7"><a href="ab-samples.html#cb19-7"></a>heat_num &lt;-<span class="st"> </span><span class="kw">rowSums</span>(heat <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>)</span>
<span id="cb19-8"><a href="ab-samples.html#cb19-8"></a>heat2 &lt;-<span class="st"> </span>heat[<span class="kw">which</span>(heat_num <span class="op">&gt;</span><span class="st"> </span><span class="kw">ncol</span>(heat)<span class="op">/</span><span class="dv">10</span>),]</span>
<span id="cb19-9"><a href="ab-samples.html#cb19-9"></a></span>
<span id="cb19-10"><a href="ab-samples.html#cb19-10"></a><span class="co"># limit to variants that have a high variance</span></span>
<span id="cb19-11"><a href="ab-samples.html#cb19-11"></a>heat_var_num &lt;-<span class="st"> </span>matrixStats<span class="op">::</span><span class="kw">rowVars</span>(heat2)</span>
<span id="cb19-12"><a href="ab-samples.html#cb19-12"></a>heat3 &lt;-<span class="st"> </span>heat2[<span class="kw">which</span>(heat_var_num <span class="op">&gt;</span><span class="st">  </span><span class="kw">quantile</span>(heat_var_num,  <span class="fl">0.50</span>)) ,]</span>
<span id="cb19-13"><a href="ab-samples.html#cb19-13"></a></span>
<span id="cb19-14"><a href="ab-samples.html#cb19-14"></a>dat<span class="op">$</span>dummy &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb19-15"><a href="ab-samples.html#cb19-15"></a>annot.data &lt;-<span class="st"> </span><span class="kw">aggregate</span>(dummy <span class="op">~</span><span class="st"> </span>mouse.id <span class="op">+</span><span class="st"> </span>mouse.group <span class="op">+</span><span class="st"> </span>day <span class="op">+</span><span class="st"> </span>phase, dat, sum)</span>
<span id="cb19-16"><a href="ab-samples.html#cb19-16"></a>annot.data<span class="op">$</span>sample.id &lt;-<span class="st"> </span><span class="kw">paste0</span>(annot.data<span class="op">$</span>mouse.id, <span class="st">&quot;-&quot;</span>,annot.data<span class="op">$</span>day)</span>
<span id="cb19-17"><a href="ab-samples.html#cb19-17"></a></span>
<span id="cb19-18"><a href="ab-samples.html#cb19-18"></a></span>
<span id="cb19-19"><a href="ab-samples.html#cb19-19"></a>heat3.mouse.id &lt;-<span class="st"> </span>annot.data[<span class="kw">match</span>(<span class="kw">colnames</span>(heat3), annot.data<span class="op">$</span>sample.id),]<span class="op">$</span>mouse.id</span>
<span id="cb19-20"><a href="ab-samples.html#cb19-20"></a>heat3.day &lt;-<span class="st"> </span>annot.data[<span class="kw">match</span>(<span class="kw">colnames</span>(heat3), annot.data<span class="op">$</span>sample.id),]<span class="op">$</span>day</span>
<span id="cb19-21"><a href="ab-samples.html#cb19-21"></a>heat3.mouse.group &lt;-<span class="st"> </span>annot.data[<span class="kw">match</span>(<span class="kw">colnames</span>(heat3), annot.data<span class="op">$</span>sample.id),]<span class="op">$</span>mouse.group</span>
<span id="cb19-22"><a href="ab-samples.html#cb19-22"></a>heat3.phase &lt;-<span class="st"> </span>annot.data[<span class="kw">match</span>(<span class="kw">colnames</span>(heat3), annot.data<span class="op">$</span>sample.id),]<span class="op">$</span>phase</span>
<span id="cb19-23"><a href="ab-samples.html#cb19-23"></a>heat3.phase2 &lt;-<span class="st"> </span><span class="kw">ifelse</span>(heat3.phase <span class="op">==</span><span class="st"> &quot;post-treatment&quot;</span>, <span class="dv">6</span>, <span class="ot">NA</span>)</span>
<span id="cb19-24"><a href="ab-samples.html#cb19-24"></a></span>
<span id="cb19-25"><a href="ab-samples.html#cb19-25"></a>col_fun =<span class="st"> </span><span class="kw">colorRamp2</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>, <span class="dv">1</span>), <span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;yellow&quot;</span>, <span class="st">&quot;red&quot;</span>))</span>
<span id="cb19-26"><a href="ab-samples.html#cb19-26"></a></span>
<span id="cb19-27"><a href="ab-samples.html#cb19-27"></a></span>
<span id="cb19-28"><a href="ab-samples.html#cb19-28"></a><span class="co"># order the heatmap by treatment group</span></span>
<span id="cb19-29"><a href="ab-samples.html#cb19-29"></a></span>
<span id="cb19-30"><a href="ab-samples.html#cb19-30"></a><span class="co">#pdf(&quot;heat3.pdf&quot;, width= 20, height = 25)</span></span>
<span id="cb19-31"><a href="ab-samples.html#cb19-31"></a><span class="kw">Heatmap</span>(heat3, <span class="dt">name =</span> <span class="st">&quot;AF&quot;</span>, <span class="dt">col =</span> col_fun, <span class="dt">border =</span> <span class="ot">TRUE</span>,</span>
<span id="cb19-32"><a href="ab-samples.html#cb19-32"></a><span class="dt">top_annotation =</span> <span class="kw">HeatmapAnnotation</span>(<span class="dt">num =</span> <span class="kw">anno_lines</span>(<span class="kw">colSums</span>(heat3),</span>
<span id="cb19-33"><a href="ab-samples.html#cb19-33"></a><span class="dt">smooth =</span> <span class="ot">TRUE</span>,<span class="dt">border =</span> <span class="ot">TRUE</span>),</span>
<span id="cb19-34"><a href="ab-samples.html#cb19-34"></a><span class="dt">day=</span><span class="kw">anno_simple</span>(heat3.day, <span class="dt">pch =</span>heat3.phase2 )),</span>
<span id="cb19-35"><a href="ab-samples.html#cb19-35"></a><span class="dt">cluster_columns =</span>F,</span>
<span id="cb19-36"><a href="ab-samples.html#cb19-36"></a><span class="dt">column_split =</span> heat3.mouse.group,</span>
<span id="cb19-37"><a href="ab-samples.html#cb19-37"></a> <span class="dt">column_names_gp =</span> <span class="kw">gpar</span>(<span class="dt">fontsize =</span><span class="dv">18</span>),</span>
<span id="cb19-38"><a href="ab-samples.html#cb19-38"></a> <span class="dt">row_names_gp =</span> <span class="kw">gpar</span>(<span class="dt">fontsize =</span> <span class="dv">8</span>),</span>
<span id="cb19-39"><a href="ab-samples.html#cb19-39"></a> <span class="dt">show_row_dend =</span> F,</span>
<span id="cb19-40"><a href="ab-samples.html#cb19-40"></a>  <span class="dt">show_row_names =</span> F,</span>
<span id="cb19-41"><a href="ab-samples.html#cb19-41"></a> <span class="dt">show_column_dend =</span> F</span>
<span id="cb19-42"><a href="ab-samples.html#cb19-42"></a>)</span></code></pre></div>
<p><img src="oligomm-claudia_files/figure-html/unnamed-chunk-8-1.png" width="1440" /></p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="ab-samples.html#cb20-1"></a><span class="co">#dev.off()</span></span></code></pre></div>
</div>
<div id="akkermansia-muciniphila" class="section level2" number="4.6">
<h2><span class="header-section-number">4.6</span> Akkermansia Muciniphila</h2>
<div id="area-plot-1" class="section level3" number="4.6.1">
<h3><span class="header-section-number">4.6.1</span> area plot 1</h3>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="ab-samples.html#cb21-1"></a>    dat &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;data/rds/omm_ab.rds&quot;</span>)</span>
<span id="cb21-2"><a href="ab-samples.html#cb21-2"></a>    <span class="kw">message</span>(genome)</span>
<span id="cb21-3"><a href="ab-samples.html#cb21-3"></a>    dat<span class="op">$</span>variant.id &lt;-<span class="st"> </span><span class="kw">paste0</span>(dat<span class="op">$</span>POS, <span class="st">&quot;-&quot;</span>,dat<span class="op">$</span>REF, <span class="st">&quot;-&quot;</span>, dat<span class="op">$</span>ALT)</span>
<span id="cb21-4"><a href="ab-samples.html#cb21-4"></a>    dat &lt;-<span class="st"> </span>dat[<span class="kw">which</span>(dat<span class="op">$</span>chr <span class="op">==</span><span class="st"> &quot;Akkermansia Muciniphila&quot;</span>),]</span>
<span id="cb21-5"><a href="ab-samples.html#cb21-5"></a>    data.wide &lt;-<span class="st"> </span><span class="kw">dcast</span>(dat, day <span class="op">+</span><span class="st"> </span>mouse.id <span class="op">+</span><span class="st"> </span>mouse.group<span class="op">~</span>variant.id, <span class="dt">value.var =</span> <span class="st">&quot;AF&quot;</span>)</span>
<span id="cb21-6"><a href="ab-samples.html#cb21-6"></a>    data.wide[<span class="kw">is.na</span>(data.wide)] &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb21-7"><a href="ab-samples.html#cb21-7"></a></span>
<span id="cb21-8"><a href="ab-samples.html#cb21-8"></a>    colMax &lt;-<span class="st"> </span><span class="cf">function</span>(X) <span class="kw">apply</span>(X, <span class="dv">2</span>, max)</span>
<span id="cb21-9"><a href="ab-samples.html#cb21-9"></a>    dat_mat  &lt;-<span class="st"> </span>data.wide[,<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)]</span>
<span id="cb21-10"><a href="ab-samples.html#cb21-10"></a></span>
<span id="cb21-11"><a href="ab-samples.html#cb21-11"></a>    <span class="co"># filter variants </span></span>
<span id="cb21-12"><a href="ab-samples.html#cb21-12"></a>    data.wide.reduced &lt;-<span class="st"> </span><span class="kw">cbind</span>(data.wide[,<span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)],dat_mat[,<span class="kw">which</span>(<span class="kw">colMax</span>(dat_mat)<span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>)])</span>
<span id="cb21-13"><a href="ab-samples.html#cb21-13"></a>    <span class="co">#data.wide.reduced &lt;- data.wide</span></span>
<span id="cb21-14"><a href="ab-samples.html#cb21-14"></a>    <span class="cf">if</span>(<span class="kw">ncol</span>(data.wide.reduced)  <span class="op">&gt;</span><span class="st"> </span><span class="dv">3</span>) {</span>
<span id="cb21-15"><a href="ab-samples.html#cb21-15"></a>        dat2 &lt;-<span class="st"> </span><span class="kw">melt</span>(data.wide.reduced, <span class="dt">id.vars =</span> <span class="kw">c</span>(<span class="st">&quot;day&quot;</span>,<span class="st">&quot;mouse.id&quot;</span>, <span class="st">&quot;mouse.group&quot;</span>))</span>
<span id="cb21-16"><a href="ab-samples.html#cb21-16"></a>        dat3 &lt;-<span class="st"> </span>dat2 <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(day, mouse.id) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">Nor =</span> value<span class="op">/</span><span class="kw">sum</span>(value))</span>
<span id="cb21-17"><a href="ab-samples.html#cb21-17"></a></span>
<span id="cb21-18"><a href="ab-samples.html#cb21-18"></a>        <span class="kw">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb21-19"><a href="ab-samples.html#cb21-19"></a></span>
<span id="cb21-20"><a href="ab-samples.html#cb21-20"></a>        col_list &lt;-<span class="st"> </span><span class="kw">sort</span>(<span class="kw">unique</span>(dat3<span class="op">$</span>variable))</span>
<span id="cb21-21"><a href="ab-samples.html#cb21-21"></a>        cols &lt;-randomcoloR<span class="op">::</span><span class="kw">randomColor</span>(<span class="kw">length</span>(<span class="kw">unique</span>(dat3<span class="op">$</span>variable)))   </span>
<span id="cb21-22"><a href="ab-samples.html#cb21-22"></a></span>
<span id="cb21-23"><a href="ab-samples.html#cb21-23"></a>        <span class="co"># Muller plot</span></span>
<span id="cb21-24"><a href="ab-samples.html#cb21-24"></a>        p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(dat3, <span class="kw">aes</span>(<span class="dt">x =</span> day, <span class="dt">y =</span> Nor, <span class="dt">group =</span> variable,</span>
<span id="cb21-25"><a href="ab-samples.html#cb21-25"></a>        <span class="dt">fill=</span>variable))</span>
<span id="cb21-26"><a href="ab-samples.html#cb21-26"></a>        p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_area</span>(<span class="dt">color =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">size =</span> <span class="fl">0.1</span>) </span>
<span id="cb21-27"><a href="ab-samples.html#cb21-27"></a>        p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="st"> </span>mouse.group <span class="op">+</span><span class="st"> </span>mouse.id, <span class="dt">ncol=</span><span class="dv">3</span>)</span>
<span id="cb21-28"><a href="ab-samples.html#cb21-28"></a>        p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme_minimal</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb21-29"><a href="ab-samples.html#cb21-29"></a>        p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Fraction&quot;</span>)</span>
<span id="cb21-30"><a href="ab-samples.html#cb21-30"></a>        p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span> cols, <span class="dt">breaks =</span> col_list)</span>
<span id="cb21-31"><a href="ab-samples.html#cb21-31"></a>        p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(genome)</span>
<span id="cb21-32"><a href="ab-samples.html#cb21-32"></a>        <span class="kw">print</span>(p)</span>
<span id="cb21-33"><a href="ab-samples.html#cb21-33"></a>    }</span></code></pre></div>
</div>
<div id="line-plot" class="section level3" number="4.6.2">
<h3><span class="header-section-number">4.6.2</span> line plot</h3>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="ab-samples.html#cb22-1"></a>    dat &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;data/rds/omm_ab.rds&quot;</span>)</span>
<span id="cb22-2"><a href="ab-samples.html#cb22-2"></a>    <span class="kw">message</span>(genome)</span>
<span id="cb22-3"><a href="ab-samples.html#cb22-3"></a>    dat<span class="op">$</span>variant.id &lt;-<span class="st"> </span><span class="kw">paste0</span>(dat<span class="op">$</span>POS, <span class="st">&quot;-&quot;</span>,dat<span class="op">$</span>REF, <span class="st">&quot;-&quot;</span>, dat<span class="op">$</span>ALT)</span>
<span id="cb22-4"><a href="ab-samples.html#cb22-4"></a>    dat &lt;-<span class="st"> </span>dat[<span class="kw">which</span>(dat<span class="op">$</span>chr <span class="op">==</span><span class="st"> &quot;Akkermansia Muciniphila&quot;</span>),]</span>
<span id="cb22-5"><a href="ab-samples.html#cb22-5"></a>    data.wide &lt;-<span class="st"> </span><span class="kw">dcast</span>(dat, day <span class="op">+</span><span class="st"> </span>mouse.id <span class="op">+</span><span class="st"> </span>mouse.group<span class="op">~</span>variant.id, <span class="dt">value.var =</span> <span class="st">&quot;AF&quot;</span>)</span>
<span id="cb22-6"><a href="ab-samples.html#cb22-6"></a>    data.wide[<span class="kw">is.na</span>(data.wide)] &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb22-7"><a href="ab-samples.html#cb22-7"></a>    dat2 &lt;-<span class="st"> </span><span class="kw">melt</span>(data.wide, <span class="dt">id.vars =</span> <span class="kw">c</span>(<span class="st">&quot;day&quot;</span>,<span class="st">&quot;mouse.id&quot;</span>, <span class="st">&quot;mouse.group&quot;</span>))</span>
<span id="cb22-8"><a href="ab-samples.html#cb22-8"></a></span>
<span id="cb22-9"><a href="ab-samples.html#cb22-9"></a>    <span class="kw">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb22-10"><a href="ab-samples.html#cb22-10"></a>    col_list &lt;-<span class="st"> </span><span class="kw">sort</span>(<span class="kw">unique</span>(dat3<span class="op">$</span>variable))</span>
<span id="cb22-11"><a href="ab-samples.html#cb22-11"></a>    cols &lt;-randomcoloR<span class="op">::</span><span class="kw">randomColor</span>(<span class="kw">length</span>(<span class="kw">unique</span>(dat3<span class="op">$</span>variable)))   </span>
<span id="cb22-12"><a href="ab-samples.html#cb22-12"></a></span>
<span id="cb22-13"><a href="ab-samples.html#cb22-13"></a>        p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(dat2, <span class="kw">aes</span>(<span class="dt">x =</span> day, <span class="dt">y =</span>value))</span>
<span id="cb22-14"><a href="ab-samples.html#cb22-14"></a>        p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">group =</span> variable), <span class="dt">alpha=</span> <span class="fl">0.2</span>) </span>
<span id="cb22-15"><a href="ab-samples.html#cb22-15"></a>        p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>mouse.group <span class="op">+</span><span class="st"> </span>mouse.id, <span class="dt">ncol=</span><span class="dv">3</span>)</span>
<span id="cb22-16"><a href="ab-samples.html#cb22-16"></a>        plotly<span class="op">::</span><span class="kw">ggplotly</span>(p)</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="resequencing-samples.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
