<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Section 7 Comparision of lofreq and varscan | Claudia’s data</title>
  <meta name="description" content="Section 7 Comparision of lofreq and varscan | Claudia’s data" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="Section 7 Comparision of lofreq and varscan | Claudia’s data" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Section 7 Comparision of lofreq and varscan | Claudia’s data" />
  
  
  

<meta name="author" content="Philipp C. Muench" />


<meta name="date" content="2020-04-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Allele-frequency.html"/>

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Analysis Report</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#how-to-generate-this-report"><i class="fa fa-check"></i>How to generate this report</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="Setup.html"><a href="Setup.html"><i class="fa fa-check"></i><b>1</b> Setup</a><ul>
<li class="chapter" data-level="1.1" data-path="Setup.html"><a href="Setup.html#Workspace_setup"><i class="fa fa-check"></i><b>1.1</b> Workspace setup</a></li>
<li class="chapter" data-level="1.2" data-path="Setup.html"><a href="Setup.html#dependencies"><i class="fa fa-check"></i><b>1.2</b> Dependencies</a></li>
<li class="chapter" data-level="1.3" data-path="Setup.html"><a href="Setup.html#Data_structures"><i class="fa fa-check"></i><b>1.3</b> Data structure</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="study-design.html"><a href="study-design.html"><i class="fa fa-check"></i><b>2</b> Study design</a><ul>
<li class="chapter" data-level="2.1" data-path="study-design.html"><a href="study-design.html#experiment"><i class="fa fa-check"></i><b>2.1</b> Experiment</a></li>
<li class="chapter" data-level="2.2" data-path="study-design.html"><a href="study-design.html#metadata"><i class="fa fa-check"></i><b>2.2</b> Metadata</a></li>
<li class="chapter" data-level="2.3" data-path="study-design.html"><a href="study-design.html#oligomm"><i class="fa fa-check"></i><b>2.3</b> OligoMM</a><ul>
<li class="chapter" data-level="2.3.1" data-path="study-design.html"><a href="study-design.html#crispr"><i class="fa fa-check"></i><b>2.3.1</b> CRISPR</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="Variant-statistics.html"><a href="Variant-statistics.html"><i class="fa fa-check"></i><b>3</b> Variant statistics</a><ul>
<li class="chapter" data-level="3.1" data-path="Variant-statistics.html"><a href="Variant-statistics.html#process-of-raw-data"><i class="fa fa-check"></i><b>3.1</b> Process of <em>raw</em> data</a></li>
<li class="chapter" data-level="3.2" data-path="Variant-statistics.html"><a href="Variant-statistics.html#quality-of-variants"><i class="fa fa-check"></i><b>3.2</b> Quality of variants</a><ul>
<li class="chapter" data-level="3.2.1" data-path="Variant-statistics.html"><a href="Variant-statistics.html#lofreq"><i class="fa fa-check"></i><b>3.2.1</b> lofreq</a></li>
<li class="chapter" data-level="3.2.2" data-path="Variant-statistics.html"><a href="Variant-statistics.html#all-samples"><i class="fa fa-check"></i><b>3.2.2</b> all samples</a></li>
<li class="chapter" data-level="3.2.3" data-path="Variant-statistics.html"><a href="Variant-statistics.html#by-mouse"><i class="fa fa-check"></i><b>3.2.3</b> by mouse</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="Variant-statistics.html"><a href="Variant-statistics.html#number-of-variants"><i class="fa fa-check"></i><b>3.3</b> Number of variants</a><ul>
<li class="chapter" data-level="3.3.1" data-path="Variant-statistics.html"><a href="Variant-statistics.html#by-sample"><i class="fa fa-check"></i><b>3.3.1</b> by sample</a></li>
<li class="chapter" data-level="3.3.2" data-path="Variant-statistics.html"><a href="Variant-statistics.html#by-mouse-and-genome"><i class="fa fa-check"></i><b>3.3.2</b> by mouse and genome</a></li>
<li class="chapter" data-level="3.3.3" data-path="Variant-statistics.html"><a href="Variant-statistics.html#by-extendet-metadata"><i class="fa fa-check"></i><b>3.3.3</b> by extendet metadata</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="Variant-statistics.html"><a href="Variant-statistics.html#orfs-with-variants"><i class="fa fa-check"></i><b>3.4</b> ORFs with variants</a><ul>
<li class="chapter" data-level="3.4.1" data-path="Variant-statistics.html"><a href="Variant-statistics.html#data"><i class="fa fa-check"></i><b>3.4.1</b> Data</a></li>
<li class="chapter" data-level="3.4.2" data-path="Variant-statistics.html"><a href="Variant-statistics.html#statistics"><i class="fa fa-check"></i><b>3.4.2</b> Statistics</a></li>
<li class="chapter" data-level="3.4.3" data-path="Variant-statistics.html"><a href="Variant-statistics.html#minor-to-major-transistion"><i class="fa fa-check"></i><b>3.4.3</b> Minor to major transistion</a></li>
<li class="chapter" data-level="3.4.4" data-path="Variant-statistics.html"><a href="Variant-statistics.html#gene-centric-analysis"><i class="fa fa-check"></i><b>3.4.4</b> Gene-centric analysis</a></li>
<li class="chapter" data-level="3.4.5" data-path="Variant-statistics.html"><a href="Variant-statistics.html#by-variant-static"><i class="fa fa-check"></i><b>3.4.5</b> by variant (static)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="Variant-statistics-filtered.html"><a href="Variant-statistics-filtered.html"><i class="fa fa-check"></i><b>4</b> Variant statistics relative to <em>I_mi</em></a><ul>
<li class="chapter" data-level="4.1" data-path="Variant-statistics-filtered.html"><a href="Variant-statistics-filtered.html#number-of-variants-1"><i class="fa fa-check"></i><b>4.1</b> Number of variants</a><ul>
<li class="chapter" data-level="4.1.1" data-path="Variant-statistics-filtered.html"><a href="Variant-statistics-filtered.html#by-sample-1"><i class="fa fa-check"></i><b>4.1.1</b> by sample</a></li>
<li class="chapter" data-level="4.1.2" data-path="Variant-statistics-filtered.html"><a href="Variant-statistics-filtered.html#by-extendet-metadata-1"><i class="fa fa-check"></i><b>4.1.2</b> by extendet metadata</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="Variant-statistics-filtered.html"><a href="Variant-statistics-filtered.html#orfs-with-variants-1"><i class="fa fa-check"></i><b>4.2</b> ORFs with variants</a><ul>
<li class="chapter" data-level="4.2.1" data-path="Variant-statistics-filtered.html"><a href="Variant-statistics-filtered.html#data-1"><i class="fa fa-check"></i><b>4.2.1</b> Data</a></li>
<li class="chapter" data-level="4.2.2" data-path="Variant-statistics-filtered.html"><a href="Variant-statistics-filtered.html#statistics-1"><i class="fa fa-check"></i><b>4.2.2</b> Statistics</a></li>
<li class="chapter" data-level="4.2.3" data-path="Variant-statistics-filtered.html"><a href="Variant-statistics-filtered.html#minor-to-major-transistion-1"><i class="fa fa-check"></i><b>4.2.3</b> Minor to major transistion</a></li>
<li class="chapter" data-level="4.2.4" data-path="Variant-statistics-filtered.html"><a href="Variant-statistics-filtered.html#gene-centric-analysis-1"><i class="fa fa-check"></i><b>4.2.4</b> Gene-centric analysis</a></li>
<li class="chapter" data-level="4.2.5" data-path="Variant-statistics-filtered.html"><a href="Variant-statistics-filtered.html#by-variant-static-1"><i class="fa fa-check"></i><b>4.2.5</b> by variant (static)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="variant-effects.html"><a href="variant-effects.html"><i class="fa fa-check"></i><b>5</b> Variant effects</a><ul>
<li class="chapter" data-level="5.1" data-path="variant-effects.html"><a href="variant-effects.html#snpeff-run"><i class="fa fa-check"></i><b>5.1</b> snpEff run</a></li>
<li class="chapter" data-level="5.2" data-path="variant-effects.html"><a href="variant-effects.html#number-of-effects-by-type"><i class="fa fa-check"></i><b>5.2</b> number of effects by type</a></li>
<li class="chapter" data-level="5.3" data-path="variant-effects.html"><a href="variant-effects.html#number-of-effects-by-type-by-genome-vartype"><i class="fa fa-check"></i><b>5.3</b> number of effects by type by genome (vartype)</a></li>
<li class="chapter" data-level="5.4" data-path="variant-effects.html"><a href="variant-effects.html#number-of-effects-by-type-by-genome-effect"><i class="fa fa-check"></i><b>5.4</b> number of effects by type by genome (effect)</a></li>
<li class="chapter" data-level="5.5" data-path="variant-effects.html"><a href="variant-effects.html#by-genome-and-mouse"><i class="fa fa-check"></i><b>5.5</b> by genome and mouse</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Allele-frequency.html"><a href="Allele-frequency.html"><i class="fa fa-check"></i><b>6</b> Allele frequency</a><ul>
<li class="chapter" data-level="6.1" data-path="Allele-frequency.html"><a href="Allele-frequency.html#major-allele-frequency"><i class="fa fa-check"></i><b>6.1</b> major allele frequency</a><ul>
<li class="chapter" data-level="6.1.1" data-path="Allele-frequency.html"><a href="Allele-frequency.html#all-samples-and-all-genomes"><i class="fa fa-check"></i><b>6.1.1</b> all samples and all genomes</a></li>
<li class="chapter" data-level="6.1.2" data-path="Allele-frequency.html"><a href="Allele-frequency.html#by-sample-and-genome"><i class="fa fa-check"></i><b>6.1.2</b> by sample and genome</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="comparision-of-lofreq-and-varscan.html"><a href="comparision-of-lofreq-and-varscan.html"><i class="fa fa-check"></i><b>7</b> Comparision of lofreq and varscan</a><ul>
<li class="chapter" data-level="7.1" data-path="comparision-of-lofreq-and-varscan.html"><a href="comparision-of-lofreq-and-varscan.html#major-allele-frequency-1"><i class="fa fa-check"></i><b>7.1</b> major allele frequency</a><ul>
<li class="chapter" data-level="7.1.1" data-path="comparision-of-lofreq-and-varscan.html"><a href="comparision-of-lofreq-and-varscan.html#all-samples-and-all-genomes-1"><i class="fa fa-check"></i><b>7.1.1</b> all samples and all genomes</a></li>
<li class="chapter" data-level="7.1.2" data-path="comparision-of-lofreq-and-varscan.html"><a href="comparision-of-lofreq-and-varscan.html#by-sample-and-genome-1"><i class="fa fa-check"></i><b>7.1.2</b> by sample and genome</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://pmuench.com" target="blank">Philipp C. Muench</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Claudia’s data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="comparision-of-lofreq-and-varscan" class="section level1">
<h1><span class="header-section-number">Section 7</span> Comparision of lofreq and varscan</h1>
<p>without <em>E. Coli</em></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">dat.lo &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;data/rds/variants_lofreq.rds&quot;</span>)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">dat.lo &lt;-<span class="st"> </span>dat.lo[<span class="kw">which</span>(dat.lo<span class="op">$</span>ecoli<span class="op">==</span><span class="st"> </span><span class="ot">FALSE</span>),]</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">dat.va &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;data/rds/variants_varscan.rds&quot;</span>)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">dat.va &lt;-<span class="st"> </span>dat.va[<span class="kw">which</span>(dat.va<span class="op">$</span>ecoli<span class="op">==</span><span class="st"> </span><span class="ot">FALSE</span>),]</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"></a>
<a class="sourceLine" id="cb1-6" data-line-number="6">dat.va<span class="op">$</span>dummy &lt;-<span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb1-7" data-line-number="7">dat.lo<span class="op">$</span>dummy &lt;-<span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb1-8" data-line-number="8">dat.va.by.sample.genome &lt;-<span class="st"> </span><span class="kw">aggregate</span>(dummy <span class="op">~</span><span class="st"> </span>mouse.id <span class="op">+</span><span class="st"> </span>genome, dat.va, sum)</a>
<a class="sourceLine" id="cb1-9" data-line-number="9">dat.va.by.sample.genome<span class="op">$</span>type &lt;-<span class="st"> &quot;varscan&quot;</span></a>
<a class="sourceLine" id="cb1-10" data-line-number="10">dat.lo.by.sample.genome &lt;-<span class="st"> </span><span class="kw">aggregate</span>(dummy <span class="op">~</span><span class="st"> </span>mouse.id <span class="op">+</span><span class="st"> </span>genome, dat.lo, sum)</a>
<a class="sourceLine" id="cb1-11" data-line-number="11">dat.lo.by.sample.genome<span class="op">$</span>type &lt;-<span class="st"> &quot;lofreq&quot;</span></a>
<a class="sourceLine" id="cb1-12" data-line-number="12"></a>
<a class="sourceLine" id="cb1-13" data-line-number="13">joined.df &lt;-<span class="st"> </span><span class="kw">rbind</span>(dat.lo.by.sample.genome, dat.va.by.sample.genome)</a>
<a class="sourceLine" id="cb1-14" data-line-number="14"></a>
<a class="sourceLine" id="cb1-15" data-line-number="15">p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(joined.df, <span class="kw">aes</span>(<span class="dt">x =</span> mouse.id, <span class="dt">y=</span>dummy, <span class="dt">fill =</span> type))</a>
<a class="sourceLine" id="cb1-16" data-line-number="16">p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_log10</span>() <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Mouse ID&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;number of variants&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>()</a>
<a class="sourceLine" id="cb1-17" data-line-number="17">p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="dt">position =</span> <span class="kw">position_dodge2</span>(<span class="dt">preserve =</span> <span class="st">&quot;single&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="st"> </span>genome)</a>
<a class="sourceLine" id="cb1-18" data-line-number="18">p</a></code></pre></div>
<p><img src="oligomm-claudia_files/figure-html/unnamed-chunk-1-1.png" width="1440" style="display: block; margin: auto;" /></p>
<div id="major-allele-frequency-1" class="section level2">
<h2><span class="header-section-number">7.1</span> major allele frequency</h2>
<div id="all-samples-and-all-genomes-1" class="section level3">
<h3><span class="header-section-number">7.1.1</span> all samples and all genomes</h3>
<div id="lofreq-1" class="section level4">
<h4><span class="header-section-number">7.1.1.1</span> lofreq</h4>
<p>without <em>E. Coli</em></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(dat.lo, <span class="kw">aes</span>(<span class="dt">x =</span> majorAF, <span class="dt">y =</span> mouse.id))</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span>ggridges<span class="op">::</span><span class="kw">geom_density_ridges</span>(<span class="dt">jittered_points =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">  <span class="dt">position =</span> ggridges<span class="op">::</span><span class="kw">position_points_jitter</span>(<span class="dt">width =</span> <span class="fl">0.01</span>, <span class="dt">height =</span> <span class="dv">0</span>),</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">  <span class="dt">point_shape =</span> <span class="st">&#39;|&#39;</span>, <span class="dt">point_size =</span> <span class="dv">1</span>, <span class="dt">point_alpha =</span> <span class="fl">0.5</span>, <span class="dt">alpha =</span> <span class="fl">0.1</span>)</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() </a>
<a class="sourceLine" id="cb2-6" data-line-number="6">p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">strip.background =</span> <span class="kw">element_blank</span>()) </a>
<a class="sourceLine" id="cb2-7" data-line-number="7">p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Major allele frequency&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Count&quot;</span>) </a>
<a class="sourceLine" id="cb2-8" data-line-number="8">p</a></code></pre></div>
<pre><code>## Picking joint bandwidth of 0.0632</code></pre>
<p><img src="oligomm-claudia_files/figure-html/unnamed-chunk-2-1.png" width="1440" /></p>
</div>
<div id="varscan" class="section level4">
<h4><span class="header-section-number">7.1.1.2</span> varscan</h4>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(dat.va, <span class="kw">aes</span>(<span class="dt">x =</span> majorAF, <span class="dt">y =</span> mouse.id))</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span>ggridges<span class="op">::</span><span class="kw">geom_density_ridges</span>(<span class="dt">jittered_points =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">  <span class="dt">position =</span> ggridges<span class="op">::</span><span class="kw">position_points_jitter</span>(<span class="dt">width =</span> <span class="fl">0.01</span>, <span class="dt">height =</span> <span class="dv">0</span>),</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">  <span class="dt">point_shape =</span> <span class="st">&#39;|&#39;</span>, <span class="dt">point_size =</span> <span class="dv">1</span>, <span class="dt">point_alpha =</span> <span class="fl">0.5</span>, <span class="dt">alpha =</span> <span class="fl">0.1</span>)</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() </a>
<a class="sourceLine" id="cb4-6" data-line-number="6">p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">strip.background =</span> <span class="kw">element_blank</span>()) </a>
<a class="sourceLine" id="cb4-7" data-line-number="7">p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Major allele frequency&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Count&quot;</span>) </a>
<a class="sourceLine" id="cb4-8" data-line-number="8">p</a></code></pre></div>
<pre><code>## Picking joint bandwidth of 0.0263</code></pre>
<p><img src="oligomm-claudia_files/figure-html/unnamed-chunk-3-1.png" width="1440" /></p>
</div>
</div>
<div id="by-sample-and-genome-1" class="section level3">
<h3><span class="header-section-number">7.1.2</span> by sample and genome</h3>
<div id="lofreq-2" class="section level4">
<h4><span class="header-section-number">7.1.2.1</span> lofreq</h4>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">plotAFofSample &lt;-<span class="st"> </span><span class="cf">function</span>(<span class="dt">dat =</span> dat.lo) {</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">  p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(dat, <span class="kw">aes</span>(<span class="dt">x =</span> majorAF, <span class="dt">y =</span> genome, <span class="dt">color =</span> genome))</a>
<a class="sourceLine" id="cb6-3" data-line-number="3">  p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span>ggridges<span class="op">::</span><span class="kw">geom_density_ridges</span>(<span class="dt">jittered_points =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb6-4" data-line-number="4">    <span class="dt">position =</span> ggridges<span class="op">::</span><span class="kw">position_points_jitter</span>(<span class="dt">width =</span> <span class="fl">0.01</span>, <span class="dt">height =</span> <span class="dv">0</span>),</a>
<a class="sourceLine" id="cb6-5" data-line-number="5">    <span class="dt">point_shape =</span> <span class="st">&#39;|&#39;</span>, <span class="dt">point_size =</span> <span class="dv">2</span>, <span class="dt">point_alpha =</span> <span class="dv">1</span>, <span class="dt">alpha =</span> <span class="fl">0.1</span>)</a>
<a class="sourceLine" id="cb6-6" data-line-number="6">  p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>desc)</a>
<a class="sourceLine" id="cb6-7" data-line-number="7">  p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> omm_colors)</a>
<a class="sourceLine" id="cb6-8" data-line-number="8">  <span class="co">#p &lt;- p + facet_grid(phase ~., space =  &quot;free&quot;, scales= &quot;free&quot;)</span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9">  p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">strip.background =</span> <span class="kw">element_blank</span>()) </a>
<a class="sourceLine" id="cb6-10" data-line-number="10">  p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Major allele frequency&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Genome&quot;</span>) </a>
<a class="sourceLine" id="cb6-11" data-line-number="11">  <span class="kw">return</span>(p)</a>
<a class="sourceLine" id="cb6-12" data-line-number="12">}</a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="kw">suppressWarnings</span>(<span class="kw">plotAFofSample</span>(dat.lo))</a></code></pre></div>
<pre><code>## Picking joint bandwidth of 0.0565</code></pre>
<pre><code>## Picking joint bandwidth of 0.0601</code></pre>
<pre><code>## Picking joint bandwidth of 0.0492</code></pre>
<pre><code>## Picking joint bandwidth of 0.0549</code></pre>
<pre><code>## Picking joint bandwidth of 0.0513</code></pre>
<pre><code>## Picking joint bandwidth of 0.0366</code></pre>
<pre><code>## Picking joint bandwidth of 0.0626</code></pre>
<pre><code>## Picking joint bandwidth of 0.0767</code></pre>
<pre><code>## Picking joint bandwidth of 0.0482</code></pre>
<p><img src="oligomm-claudia_files/figure-html/unnamed-chunk-4-1.png" width="1440" /></p>
</div>
<div id="varscan-1" class="section level4">
<h4><span class="header-section-number">7.1.2.2</span> varscan</h4>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">plotAFofSample &lt;-<span class="st"> </span><span class="cf">function</span>(<span class="dt">dat =</span> dat) {</a>
<a class="sourceLine" id="cb16-2" data-line-number="2">  p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(dat, <span class="kw">aes</span>(<span class="dt">x =</span> majorAF, <span class="dt">y =</span> genome, <span class="dt">color =</span> genome))</a>
<a class="sourceLine" id="cb16-3" data-line-number="3">  p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span>ggridges<span class="op">::</span><span class="kw">geom_density_ridges</span>(<span class="dt">jittered_points =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb16-4" data-line-number="4">    <span class="dt">position =</span> ggridges<span class="op">::</span><span class="kw">position_points_jitter</span>(<span class="dt">width =</span> <span class="fl">0.01</span>, <span class="dt">height =</span> <span class="dv">0</span>),</a>
<a class="sourceLine" id="cb16-5" data-line-number="5">    <span class="dt">point_shape =</span> <span class="st">&#39;|&#39;</span>, <span class="dt">point_size =</span> <span class="dv">2</span>, <span class="dt">point_alpha =</span> <span class="dv">1</span>, <span class="dt">alpha =</span> <span class="fl">0.1</span>)</a>
<a class="sourceLine" id="cb16-6" data-line-number="6">  p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>desc)</a>
<a class="sourceLine" id="cb16-7" data-line-number="7">  p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> omm_colors)</a>
<a class="sourceLine" id="cb16-8" data-line-number="8">  <span class="co">#p &lt;- p + facet_grid(phase ~., space =  &quot;free&quot;, scales= &quot;free&quot;)</span></a>
<a class="sourceLine" id="cb16-9" data-line-number="9">  p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">strip.background =</span> <span class="kw">element_blank</span>()) </a>
<a class="sourceLine" id="cb16-10" data-line-number="10">  p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Major allele frequency&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Genome&quot;</span>) </a>
<a class="sourceLine" id="cb16-11" data-line-number="11">  <span class="kw">return</span>(p)</a>
<a class="sourceLine" id="cb16-12" data-line-number="12">}</a>
<a class="sourceLine" id="cb16-13" data-line-number="13"><span class="kw">suppressWarnings</span>(<span class="kw">plotAFofSample</span>(dat.va))</a></code></pre></div>
<pre><code>## Picking joint bandwidth of 0.028</code></pre>
<pre><code>## Picking joint bandwidth of 0.0449</code></pre>
<pre><code>## Picking joint bandwidth of 0.0393</code></pre>
<pre><code>## Picking joint bandwidth of 0.0418</code></pre>
<pre><code>## Picking joint bandwidth of 0.0378</code></pre>
<pre><code>## Picking joint bandwidth of 0.0364</code></pre>
<pre><code>## Picking joint bandwidth of 0.0488</code></pre>
<pre><code>## Picking joint bandwidth of 0.0475</code></pre>
<pre><code>## Picking joint bandwidth of 0.0356</code></pre>
<p><img src="oligomm-claudia_files/figure-html/unnamed-chunk-5-1.png" width="1440" /></p>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="Allele-frequency.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["oligomm-claudia.pdf", "oligomm-claudia.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
