<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Preface | OMM report</title>
  <meta name="description" content="Preface | OMM report" />
  <meta name="generator" content="bookdown 0.19 and GitBook 2.6.7" />

  <meta property="og:title" content="Preface | OMM report" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Preface | OMM report" />
  
  
  

<meta name="author" content="Philipp C. Muench" />


<meta name="date" content="2020-06-02" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="claudia-samples.html"/>
<script src="libs/header-attrs-2.1/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.10/datatables.js"></script>
<link href="libs/dt-core-1.10.19/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.19/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.19/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<script src="libs/plotly-binding-4.9.1/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-1.49.4/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-1.49.4/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Setup</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#how-to-generate-this-report"><i class="fa fa-check"></i>How to generate this report</a></li>
<li class="chapter" data-level="1.1" data-path="preface.html"><a href="preface.html#Workspace_setup"><i class="fa fa-check"></i><b>1.1</b> Howto generate this report</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="claudia-samples.html"><a href="claudia-samples.html"><i class="fa fa-check"></i><b>2</b> Claudia Samples</a>
<ul>
<li class="chapter" data-level="2.1" data-path="claudia-samples.html"><a href="claudia-samples.html#experiment"><i class="fa fa-check"></i><b>2.1</b> Experiment</a></li>
<li class="chapter" data-level="2.2" data-path="claudia-samples.html"><a href="claudia-samples.html#metadata"><i class="fa fa-check"></i><b>2.2</b> Metadata</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="claudia-samples.html"><a href="claudia-samples.html#crispr"><i class="fa fa-check"></i><b>2.2.1</b> CRISPR</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="resequencing.html"><a href="resequencing.html"><i class="fa fa-check"></i><b>3</b> Resequencing</a>
<ul>
<li class="chapter" data-level="3.1" data-path="resequencing.html"><a href="resequencing.html#metadata-1"><i class="fa fa-check"></i><b>3.1</b> Metadata</a></li>
<li class="chapter" data-level="3.2" data-path="resequencing.html"><a href="resequencing.html#load-in-variants"><i class="fa fa-check"></i><b>3.2</b> Load in variants</a></li>
<li class="chapter" data-level="3.3" data-path="resequencing.html"><a href="resequencing.html#kb1"><i class="fa fa-check"></i><b>3.3</b> KB1</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="resequencing.html"><a href="resequencing.html#snp"><i class="fa fa-check"></i><b>3.3.1</b> SNP</a></li>
<li class="chapter" data-level="3.3.2" data-path="resequencing.html"><a href="resequencing.html#insertion"><i class="fa fa-check"></i><b>3.3.2</b> insertion</a></li>
<li class="chapter" data-level="3.3.3" data-path="resequencing.html"><a href="resequencing.html#deletion"><i class="fa fa-check"></i><b>3.3.3</b> deletion</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="resequencing.html"><a href="resequencing.html#i46"><i class="fa fa-check"></i><b>3.4</b> I46</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="resequencing.html"><a href="resequencing.html#snp-1"><i class="fa fa-check"></i><b>3.4.1</b> SNP</a></li>
<li class="chapter" data-level="3.4.2" data-path="resequencing.html"><a href="resequencing.html#insertion-1"><i class="fa fa-check"></i><b>3.4.2</b> insertion</a></li>
<li class="chapter" data-level="3.4.3" data-path="resequencing.html"><a href="resequencing.html#deletion-1"><i class="fa fa-check"></i><b>3.4.3</b> deletion</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="resequencing.html"><a href="resequencing.html#yl58"><i class="fa fa-check"></i><b>3.5</b> YL58</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="resequencing.html"><a href="resequencing.html#snp-2"><i class="fa fa-check"></i><b>3.5.1</b> SNP</a></li>
<li class="chapter" data-level="3.5.2" data-path="resequencing.html"><a href="resequencing.html#insertion-2"><i class="fa fa-check"></i><b>3.5.2</b> insertion</a></li>
<li class="chapter" data-level="3.5.3" data-path="resequencing.html"><a href="resequencing.html#deletion-2"><i class="fa fa-check"></i><b>3.5.3</b> deletion</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="resequencing.html"><a href="resequencing.html#mutation-bias"><i class="fa fa-check"></i><b>3.6</b> mutation bias?</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="resequencing.html"><a href="resequencing.html#snp-3"><i class="fa fa-check"></i><b>3.6.1</b> SNP</a></li>
<li class="chapter" data-level="3.6.2" data-path="resequencing.html"><a href="resequencing.html#insertions"><i class="fa fa-check"></i><b>3.6.2</b> insertions</a></li>
<li class="chapter" data-level="3.6.3" data-path="resequencing.html"><a href="resequencing.html#deletions"><i class="fa fa-check"></i><b>3.6.3</b> deletions</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="resequencing.html"><a href="resequencing.html#heatmap"><i class="fa fa-check"></i><b>3.7</b> heatmap</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="resequencing.html"><a href="resequencing.html#kb1-1"><i class="fa fa-check"></i><b>3.7.1</b> KB1</a></li>
<li class="chapter" data-level="3.7.2" data-path="resequencing.html"><a href="resequencing.html#i46-1"><i class="fa fa-check"></i><b>3.7.2</b> I46</a></li>
<li class="chapter" data-level="3.7.3" data-path="resequencing.html"><a href="resequencing.html#yl58-1"><i class="fa fa-check"></i><b>3.7.3</b> YL58</a></li>
<li class="chapter" data-level="3.7.4" data-path="resequencing.html"><a href="resequencing.html#filter-present-in-all"><i class="fa fa-check"></i><b>3.7.4</b> Filter: present in all</a></li>
<li class="chapter" data-level="3.7.5" data-path="resequencing.html"><a href="resequencing.html#kb1-2"><i class="fa fa-check"></i><b>3.7.5</b> KB1</a></li>
<li class="chapter" data-level="3.7.6" data-path="resequencing.html"><a href="resequencing.html#yl58-2"><i class="fa fa-check"></i><b>3.7.6</b> YL58</a></li>
<li class="chapter" data-level="3.7.7" data-path="resequencing.html"><a href="resequencing.html#i46-2"><i class="fa fa-check"></i><b>3.7.7</b> I46</a></li>
<li class="chapter" data-level="3.7.8" data-path="resequencing.html"><a href="resequencing.html#filter-not-in-water-but-in-ab"><i class="fa fa-check"></i><b>3.7.8</b> Filter: not in water but in AB</a></li>
<li class="chapter" data-level="3.7.9" data-path="resequencing.html"><a href="resequencing.html#kb1-3"><i class="fa fa-check"></i><b>3.7.9</b> KB1</a></li>
<li class="chapter" data-level="3.7.10" data-path="resequencing.html"><a href="resequencing.html#yl58-3"><i class="fa fa-check"></i><b>3.7.10</b> YL58</a></li>
<li class="chapter" data-level="3.7.11" data-path="resequencing.html"><a href="resequencing.html#i46-3"><i class="fa fa-check"></i><b>3.7.11</b> I46</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="ab-samples.html"><a href="ab-samples.html"><i class="fa fa-check"></i><b>4</b> AB Samples</a>
<ul>
<li class="chapter" data-level="4.1" data-path="ab-samples.html"><a href="ab-samples.html#test"><i class="fa fa-check"></i><b>4.1</b> test</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">OMM report</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="preface" class="section level1 unnumbered" number="">
<h1>Preface</h1>
<p>Processing of NGS data is done using <strong>Snakemake</strong> and not covered by this report yet and will be added at a later stage. In short, preprocessing and analysis (such as <code>bwa</code> alignment and variant calling using <a href="https://csb5.github.io/lofreq/commands/">Lofreq</a> is defined using a <code>Snakefile</code> and a set of snakemake <em>rules</em> which are run on the HZI server (host) and detailed analysis (such as generation of figures and statistical tests) are done using R markdown on the <em>client</em> machine.</p>
<div id="how-to-generate-this-report" class="section level2 unnumbered" number="">
<h2>How to generate this report</h2>
<p>In <a href="https://code.visualstudio.com/">vscode</a> with bookdown addon a server can be started using <code>serve_book</code></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="preface.html#cb1-1"></a><span class="kw">setwd</span>(<span class="st">&quot;~/Cloud/ActiveProjects/OligoMM-report&quot;</span>)</span>
<span id="cb1-2"><a href="preface.html#cb1-2"></a>bookdown<span class="op">::</span><span class="kw">serve_book</span>(<span class="kw">getwd</span>())</span>
<span id="cb1-3"><a href="preface.html#cb1-3"></a><span class="co"># servr::daemon_stop(1)</span></span></code></pre></div>
<p>This will starts a browser window pointing to <a href="http://127.0.0.1:7853">http://127.0.0.1:7853</a> presenting the report written in <a href="https://rmarkdown.rstudio.com/">R Markdown</a> and will update as soon as there are changes on <code>.Rmd</code> files in the workspace.</p>
<p>To end bookdown server type <code>servr::daemon_stop(1)</code> where <code>1</code> is the ID of the server running</p>
</div>
<div id="Workspace_setup" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Howto generate this report</h2>
<p>We specify the folder where the analysis is located.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="preface.html#cb2-1"></a><span class="kw">setwd</span>(<span class="st">&quot;~/Cloud/ActiveProjects/OligoMM-report&quot;</span>)</span></code></pre></div>
<p>Functions that are used frequent are defined in <code>utils.R</code> and are mostly helper functions for data manipulation (e.g. translation of sample ids to groups). Other packages can be load via cran using the <code>install.packages()</code> function.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="preface.html#cb3-1"></a><span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>())</span>
<span id="cb3-2"><a href="preface.html#cb3-2"></a><span class="kw">source</span>(<span class="st">&quot;utils.R&quot;</span>)</span>
<span id="cb3-3"><a href="preface.html#cb3-3"></a><span class="kw">library</span>(<span class="st">&quot;vcfR&quot;</span>)</span>
<span id="cb3-4"><a href="preface.html#cb3-4"></a><span class="kw">library</span>(<span class="st">&quot;rmdformats&quot;</span>)</span>
<span id="cb3-5"><a href="preface.html#cb3-5"></a><span class="kw">library</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb3-6"><a href="preface.html#cb3-6"></a><span class="kw">library</span>(<span class="st">&quot;reshape2&quot;</span>)</span>
<span id="cb3-7"><a href="preface.html#cb3-7"></a><span class="kw">library</span>(<span class="st">&quot;ggridges&quot;</span>)</span>
<span id="cb3-8"><a href="preface.html#cb3-8"></a><span class="kw">library</span>(<span class="st">&quot;knitr&quot;</span>)</span>
<span id="cb3-9"><a href="preface.html#cb3-9"></a><span class="kw">library</span>(<span class="st">&quot;kableExtra&quot;</span>)</span>
<span id="cb3-10"><a href="preface.html#cb3-10"></a><span class="kw">library</span>(<span class="st">&quot;plotly&quot;</span>)</span>
<span id="cb3-11"><a href="preface.html#cb3-11"></a><span class="kw">library</span>(<span class="st">&quot;DT&quot;</span>)</span>
<span id="cb3-12"><a href="preface.html#cb3-12"></a><span class="kw">library</span>(<span class="st">&quot;rtracklayer&quot;</span>)</span>
<span id="cb3-13"><a href="preface.html#cb3-13"></a><span class="kw">library</span>(<span class="st">&quot;bookdown&quot;</span>)</span></code></pre></div>
<p><em>Note:</em> for <code>rtracklayer</code> you might need <em>curl</em></p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="preface.html#cb4-1"></a><span class="fu">sudo</span> apt-get install libcurl4-openssl-dev</span></code></pre></div>
<p>Data structure {#Data_structures}</p>
<dl>
<dt><code>data/raw</code></dt>
<dd>files from host machine, contains <code>coverage</code>, <em>lofreq</em> and <em>varscan</em> output
</dd>
<dt><code>data/annotation</code></dt>
<dd>annotation files for OMM12 genomes e.g. phage location (<em>pro-hunter</em>, <em>fasta</em> files and <em>Prokka</em> annotation)
</dd>
<dt><code>data/tables</code></dt>
<dd>tables in tsv format for e.g. import with Excel
</dd>
<dt><code>data/annotation/gff</code></dt>
<dd><em>Prokka</em> annotation
</dd>
<dt><code>data/rda</code></dt>
<dd>processed raw data in <em>rds</em> format generated by <code>01-load-data.rmd</code>
</dd>
<dt><code>docs</code></dt>
<dd><code>html</code> files generated via <code>bookdown::serve_book(getwd())</code>
</dd>
</dl>
<p>Build this book</p>
<p>You need to install <a href="https://pandoc.org/installing.html">Pandoc</a>.
<code>bookdown::serve_book(getwd())</code> and <code>bookdown::render_book(getwd())</code></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="claudia-samples.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
