<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Section 1 Setup | Claudia’s data</title>
  <meta name="description" content="Section 1 Setup | Claudia’s data" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="Section 1 Setup | Claudia’s data" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Section 1 Setup | Claudia’s data" />
  
  
  

<meta name="author" content="Philipp C. Muench" />


<meta name="date" content="2020-04-02" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="study-design.html"/>
<script src="libs/header-attrs-2.1.1/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.13/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<script src="libs/plotly-binding-4.9.2/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-1.52.2/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-1.52.2/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Analysis Report</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#how-to-generate-this-report"><i class="fa fa-check"></i>How to generate this report</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="Setup.html"><a href="Setup.html"><i class="fa fa-check"></i><b>1</b> Setup</a>
<ul>
<li class="chapter" data-level="1.1" data-path="Setup.html"><a href="Setup.html#Workspace_setup"><i class="fa fa-check"></i><b>1.1</b> Workspace setup</a></li>
<li class="chapter" data-level="1.2" data-path="Setup.html"><a href="Setup.html#dependencies"><i class="fa fa-check"></i><b>1.2</b> Dependencies</a></li>
<li class="chapter" data-level="1.3" data-path="Setup.html"><a href="Setup.html#Data_structures"><i class="fa fa-check"></i><b>1.3</b> Data structure</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="study-design.html"><a href="study-design.html"><i class="fa fa-check"></i><b>2</b> Study design</a>
<ul>
<li class="chapter" data-level="2.1" data-path="study-design.html"><a href="study-design.html#experiment"><i class="fa fa-check"></i><b>2.1</b> Experiment</a></li>
<li class="chapter" data-level="2.2" data-path="study-design.html"><a href="study-design.html#metadata"><i class="fa fa-check"></i><b>2.2</b> Metadata</a></li>
<li class="chapter" data-level="2.3" data-path="study-design.html"><a href="study-design.html#oligomm"><i class="fa fa-check"></i><b>2.3</b> OligoMM</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="study-design.html"><a href="study-design.html#crispr"><i class="fa fa-check"></i><b>2.3.1</b> CRISPR</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="Variant-statistics.html"><a href="Variant-statistics.html"><i class="fa fa-check"></i><b>3</b> Variant statistics</a>
<ul>
<li class="chapter" data-level="3.1" data-path="Variant-statistics.html"><a href="Variant-statistics.html#process-of-raw-data"><i class="fa fa-check"></i><b>3.1</b> Process of <em>raw</em> data</a></li>
<li class="chapter" data-level="3.2" data-path="Variant-statistics.html"><a href="Variant-statistics.html#quality-of-variants"><i class="fa fa-check"></i><b>3.2</b> Quality of variants</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="Variant-statistics.html"><a href="Variant-statistics.html#all-samples"><i class="fa fa-check"></i><b>3.2.1</b> all samples</a></li>
<li class="chapter" data-level="3.2.2" data-path="Variant-statistics.html"><a href="Variant-statistics.html#by-mouse"><i class="fa fa-check"></i><b>3.2.2</b> by mouse</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="Variant-statistics.html"><a href="Variant-statistics.html#number-of-variants"><i class="fa fa-check"></i><b>3.3</b> Number of variants</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="Variant-statistics.html"><a href="Variant-statistics.html#by-sample"><i class="fa fa-check"></i><b>3.3.1</b> by sample</a></li>
<li class="chapter" data-level="3.3.2" data-path="Variant-statistics.html"><a href="Variant-statistics.html#by-mouse-and-genome"><i class="fa fa-check"></i><b>3.3.2</b> by mouse and genome</a></li>
<li class="chapter" data-level="3.3.3" data-path="Variant-statistics.html"><a href="Variant-statistics.html#by-extendet-metadata"><i class="fa fa-check"></i><b>3.3.3</b> by extendet metadata</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="Variant-statistics.html"><a href="Variant-statistics.html#orfs-with-variants"><i class="fa fa-check"></i><b>3.4</b> ORFs with variants</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="Variant-statistics.html"><a href="Variant-statistics.html#data"><i class="fa fa-check"></i><b>3.4.1</b> Data</a></li>
<li class="chapter" data-level="3.4.2" data-path="Variant-statistics.html"><a href="Variant-statistics.html#statistics"><i class="fa fa-check"></i><b>3.4.2</b> Statistics</a></li>
<li class="chapter" data-level="3.4.3" data-path="Variant-statistics.html"><a href="Variant-statistics.html#minor-to-major-transistion"><i class="fa fa-check"></i><b>3.4.3</b> Minor to major transistion</a></li>
<li class="chapter" data-level="3.4.4" data-path="Variant-statistics.html"><a href="Variant-statistics.html#gene-centric-analysis"><i class="fa fa-check"></i><b>3.4.4</b> Gene-centric analysis</a></li>
<li class="chapter" data-level="3.4.5" data-path="Variant-statistics.html"><a href="Variant-statistics.html#by-variant-static"><i class="fa fa-check"></i><b>3.4.5</b> by variant (static)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="variant-effects.html"><a href="variant-effects.html"><i class="fa fa-check"></i><b>4</b> Variant effects</a>
<ul>
<li class="chapter" data-level="4.1" data-path="variant-effects.html"><a href="variant-effects.html#snpeff-run"><i class="fa fa-check"></i><b>4.1</b> snpEff run</a></li>
<li class="chapter" data-level="4.2" data-path="variant-effects.html"><a href="variant-effects.html#number-of-effects-by-type"><i class="fa fa-check"></i><b>4.2</b> number of effects by type</a></li>
<li class="chapter" data-level="4.3" data-path="variant-effects.html"><a href="variant-effects.html#number-of-effects-by-type-by-genome-vartype"><i class="fa fa-check"></i><b>4.3</b> number of effects by type by genome (vartype)</a></li>
<li class="chapter" data-level="4.4" data-path="variant-effects.html"><a href="variant-effects.html#number-of-effects-by-type-by-genome-effect"><i class="fa fa-check"></i><b>4.4</b> number of effects by type by genome (effect)</a></li>
<li class="chapter" data-level="4.5" data-path="variant-effects.html"><a href="variant-effects.html#by-genome-and-mouse"><i class="fa fa-check"></i><b>4.5</b> by genome and mouse</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Allele-frequency.html"><a href="Allele-frequency.html"><i class="fa fa-check"></i><b>5</b> Allele frequency</a>
<ul>
<li class="chapter" data-level="5.1" data-path="Allele-frequency.html"><a href="Allele-frequency.html#major-allele-frequency"><i class="fa fa-check"></i><b>5.1</b> major allele frequency</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="Allele-frequency.html"><a href="Allele-frequency.html#all-samples-and-all-genomes"><i class="fa fa-check"></i><b>5.1.1</b> all samples and all genomes</a></li>
<li class="chapter" data-level="5.1.2" data-path="Allele-frequency.html"><a href="Allele-frequency.html#by-sample-and-genome"><i class="fa fa-check"></i><b>5.1.2</b> by sample and genome</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://pmuench.com" target="blank">Philipp C. Muench</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Claudia’s data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="Setup" class="section level1" number="1">
<h1><span class="header-section-number">Section 1</span> Setup</h1>
<div id="Workspace_setup" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Workspace setup</h2>
<p>We specify the folder where the analysis is located.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="Setup.html#cb2-1"></a><span class="kw">setwd</span>(<span class="st">&quot;~/projects/oligomm-claudia/&quot;</span>)</span></code></pre></div>
</div>
<div id="dependencies" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Dependencies</h2>
<p>Functions that are used frequent are defined in <code>utils.R</code> and are mostly helper functions for data manipulation (e.g. translation of sample ids to groups). Other packages can be load via cran using the <code>install.packages()</code> function.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="Setup.html#cb3-1"></a><span class="kw">rm</span>(<span class="dt">list =</span> <span class="kw">ls</span>())</span>
<span id="cb3-2"><a href="Setup.html#cb3-2"></a><span class="kw">source</span>(<span class="st">&quot;utils.R&quot;</span>)</span>
<span id="cb3-3"><a href="Setup.html#cb3-3"></a><span class="kw">library</span>(<span class="st">&quot;vcfR&quot;</span>)</span>
<span id="cb3-4"><a href="Setup.html#cb3-4"></a><span class="kw">library</span>(<span class="st">&quot;rmdformats&quot;</span>)</span>
<span id="cb3-5"><a href="Setup.html#cb3-5"></a><span class="kw">library</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb3-6"><a href="Setup.html#cb3-6"></a><span class="kw">library</span>(<span class="st">&quot;reshape2&quot;</span>)</span>
<span id="cb3-7"><a href="Setup.html#cb3-7"></a><span class="kw">library</span>(<span class="st">&quot;ggridges&quot;</span>)</span>
<span id="cb3-8"><a href="Setup.html#cb3-8"></a><span class="kw">library</span>(<span class="st">&quot;knitr&quot;</span>)</span>
<span id="cb3-9"><a href="Setup.html#cb3-9"></a><span class="kw">library</span>(<span class="st">&quot;kableExtra&quot;</span>)</span>
<span id="cb3-10"><a href="Setup.html#cb3-10"></a><span class="kw">library</span>(<span class="st">&quot;plotly&quot;</span>)</span>
<span id="cb3-11"><a href="Setup.html#cb3-11"></a><span class="kw">library</span>(<span class="st">&quot;DT&quot;</span>)</span>
<span id="cb3-12"><a href="Setup.html#cb3-12"></a><span class="kw">library</span>(<span class="st">&quot;rtracklayer&quot;</span>)</span></code></pre></div>
<p>for <code>rtracklayer</code> you might need curl</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="Setup.html#cb4-1"></a><span class="fu">sudo</span> apt-get install libcurl4-openssl-dev</span></code></pre></div>
</div>
<div id="Data_structures" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Data structure</h2>
<dl>
<dt><code>data/raw</code></dt>
<dd>files from host machine, contains <code>coverage</code> and <code>lofreq</code> data
</dd>
<dt><code>data/annotation</code></dt>
<dd>annotation files for OMM12 genomes e.g. phage location (<em>pro-hunter</em>, <em>fasta</em> files and <em>Prokka</em> annotation)
</dd>
<dt><code>data/tables</code></dt>
<dd>tables in tsv format for e.g. import with Excel
</dd>
<dt><code>data/annotation/gff</code></dt>
<dd><em>Prokka</em> annotation
</dd>
<dt><code>data/rda</code></dt>
<dd>processed raw data in <em>rds</em> format generated by <code>01-load-data.rmd</code>
</dd>
<dt><code>docs</code></dt>
<dd><code>html</code> files generated via <code>bookdown::serve_book(getwd())</code>
</dd>
</dl>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="study-design.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["oligomm-claudia.pdf", "oligomm-claudia.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
