<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Section 2 Variant statistics | Claudia’s data</title>
  <meta name="description" content="Section 2 Variant statistics | Claudia’s data" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="Section 2 Variant statistics | Claudia’s data" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Section 2 Variant statistics | Claudia’s data" />
  
  
  

<meta name="author" content="Philipp C. Muench" />


<meta name="date" content="2020-03-26" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Setup.html"/>
<link rel="next" href="Allele-frequency.html"/>
<script src="libs/header-attrs-2.1.1/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.12/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<script src="libs/plotly-binding-4.9.2/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-1.52.2/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-1.52.2/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Analysis Report</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#how-to-generate-this-report"><i class="fa fa-check"></i>How to generate this report</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="Setup.html"><a href="Setup.html"><i class="fa fa-check"></i><b>1</b> Setup</a>
<ul>
<li class="chapter" data-level="1.1" data-path="Setup.html"><a href="Setup.html#dependencies"><i class="fa fa-check"></i><b>1.1</b> Dependencies</a></li>
<li class="chapter" data-level="1.2" data-path="Setup.html"><a href="Setup.html#Workspace_setup"><i class="fa fa-check"></i><b>1.2</b> Workspace setup</a></li>
<li class="chapter" data-level="1.3" data-path="Setup.html"><a href="Setup.html#Data_structures"><i class="fa fa-check"></i><b>1.3</b> Data structure</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="Variant-statistics.html"><a href="Variant-statistics.html"><i class="fa fa-check"></i><b>2</b> Variant statistics</a>
<ul>
<li class="chapter" data-level="2.1" data-path="Variant-statistics.html"><a href="Variant-statistics.html#process-of-raw-data"><i class="fa fa-check"></i><b>2.1</b> Process of <em>raw</em> data</a></li>
<li class="chapter" data-level="2.2" data-path="Variant-statistics.html"><a href="Variant-statistics.html#number-of-variants"><i class="fa fa-check"></i><b>2.2</b> Number of variants</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="Allele-frequency.html"><a href="Allele-frequency.html"><i class="fa fa-check"></i><b>3</b> Allele frequency</a>
<ul>
<li class="chapter" data-level="3.1" data-path="Allele-frequency.html"><a href="Allele-frequency.html#major-allele-frequency"><i class="fa fa-check"></i><b>3.1</b> major allele frequency</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://pmuench.com" target="blank">Philipp C. Muench</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Claudia’s data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="Variant_statistics" class="section level1" number="2">
<h1><span class="header-section-number">Section 2</span> Variant statistics</h1>
<div id="process-of-raw-data" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Process of <em>raw</em> data</h2>
<p>Since the raw output of the <strong>Snakemake</strong> pipeline produces one <code>vcf</code> file per sample (including all contigs of the OligoMM reference genomes) we now produce a single <code>rds</code> file that contains the merged variant information of the whole study.</p>
<p>We define a function that iterates over a set of <code>vcf</code> files (<code>vcf.files</code>)</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="Variant-statistics.html#cb4-1"></a><span class="co">#&#39; Processes a list of vcf files</span></span>
<span id="cb4-2"><a href="Variant-statistics.html#cb4-2"></a><span class="co">#&#39; </span></span>
<span id="cb4-3"><a href="Variant-statistics.html#cb4-3"></a><span class="co">#&#39; @param vcf.files a list of file path of files in vcf format</span></span>
<span id="cb4-4"><a href="Variant-statistics.html#cb4-4"></a><span class="co">#&#39; @param contig_mapping mapping file of vcf CHR to OMM reference genomes</span></span>
<span id="cb4-5"><a href="Variant-statistics.html#cb4-5"></a><span class="co">#&#39; @return a dataframe holding the merged annotated variant information</span></span>
<span id="cb4-6"><a href="Variant-statistics.html#cb4-6"></a></span>
<span id="cb4-7"><a href="Variant-statistics.html#cb4-7"></a>vcfToDataframe &lt;-<span class="st"> </span><span class="cf">function</span>(vcf.files, <span class="dt">contig_mapping =</span> <span class="kw">read.csv2</span>(<span class="st">&quot;data/contig_genome_mapping.csv&quot;</span>, <span class="dt">sep =</span><span class="st">&quot;;&quot;</span>, <span class="dt">header=</span>T, <span class="dt">stringsAsFactors =</span> F) ) {</span>
<span id="cb4-8"><a href="Variant-statistics.html#cb4-8"></a>    <span class="kw">require</span>(vcfR)</span>
<span id="cb4-9"><a href="Variant-statistics.html#cb4-9"></a>    res &lt;-<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb4-10"><a href="Variant-statistics.html#cb4-10"></a>    <span class="cf">for</span> (file <span class="cf">in</span> vcf.files) {</span>
<span id="cb4-11"><a href="Variant-statistics.html#cb4-11"></a>        vcf.content &lt;-<span class="st"> </span>vcfR<span class="op">::</span><span class="kw">read.vcfR</span>(file, <span class="dt">verbose =</span> <span class="ot">FALSE</span> )</span>
<span id="cb4-12"><a href="Variant-statistics.html#cb4-12"></a>        vcf.fix &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(vcf.content<span class="op">@</span>fix) <span class="co"># contains chr, position and substitution informations</span></span>
<span id="cb4-13"><a href="Variant-statistics.html#cb4-13"></a>        vcf.info &lt;-<span class="st"> </span>vcfR<span class="op">::</span><span class="kw">INFO2df</span>(vcf.content) <span class="co"># get INFO field, contains DP, AF informations</span></span>
<span id="cb4-14"><a href="Variant-statistics.html#cb4-14"></a>    <span class="co">#  vcf.ann &lt;- data.frame(do.call(&#39;rbind&#39;, strsplit(as.character(vcf.info$ANN),&#39;|&#39;,fixed = TRUE))) # split ANN field, contains information if files are processed by snpEff</span></span>
<span id="cb4-15"><a href="Variant-statistics.html#cb4-15"></a>        <span class="cf">if</span>(<span class="kw">nrow</span>(vcf.fix) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>) {</span>
<span id="cb4-16"><a href="Variant-statistics.html#cb4-16"></a>            dat &lt;-<span class="st">  </span><span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(vcf.fix[,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">5</span>, <span class="dv">6</span>)], vcf.info[,<span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>)]))</span>
<span id="cb4-17"><a href="Variant-statistics.html#cb4-17"></a>            dat<span class="op">$</span>majorAF &lt;-<span class="st"> </span><span class="kw">sapply</span>(dat<span class="op">$</span>AF, minorAfToMajorAf) <span class="co"># transforms e.g. AF of 0.1 to 0.9, 0.9 stays 0.9 and 0.5 stays 0.5</span></span>
<span id="cb4-18"><a href="Variant-statistics.html#cb4-18"></a>            dat<span class="op">$</span>genome &lt;-<span class="st"> </span>contig_mapping[<span class="kw">match</span>(dat<span class="op">$</span>CHROM, contig_mapping<span class="op">$</span>contig),]<span class="op">$</span>genome <span class="co"># map chr information to genome name e.g. NHMU01000001.1 -&gt; i48</span></span>
<span id="cb4-19"><a href="Variant-statistics.html#cb4-19"></a>            dat<span class="op">$</span>genome_hr &lt;-<span class="st"> </span><span class="kw">translateGenomeIdToFullName</span>(dat<span class="op">$</span>genome)</span>
<span id="cb4-20"><a href="Variant-statistics.html#cb4-20"></a>            dat<span class="op">$</span>day &lt;-<span class="st"> </span><span class="kw">as.integer</span>(<span class="kw">substr</span>(<span class="kw">basename</span>(file), <span class="dv">6</span>, <span class="dv">7</span>)) <span class="co"># get day information from file name</span></span>
<span id="cb4-21"><a href="Variant-statistics.html#cb4-21"></a>            dat<span class="op">$</span>mouse.id &lt;-<span class="st"> </span><span class="kw">substr</span>(<span class="kw">basename</span>(file), <span class="dv">1</span>, <span class="dv">4</span>) <span class="co"># get mouse ID from file name, e.g. &quot;1681&quot;</span></span>
<span id="cb4-22"><a href="Variant-statistics.html#cb4-22"></a>            dat<span class="op">$</span>phase &lt;-<span class="st"> </span><span class="kw">translateMouseIdToClaudiaGroup</span>(<span class="kw">as.character</span>(<span class="kw">as.matrix</span>(dat<span class="op">$</span>mouse.id)))</span>
<span id="cb4-23"><a href="Variant-statistics.html#cb4-23"></a>            dat<span class="op">$</span>phase_num &lt;-<span class="st"> </span><span class="kw">binDaysByPhaseGroup</span>(dat<span class="op">$</span>day)</span>
<span id="cb4-24"><a href="Variant-statistics.html#cb4-24"></a>            dat<span class="op">$</span>dp &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">as.matrix</span>(vcf.info<span class="op">$</span>DP))</span>
<span id="cb4-25"><a href="Variant-statistics.html#cb4-25"></a>            res[[<span class="kw">basename</span>(file)]] &lt;-<span class="st"> </span>dat <span class="co"># add vcf df to list</span></span>
<span id="cb4-26"><a href="Variant-statistics.html#cb4-26"></a>        }</span>
<span id="cb4-27"><a href="Variant-statistics.html#cb4-27"></a>    }</span>
<span id="cb4-28"><a href="Variant-statistics.html#cb4-28"></a>    df &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">do.call</span>(rbind, res)) <span class="co"># merge list to df</span></span>
<span id="cb4-29"><a href="Variant-statistics.html#cb4-29"></a>    <span class="kw">return</span>(df)</span>
<span id="cb4-30"><a href="Variant-statistics.html#cb4-30"></a>}</span></code></pre></div>
<p>We apply this function on all 24 files that live in <code>data/raw/lofreq/*.vcf</code></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="Variant-statistics.html#cb5-1"></a>contig_mapping &lt;-<span class="st"> </span><span class="kw">read.csv2</span>(<span class="st">&quot;data/contig_genome_mapping.csv&quot;</span>, <span class="dt">sep =</span><span class="st">&quot;;&quot;</span>, <span class="dt">header=</span>T, <span class="dt">stringsAsFactors =</span> F) <span class="co"># this file contains contig names of the 12 OligoMM genomes</span></span>
<span id="cb5-2"><a href="Variant-statistics.html#cb5-2"></a>vcf.files &lt;-<span class="st"> </span><span class="kw">Sys.glob</span>(<span class="st">&quot;data/raw/lofreq/*.vcf&quot;</span>)</span>
<span id="cb5-3"><a href="Variant-statistics.html#cb5-3"></a>dat &lt;-<span class="st"> </span><span class="kw">suppressWarnings</span>(<span class="kw">vcfToDataframe</span>(vcf.files, contig_mapping))</span>
<span id="cb5-4"><a href="Variant-statistics.html#cb5-4"></a><span class="kw">saveRDS</span>(dat, <span class="dt">file =</span> <span class="st">&quot;data/rds/variants_unannotated.rds&quot;</span>)</span></code></pre></div>
<p>The dataset is written to <code>data/rds/variants_unannotated.rds</code>) and will be used in later sections.</p>
</div>
<div id="number-of-variants" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Number of variants</h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="Variant-statistics.html#cb6-1"></a>dat &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;data/rds/variants_unannotated.rds&quot;</span>)</span></code></pre></div>
<p>This set contains 1796 variants found in 12 genomes. The detailed breakdow of variants per genome is:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="Variant-statistics.html#cb7-1"></a>dat<span class="op">$</span>dummy &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb7-2"><a href="Variant-statistics.html#cb7-2"></a>dat.by.sample.genome &lt;-<span class="st"> </span><span class="kw">aggregate</span>(dummy <span class="op">~</span><span class="st"> </span>mouse.id <span class="op">+</span><span class="st"> </span>genome, dat, sum)</span>
<span id="cb7-3"><a href="Variant-statistics.html#cb7-3"></a><span class="kw">colnames</span>(dat.by.sample.genome) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;genome&quot;</span>, <span class="st">&quot;number_variants&quot;</span>)</span>
<span id="cb7-4"><a href="Variant-statistics.html#cb7-4"></a>DT<span class="op">::</span><span class="kw">datatable</span>(dat.by.sample.genome)</span></code></pre></div>
<div id="htmlwidget-2618ab1f19df957d2bb2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-2618ab1f19df957d2bb2">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157"],["1607","1660","1664","1789","1880","I_mi","1423","1425","1607","1612","1660","1664","1750","1753","1779","1783","1789","1801","1807","1814","1815","1877","1880","1881","1882","1883","1884","1885","I_cc","I_mi","1425","1753","1789","I_mi","I_mi","I_mi","1789","1801","I_mi","1423","1425","1607","1612","1660","1664","1750","1753","1779","1783","1789","1801","1807","1814","1815","1880","1881","1882","1883","1884","1885","I_cc","I_mi","1425","1612","1660","1664","1753","1779","1783","1789","1801","1807","1814","1815","1880","1881","1882","1883","1884","1885","I_cc","I_mi","1423","1425","1607","1612","1660","1664","1750","1753","1779","1783","1789","1801","1807","1814","1815","1877","1880","1881","1882","1883","1884","1885","I_cc","I_mi","1423","1425","1750","1779","1783","1814","1815","1880","1881","1882","1883","1884","1885","I_cc","1607","1612","1660","1750","1753","1779","1783","1789","1801","1807","1814","1880","1881","1882","1883","1884","1885","I_cc","I_mi","1423","1425","1607","1612","1660","1664","1750","1753","1779","1789","1801","1807","1877","1880","1882","1883","1885","I_mi"],["i46","i46","i46","i46","i46","i46","i48","i48","i48","i48","i48","i48","i48","i48","i48","i48","i48","i48","i48","i48","i48","i48","i48","i48","i48","i48","i48","i48","i48","i48","i49","i49","i49","i49","kb1","kb18","yl2","yl2","yl2","yl27","yl27","yl27","yl27","yl27","yl27","yl27","yl27","yl27","yl27","yl27","yl27","yl27","yl27","yl27","yl27","yl27","yl27","yl27","yl27","yl27","yl27","yl27","yl31","yl31","yl31","yl31","yl31","yl31","yl31","yl31","yl31","yl31","yl31","yl31","yl31","yl31","yl31","yl31","yl31","yl31","yl31","yl31","yl32","yl32","yl32","yl32","yl32","yl32","yl32","yl32","yl32","yl32","yl32","yl32","yl32","yl32","yl32","yl32","yl32","yl32","yl32","yl32","yl32","yl32","yl32","yl32","yl44","yl44","yl44","yl44","yl44","yl44","yl44","yl44","yl44","yl44","yl44","yl44","yl44","yl44","yl45","yl45","yl45","yl45","yl45","yl45","yl45","yl45","yl45","yl45","yl45","yl45","yl45","yl45","yl45","yl45","yl45","yl45","yl45","yl58","yl58","yl58","yl58","yl58","yl58","yl58","yl58","yl58","yl58","yl58","yl58","yl58","yl58","yl58","yl58","yl58","yl58"],[3,2,1,1,13,22,20,23,18,18,18,21,21,23,16,20,21,21,20,40,21,24,39,37,32,39,33,37,18,16,13,1,1,18,3,1,3,3,3,6,5,5,3,5,6,4,2,7,6,6,6,6,4,3,9,5,2,8,8,8,5,4,2,1,10,6,12,8,1,14,10,6,4,1,8,12,11,8,4,9,1,7,35,21,34,28,25,29,25,28,14,32,21,30,25,31,28,32,30,26,21,23,27,23,34,21,1,3,1,2,1,1,2,3,2,7,5,7,6,1,2,1,4,10,9,5,2,5,12,5,1,9,5,3,14,5,14,4,13,7,7,3,1,3,3,2,2,2,3,3,1,3,1,1,1,1,2]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>sample<\/th>\n      <th>genome<\/th>\n      <th>number_variants<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="Variant-statistics.html#cb8-1"></a>p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(dat.by.sample.genome, <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">reorder</span>(sample, number_variants), <span class="dt">y =</span> number_variants, <span class="dt">fill =</span> genome))</span>
<span id="cb8-2"><a href="Variant-statistics.html#cb8-2"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">coord_flip</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>()</span>
<span id="cb8-3"><a href="Variant-statistics.html#cb8-3"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;number of variants&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;sample&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> omm_colors)</span>
<span id="cb8-4"><a href="Variant-statistics.html#cb8-4"></a>plotly<span class="op">::</span><span class="kw">ggplotly</span>(p)</span></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-7"></span>
<div id="htmlwidget-066f7e0458cb13f15981" style="width:672px;height:480px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-066f7e0458cb13f15981">{"x":{"data":[{"orientation":"h","width":[0.899999999999999,0.9,0.899999999999999,0.9,0.899999999999999,0.9],"base":[62,65,65,74,99,88],"x":[3,2,1,1,13,22],"y":[11,4,12,2,19,5],"text":["reorder(sample, number_variants): 1607<br />number_variants:  3<br />genome: i46","reorder(sample, number_variants): 1660<br />number_variants:  2<br />genome: i46","reorder(sample, number_variants): 1664<br />number_variants:  1<br />genome: i46","reorder(sample, number_variants): 1789<br />number_variants:  1<br />genome: i46","reorder(sample, number_variants): 1880<br />number_variants: 13<br />genome: i46","reorder(sample, number_variants): I_mi<br />number_variants: 22<br />genome: i46"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(35,139,69,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"i46","legendgroup":"i46","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"h","width":[0.899999999999999,0.899999999999999,0.899999999999999,0.9,0.9,0.899999999999999,0.9,0.899999999999999,0.9,0.9,0.9,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.9],"base":[49,51,44,34,47,44,42,54,38,42,53,64,43,41,34,35,60,50,45,59,51,61,45,72],"x":[20,23,18,18,18,21,21,23,16,20,21,21,20,40,21,24,39,37,32,39,33,37,18,16],"y":[18,10,11,3,4,12,7,13,1,6,2,16,8,17,14,24,19,23,15,20,21,22,9,5],"text":["reorder(sample, number_variants): 1423<br />number_variants: 20<br />genome: i48","reorder(sample, number_variants): 1425<br />number_variants: 23<br />genome: i48","reorder(sample, number_variants): 1607<br />number_variants: 18<br />genome: i48","reorder(sample, number_variants): 1612<br />number_variants: 18<br />genome: i48","reorder(sample, number_variants): 1660<br />number_variants: 18<br />genome: i48","reorder(sample, number_variants): 1664<br />number_variants: 21<br />genome: i48","reorder(sample, number_variants): 1750<br />number_variants: 21<br />genome: i48","reorder(sample, number_variants): 1753<br />number_variants: 23<br />genome: i48","reorder(sample, number_variants): 1779<br />number_variants: 16<br />genome: i48","reorder(sample, number_variants): 1783<br />number_variants: 20<br />genome: i48","reorder(sample, number_variants): 1789<br />number_variants: 21<br />genome: i48","reorder(sample, number_variants): 1801<br />number_variants: 21<br />genome: i48","reorder(sample, number_variants): 1807<br />number_variants: 20<br />genome: i48","reorder(sample, number_variants): 1814<br />number_variants: 40<br />genome: i48","reorder(sample, number_variants): 1815<br />number_variants: 21<br />genome: i48","reorder(sample, number_variants): 1877<br />number_variants: 24<br />genome: i48","reorder(sample, number_variants): 1880<br />number_variants: 39<br />genome: i48","reorder(sample, number_variants): 1881<br />number_variants: 37<br />genome: i48","reorder(sample, number_variants): 1882<br />number_variants: 32<br />genome: i48","reorder(sample, number_variants): 1883<br />number_variants: 39<br />genome: i48","reorder(sample, number_variants): 1884<br />number_variants: 33<br />genome: i48","reorder(sample, number_variants): 1885<br />number_variants: 37<br />genome: i48","reorder(sample, number_variants): I_cc<br />number_variants: 18<br />genome: i48","reorder(sample, number_variants): I_mi<br />number_variants: 16<br />genome: i48"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(158,154,200,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"i48","legendgroup":"i48","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"h","width":[0.899999999999999,0.899999999999999,0.9,0.9],"base":[38,53,52,54],"x":[13,1,1,18],"y":[10,13,2,5],"text":["reorder(sample, number_variants): 1425<br />number_variants: 13<br />genome: i49","reorder(sample, number_variants): 1753<br />number_variants:  1<br />genome: i49","reorder(sample, number_variants): 1789<br />number_variants:  1<br />genome: i49","reorder(sample, number_variants): I_mi<br />number_variants: 18<br />genome: i49"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(255,255,178,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"i49","legendgroup":"i49","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"h","width":0.9,"base":51,"x":[3],"y":[5],"text":"reorder(sample, number_variants): I_mi<br />number_variants:  3<br />genome: kb1","type":"bar","marker":{"autocolorscale":false,"color":"rgba(150,150,150,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"kb1","legendgroup":"kb1","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"h","width":0.9,"base":50,"x":[1],"y":[5],"text":"reorder(sample, number_variants): I_mi<br />number_variants:  1<br />genome: kb18","type":"bar","marker":{"autocolorscale":false,"color":"rgba(150,150,150,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"kb18","legendgroup":"kb18","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"h","width":[0.9,0.899999999999999,0.9],"base":[49,61,47],"x":[3,3,3],"y":[2,16,5],"text":["reorder(sample, number_variants): 1789<br />number_variants:  3<br />genome: yl2","reorder(sample, number_variants): 1801<br />number_variants:  3<br />genome: yl2","reorder(sample, number_variants): I_mi<br />number_variants:  3<br />genome: yl2"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(106,81,163,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"yl2","legendgroup":"yl2","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"h","width":[0.899999999999999,0.899999999999999,0.899999999999999,0.9,0.9,0.899999999999999,0.9,0.899999999999999,0.9,0.9,0.9,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.9],"base":[43,33,39,31,42,38,38,51,31,36,43,55,37,37,31,51,45,43,51,43,53,40,43],"x":[6,5,5,3,5,6,4,2,7,6,6,6,6,4,3,9,5,2,8,8,8,5,4],"y":[18,10,11,3,4,12,7,13,1,6,2,16,8,17,14,19,23,15,20,21,22,9,5],"text":["reorder(sample, number_variants): 1423<br />number_variants:  6<br />genome: yl27","reorder(sample, number_variants): 1425<br />number_variants:  5<br />genome: yl27","reorder(sample, number_variants): 1607<br />number_variants:  5<br />genome: yl27","reorder(sample, number_variants): 1612<br />number_variants:  3<br />genome: yl27","reorder(sample, number_variants): 1660<br />number_variants:  5<br />genome: yl27","reorder(sample, number_variants): 1664<br />number_variants:  6<br />genome: yl27","reorder(sample, number_variants): 1750<br />number_variants:  4<br />genome: yl27","reorder(sample, number_variants): 1753<br />number_variants:  2<br />genome: yl27","reorder(sample, number_variants): 1779<br />number_variants:  7<br />genome: yl27","reorder(sample, number_variants): 1783<br />number_variants:  6<br />genome: yl27","reorder(sample, number_variants): 1789<br />number_variants:  6<br />genome: yl27","reorder(sample, number_variants): 1801<br />number_variants:  6<br />genome: yl27","reorder(sample, number_variants): 1807<br />number_variants:  6<br />genome: yl27","reorder(sample, number_variants): 1814<br />number_variants:  4<br />genome: yl27","reorder(sample, number_variants): 1815<br />number_variants:  3<br />genome: yl27","reorder(sample, number_variants): 1880<br />number_variants:  9<br />genome: yl27","reorder(sample, number_variants): 1881<br />number_variants:  5<br />genome: yl27","reorder(sample, number_variants): 1882<br />number_variants:  2<br />genome: yl27","reorder(sample, number_variants): 1883<br />number_variants:  8<br />genome: yl27","reorder(sample, number_variants): 1884<br />number_variants:  8<br />genome: yl27","reorder(sample, number_variants): 1885<br />number_variants:  8<br />genome: yl27","reorder(sample, number_variants): I_cc<br />number_variants:  5<br />genome: yl27","reorder(sample, number_variants): I_mi<br />number_variants:  4<br />genome: yl27"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(82,82,82,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"yl27","legendgroup":"yl27","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"h","width":[0.899999999999999,0.9,0.9,0.899999999999999,0.899999999999999,0.9,0.9,0.9,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.9],"base":[31,30,32,32,39,23,35,29,45,31,33,30,43,33,32,43,39,44,39,36],"x":[2,1,10,6,12,8,1,14,10,6,4,1,8,12,11,8,4,9,1,7],"y":[10,3,4,12,13,1,6,2,16,8,17,14,19,23,15,20,21,22,9,5],"text":["reorder(sample, number_variants): 1425<br />number_variants:  2<br />genome: yl31","reorder(sample, number_variants): 1612<br />number_variants:  1<br />genome: yl31","reorder(sample, number_variants): 1660<br />number_variants: 10<br />genome: yl31","reorder(sample, number_variants): 1664<br />number_variants:  6<br />genome: yl31","reorder(sample, number_variants): 1753<br />number_variants: 12<br />genome: yl31","reorder(sample, number_variants): 1779<br />number_variants:  8<br />genome: yl31","reorder(sample, number_variants): 1783<br />number_variants:  1<br />genome: yl31","reorder(sample, number_variants): 1789<br />number_variants: 14<br />genome: yl31","reorder(sample, number_variants): 1801<br />number_variants: 10<br />genome: yl31","reorder(sample, number_variants): 1807<br />number_variants:  6<br />genome: yl31","reorder(sample, number_variants): 1814<br />number_variants:  4<br />genome: yl31","reorder(sample, number_variants): 1815<br />number_variants:  1<br />genome: yl31","reorder(sample, number_variants): 1880<br />number_variants:  8<br />genome: yl31","reorder(sample, number_variants): 1881<br />number_variants: 12<br />genome: yl31","reorder(sample, number_variants): 1882<br />number_variants: 11<br />genome: yl31","reorder(sample, number_variants): 1883<br />number_variants:  8<br />genome: yl31","reorder(sample, number_variants): 1884<br />number_variants:  4<br />genome: yl31","reorder(sample, number_variants): 1885<br />number_variants:  9<br />genome: yl31","reorder(sample, number_variants): I_cc<br />number_variants:  1<br />genome: yl31","reorder(sample, number_variants): I_mi<br />number_variants:  7<br />genome: yl31"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(136,65,157,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"yl31","legendgroup":"yl31","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"h","width":[0.899999999999999,0.899999999999999,0.899999999999999,0.9,0.9,0.899999999999999,0.9,0.899999999999999,0.9,0.9,0.9,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.9],"base":[8,10,5,2,7,3,13,11,9,3,8,15,6,2,2,3,13,7,11,20,12,21,5,15],"x":[35,21,34,28,25,29,25,28,14,32,21,30,25,31,28,32,30,26,21,23,27,23,34,21],"y":[18,10,11,3,4,12,7,13,1,6,2,16,8,17,14,24,19,23,15,20,21,22,9,5],"text":["reorder(sample, number_variants): 1423<br />number_variants: 35<br />genome: yl32","reorder(sample, number_variants): 1425<br />number_variants: 21<br />genome: yl32","reorder(sample, number_variants): 1607<br />number_variants: 34<br />genome: yl32","reorder(sample, number_variants): 1612<br />number_variants: 28<br />genome: yl32","reorder(sample, number_variants): 1660<br />number_variants: 25<br />genome: yl32","reorder(sample, number_variants): 1664<br />number_variants: 29<br />genome: yl32","reorder(sample, number_variants): 1750<br />number_variants: 25<br />genome: yl32","reorder(sample, number_variants): 1753<br />number_variants: 28<br />genome: yl32","reorder(sample, number_variants): 1779<br />number_variants: 14<br />genome: yl32","reorder(sample, number_variants): 1783<br />number_variants: 32<br />genome: yl32","reorder(sample, number_variants): 1789<br />number_variants: 21<br />genome: yl32","reorder(sample, number_variants): 1801<br />number_variants: 30<br />genome: yl32","reorder(sample, number_variants): 1807<br />number_variants: 25<br />genome: yl32","reorder(sample, number_variants): 1814<br />number_variants: 31<br />genome: yl32","reorder(sample, number_variants): 1815<br />number_variants: 28<br />genome: yl32","reorder(sample, number_variants): 1877<br />number_variants: 32<br />genome: yl32","reorder(sample, number_variants): 1880<br />number_variants: 30<br />genome: yl32","reorder(sample, number_variants): 1881<br />number_variants: 26<br />genome: yl32","reorder(sample, number_variants): 1882<br />number_variants: 21<br />genome: yl32","reorder(sample, number_variants): 1883<br />number_variants: 23<br />genome: yl32","reorder(sample, number_variants): 1884<br />number_variants: 27<br />genome: yl32","reorder(sample, number_variants): 1885<br />number_variants: 23<br />genome: yl32","reorder(sample, number_variants): I_cc<br />number_variants: 34<br />genome: yl32","reorder(sample, number_variants): I_mi<br />number_variants: 21<br />genome: yl32"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(116,196,118,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"yl32","legendgroup":"yl32","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"h","width":[0.899999999999999,0.899999999999999,0.9,0.9,0.9,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999],"base":[7,7,12,7,2,1,0,10,5,4,15,5,15,4],"x":[1,3,1,2,1,1,2,3,2,7,5,7,6,1],"y":[18,10,7,1,6,17,14,19,23,15,20,21,22,9],"text":["reorder(sample, number_variants): 1423<br />number_variants:  1<br />genome: yl44","reorder(sample, number_variants): 1425<br />number_variants:  3<br />genome: yl44","reorder(sample, number_variants): 1750<br />number_variants:  1<br />genome: yl44","reorder(sample, number_variants): 1779<br />number_variants:  2<br />genome: yl44","reorder(sample, number_variants): 1783<br />number_variants:  1<br />genome: yl44","reorder(sample, number_variants): 1814<br />number_variants:  1<br />genome: yl44","reorder(sample, number_variants): 1815<br />number_variants:  2<br />genome: yl44","reorder(sample, number_variants): 1880<br />number_variants:  3<br />genome: yl44","reorder(sample, number_variants): 1881<br />number_variants:  2<br />genome: yl44","reorder(sample, number_variants): 1882<br />number_variants:  7<br />genome: yl44","reorder(sample, number_variants): 1883<br />number_variants:  5<br />genome: yl44","reorder(sample, number_variants): 1884<br />number_variants:  7<br />genome: yl44","reorder(sample, number_variants): 1885<br />number_variants:  6<br />genome: yl44","reorder(sample, number_variants): I_cc<br />number_variants:  1<br />genome: yl44"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(206,18,86,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"yl44","legendgroup":"yl44","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"h","width":[0.899999999999999,0.9,0.9,0.9,0.899999999999999,0.9,0.9,0.9,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.9],"base":[3,1,3,2,2,2,0,3,3,1,0,1,0,1,1,0,1,0,2],"x":[2,1,4,10,9,5,2,5,12,5,1,9,5,3,14,5,14,4,13],"y":[11,3,4,7,13,1,6,2,16,8,17,19,23,15,20,21,22,9,5],"text":["reorder(sample, number_variants): 1607<br />number_variants:  2<br />genome: yl45","reorder(sample, number_variants): 1612<br />number_variants:  1<br />genome: yl45","reorder(sample, number_variants): 1660<br />number_variants:  4<br />genome: yl45","reorder(sample, number_variants): 1750<br />number_variants: 10<br />genome: yl45","reorder(sample, number_variants): 1753<br />number_variants:  9<br />genome: yl45","reorder(sample, number_variants): 1779<br />number_variants:  5<br />genome: yl45","reorder(sample, number_variants): 1783<br />number_variants:  2<br />genome: yl45","reorder(sample, number_variants): 1789<br />number_variants:  5<br />genome: yl45","reorder(sample, number_variants): 1801<br />number_variants: 12<br />genome: yl45","reorder(sample, number_variants): 1807<br />number_variants:  5<br />genome: yl45","reorder(sample, number_variants): 1814<br />number_variants:  1<br />genome: yl45","reorder(sample, number_variants): 1880<br />number_variants:  9<br />genome: yl45","reorder(sample, number_variants): 1881<br />number_variants:  5<br />genome: yl45","reorder(sample, number_variants): 1882<br />number_variants:  3<br />genome: yl45","reorder(sample, number_variants): 1883<br />number_variants: 14<br />genome: yl45","reorder(sample, number_variants): 1884<br />number_variants:  5<br />genome: yl45","reorder(sample, number_variants): 1885<br />number_variants: 14<br />genome: yl45","reorder(sample, number_variants): I_cc<br />number_variants:  4<br />genome: yl45","reorder(sample, number_variants): I_mi<br />number_variants: 13<br />genome: yl45"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(150,150,150,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"yl45","legendgroup":"yl45","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"orientation":"h","width":[0.899999999999999,0.899999999999999,0.899999999999999,0.9,0.9,0.899999999999999,0.9,0.899999999999999,0.9,0.9,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.9],"base":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"x":[7,7,3,1,3,3,2,2,2,3,3,1,3,1,1,1,1,2],"y":[18,10,11,3,4,12,7,13,1,2,16,8,24,19,15,20,22,5],"text":["reorder(sample, number_variants): 1423<br />number_variants:  7<br />genome: yl58","reorder(sample, number_variants): 1425<br />number_variants:  7<br />genome: yl58","reorder(sample, number_variants): 1607<br />number_variants:  3<br />genome: yl58","reorder(sample, number_variants): 1612<br />number_variants:  1<br />genome: yl58","reorder(sample, number_variants): 1660<br />number_variants:  3<br />genome: yl58","reorder(sample, number_variants): 1664<br />number_variants:  3<br />genome: yl58","reorder(sample, number_variants): 1750<br />number_variants:  2<br />genome: yl58","reorder(sample, number_variants): 1753<br />number_variants:  2<br />genome: yl58","reorder(sample, number_variants): 1779<br />number_variants:  2<br />genome: yl58","reorder(sample, number_variants): 1789<br />number_variants:  3<br />genome: yl58","reorder(sample, number_variants): 1801<br />number_variants:  3<br />genome: yl58","reorder(sample, number_variants): 1807<br />number_variants:  1<br />genome: yl58","reorder(sample, number_variants): 1877<br />number_variants:  3<br />genome: yl58","reorder(sample, number_variants): 1880<br />number_variants:  1<br />genome: yl58","reorder(sample, number_variants): 1882<br />number_variants:  1<br />genome: yl58","reorder(sample, number_variants): 1883<br />number_variants:  1<br />genome: yl58","reorder(sample, number_variants): 1885<br />number_variants:  1<br />genome: yl58","reorder(sample, number_variants): I_mi<br />number_variants:  2<br />genome: yl58"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(203,201,226,1)","line":{"width":1.88976377952756,"color":"transparent"}},"name":"yl58","legendgroup":"yl58","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":26.2283105022831,"r":7.30593607305936,"b":40.1826484018265,"l":48.9497716894977},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-5.6,117.6],"tickmode":"array","ticktext":["0","30","60","90"],"tickvals":[0,30,60,90],"categoryorder":"array","categoryarray":["0","30","60","90"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":{"text":"number of variants","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.4,24.6],"tickmode":"array","ticktext":["1779","1789","1612","1660","I_mi","1783","1750","1807","I_cc","1425","1607","1664","1753","1815","1882","1801","1814","1423","1880","1883","1884","1885","1881","1877"],"tickvals":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24],"categoryorder":"array","categoryarray":["1779","1789","1612","1660","I_mi","1783","1750","1807","I_cc","1425","1607","1664","1753","1815","1882","1801","1814","1423","1880","1883","1884","1885","1881","1877"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":{"text":"sample","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"y":0.96751968503937},"annotations":[{"text":"genome","x":1.02,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"left","yanchor":"bottom","legendTitle":true}],"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"265777a9e329":{"x":{},"y":{},"fill":{},"type":"bar"}},"cur_data":"265777a9e329","visdat":{"265777a9e329":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 2.1: Number of variants by genome (color) found in samples
</p>
</div>
<p>by mouse and genome</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="Variant-statistics.html#cb9-1"></a>dat<span class="op">$</span>dummy &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb9-2"><a href="Variant-statistics.html#cb9-2"></a>dat.by.sample &lt;-<span class="st"> </span><span class="kw">aggregate</span>(dummy <span class="op">~</span><span class="st"> </span>mouse.id, dat, sum)</span>
<span id="cb9-3"><a href="Variant-statistics.html#cb9-3"></a><span class="kw">colnames</span>(dat.by.sample) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;number_variants&quot;</span>)</span>
<span id="cb9-4"><a href="Variant-statistics.html#cb9-4"></a>DT<span class="op">::</span><span class="kw">datatable</span>(dat.by.sample)</span></code></pre></div>
<div id="htmlwidget-40fc4884317be91b8459" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-40fc4884317be91b8459">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24"],["1423","1425","1607","1612","1660","1664","1750","1753","1779","1783","1789","1801","1807","1814","1815","1877","1880","1881","1882","1883","1884","1885","I_cc","I_mi"],[69,74,65,52,67,66,63,77,54,62,75,85,63,81,55,59,112,87,77,98,84,98,63,110]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>sample<\/th>\n      <th>number_variants<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="Variant-statistics.html#cb10-1"></a>p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(dat.by.sample, <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">reorder</span>(sample, number_variants), <span class="dt">y =</span> number_variants))</span>
<span id="cb10-2"><a href="Variant-statistics.html#cb10-2"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">coord_flip</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>()</span>
<span id="cb10-3"><a href="Variant-statistics.html#cb10-3"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;number of variants&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;sample&quot;</span>)</span>
<span id="cb10-4"><a href="Variant-statistics.html#cb10-4"></a>plotly<span class="op">::</span><span class="kw">ggplotly</span>(p)</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-9"></span>
<div id="htmlwidget-897fff54aa34d7dfd520" style="width:672px;height:480px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-897fff54aa34d7dfd520">{"x":{"data":[{"orientation":"h","width":[0.899999999999999,0.899999999999999,0.899999999999999,0.9,0.899999999999999,0.899999999999999,0.9,0.899999999999999,0.9,0.9,0.899999999999999,0.899999999999999,0.9,0.899999999999999,0.9,0.9,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999,0.899999999999999],"base":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"x":[69,74,65,52,67,66,63,77,54,62,75,85,63,81,55,59,112,87,77,98,84,98,63,110],"y":[12,13,9,1,11,10,6,15,2,5,14,19,7,17,3,4,24,20,16,21,18,22,8,23],"text":["reorder(sample, number_variants): 1423<br />number_variants:  69","reorder(sample, number_variants): 1425<br />number_variants:  74","reorder(sample, number_variants): 1607<br />number_variants:  65","reorder(sample, number_variants): 1612<br />number_variants:  52","reorder(sample, number_variants): 1660<br />number_variants:  67","reorder(sample, number_variants): 1664<br />number_variants:  66","reorder(sample, number_variants): 1750<br />number_variants:  63","reorder(sample, number_variants): 1753<br />number_variants:  77","reorder(sample, number_variants): 1779<br />number_variants:  54","reorder(sample, number_variants): 1783<br />number_variants:  62","reorder(sample, number_variants): 1789<br />number_variants:  75","reorder(sample, number_variants): 1801<br />number_variants:  85","reorder(sample, number_variants): 1807<br />number_variants:  63","reorder(sample, number_variants): 1814<br />number_variants:  81","reorder(sample, number_variants): 1815<br />number_variants:  55","reorder(sample, number_variants): 1877<br />number_variants:  59","reorder(sample, number_variants): 1880<br />number_variants: 112","reorder(sample, number_variants): 1881<br />number_variants:  87","reorder(sample, number_variants): 1882<br />number_variants:  77","reorder(sample, number_variants): 1883<br />number_variants:  98","reorder(sample, number_variants): 1884<br />number_variants:  84","reorder(sample, number_variants): 1885<br />number_variants:  98","reorder(sample, number_variants): I_cc<br />number_variants:  63","reorder(sample, number_variants): I_mi<br />number_variants: 110"],"type":"bar","marker":{"autocolorscale":false,"color":"rgba(89,89,89,1)","line":{"width":1.88976377952756,"color":"transparent"}},"showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":26.2283105022831,"r":7.30593607305936,"b":40.1826484018265,"l":48.9497716894977},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-5.6,117.6],"tickmode":"array","ticktext":["0","30","60","90"],"tickvals":[0,30,60,90],"categoryorder":"array","categoryarray":["0","30","60","90"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":{"text":"sample","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.4,24.6],"tickmode":"array","ticktext":["1612","1779","1815","1877","1783","1750","1807","I_cc","1607","1664","1660","1423","1425","1789","1753","1882","1814","1884","1801","1881","1883","1885","I_mi","1880"],"tickvals":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24],"categoryorder":"array","categoryarray":["1612","1779","1815","1877","1783","1750","1807","I_cc","1607","1664","1660","1423","1425","1789","1753","1882","1814","1884","1801","1881","1883","1885","I_mi","1880"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":{"text":"number of variants","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"265757a851b7":{"x":{},"y":{},"type":"bar"}},"cur_data":"265757a851b7","visdat":{"265757a851b7":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 2.2: number of variants of all 12 OMM genomes by mouse
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="Setup.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Allele-frequency.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["oligomm-claudia.pdf", "oligomm-claudia.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
